---
title: "CarexComparativeGenomics"
author: "Marcial Escudero"
date: "7/13/2021"
output: html_document
editor_options: 
  chunk_output_type: console
---

Lerning with the tutorial

```{r}
library(RCircos)

data(RCircos.Histogram.Data)
head(RCircos.Histogram.Data)

data(RCircos.Heatmap.Data)
head(RCircos.Heatmap.Data)

data(RCircos.Link.Data)
head(RCircos.Link.Data)

data(UCSC.HG19.Human.CytoBandIdeogram)
head(UCSC.HG19.Human.CytoBandIdeogram)

data(UCSC.Mouse.GRCm38.CytoBandIdeogram)
head(UCSC.Mouse.GRCm38.CytoBandIdeogram)

data(UCSC.Baylor.3.4.Rat.cytoBandIdeogram)
head(UCSC.Baylor.3.4.Rat.cytoBandIdeogram)


chr.exclude <- NULL
cyto.info <- UCSC.HG19.Human.CytoBandIdeogram
cyto.info <- RCircos.Gene.Label.Data
tracks.inside <- 10
tracks.outside <- 0
RCircos.Set.Core.Components(cyto.info, chr.exclude,tracks.inside, tracks.outside)

rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.cyto <- RCircos.Get.Plot.Ideogram()
rcircos.position <- RCircos.Get.Plot.Positions()
RCircos.List.Plot.Parameters()

rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.params$base.per.unit <- 3000
RCircos.Reset.Plot.Parameters(rcircos.params)
RCircos.List.Plot.Parameters()

par(mai=c(0.25, 0.25, 0.25, 0.25))
plot.new()
plot.window(c(-2.5,2.5), c(-2.5, 2.5))

out.file <- "RCircosDemoHumanGenome.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)

RCircos.Set.Plot.Area()

RCircos.Chromosome.Ideogram.Plot()


data(RCircos.Gene.Label.Data)
head(RCircos.Gene.Label.Data)
name.col <- 4
side <- "in"
track.num <- 1
RCircos.Gene.Connector.Plot(RCircos.Gene.Label.Data,track.num, side)
track.num <- 2
RCircos.Gene.Name.Plot(RCircos.Gene.Label.Data,name.col,track.num, side)

data(RCircos.Heatmap.Data)
data.col <- 6
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(RCircos.Heatmap.Data, data.col, track.num, side)
data(RCircos.Scatter.Data)
data.col <- 5
track.num <- 6
side <- "in"
by.fold <- 1
RCircos.Scatter.Plot(RCircos.Scatter.Data, data.col,  track.num, side, by.fold)
data(RCircos.Line.Data)
data.col <- 5
track.num <- 7
side <- "in"
RCircos.Line.Plot(RCircos.Line.Data, data.col,track.num, side)
data(RCircos.Histogram.Data)
data.col <- 4
track.num <- 8
side <- "in"
RCircos.Histogram.Plot(RCircos.Histogram.Data,data.col, track.num, side)
data(RCircos.Tile.Data)
track.num <- 9
side <- "in"
RCircos.Tile.Plot(RCircos.Tile.Data, track.num, side)


data(RCircos.Link.Data)
track.num <- 11
RCircos.Link.Plot(RCircos.Link.Data, track.num, TRUE)
data(RCircos.Ribbon.Data)
RCircos.Ribbon.Plot(ribbon.data=RCircos.Ribbon.Data,track.num=11, by.chromosome=FALSE, twist=FALSE)


dev.off()

```


Opening some Carex data

```{r}

both <- read.csv("both.csv")

both

chr.exclude <- NULL
cyto.info <- both
tracks.inside <- 0
tracks.outside <- 0
RCircos.Set.Core.Components(cyto.info, chr.exclude,tracks.inside, tracks.outside)


rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.cyto <- RCircos.Get.Plot.Ideogram()
rcircos.position <- RCircos.Get.Plot.Positions()
RCircos.List.Plot.Parameters()

 

rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.params$base.per.unit <- 3000
RCircos.Reset.Plot.Parameters(rcircos.params)
RCircos.List.Plot.Parameters()

RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()



```

Setting up the bowtie2 analysis

```{r}

readsdata <- read.csv("batch_3.catalog.tags.tsv", sep = "")

readsdata[,3] <- paste0("M", readsdata[,3])

readsdata <- readsdata[,c(3,9)]

readsdata2 <- readsdata[readsdata[,1]%in%both[,4],]

write.csv(readsdata2, file = "reads.csv")



```


Making the comparative genomics analyses
```{r}

bowtie2 <- read.csv("bowtie2.csv", sep = "")
bowtie2

both

links <- merge(bowtie2, both, by.x = "Marker", by.y = "Band", sort = F)
links <- links[,c(2,3,4,5,6,7)]
links

RCircos.Link.Plot(links, track.num, TRUE)

out.file <- "ComparativeGenomicsLG1.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG1",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG2.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG2",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG3.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG3",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG4.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG4",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG5.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG5",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG6.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG6",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG7.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG7",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG8.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG8",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG9.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG9",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG10.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG10",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG11.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG11",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG12.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG12",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG13.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG13",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG14.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG14",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG15.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG15",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG16.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG16",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG17.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG17",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG18.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG18",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG19.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG19",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG20.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG20",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG21.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG21",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG22.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG22",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG23.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG23",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG24.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG24",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG25.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG25",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG26.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG26",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG27.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG27",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG28.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG28",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG29.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG29",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG30.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG30",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG31.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG31",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG32.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG32",], track.num, TRUE)
dev.off()




out.file <- "ComparativeGenomicschr1.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr1",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr2.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr2",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr3.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr3",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr4.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr4",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr5.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr5",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr6.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr6",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr7.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr7",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr8.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr8",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr9.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr9",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr10.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr10",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr11.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr11",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr12.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr12",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr13.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr13",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr14.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr14",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr15.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr15",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr16.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr16",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr17.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr17",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr18.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr18",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr19.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr19",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr20.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr20",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr21.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr21",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr22.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr22",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr23.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr23",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr24.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr24",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr25.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr25",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr26.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr26",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr27.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr27",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr28.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr28",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr29.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr29",], track.num, TRUE)
dev.off()

```

Let's try to plot something that could be publish

```{r}

library(RColorBrewer)
n <- 30
qual_col_pals = brewer.pal.info[brewer.pal.info$category == 'qual',]
col_vector = unlist(mapply(brewer.pal, qual_col_pals$maxcolors, rownames(qual_col_pals)))
pie(rep(1,n), col=sample(col_vector, n))

col_vector[1:30]
col_vector <- c(col_vector[1:30],rep(col_vector[30],1381))

length(col_vector)
dim (both)

library(RCircos)
both <- read.csv("both.csv")

RCircos.Get.Plot.Colors(both, col_vector)

both$PlotColor <- col_vector

chr.exclude <- NULL
cyto.info <- both
tracks.inside <- 0
tracks.outside <- 0
RCircos.Set.Core.Components(cyto.info, chr.exclude,tracks.inside, tracks.outside)


rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.params
rcircos.cyto <- RCircos.Get.Plot.Ideogram()
rcircos.cyto
rcircos.position <- RCircos.Get.Plot.Positions()
rcircos.position
RCircos.List.Plot.Parameters()

#rcircos.params <- RCircos.Get.Plot.Parameters()
#rcircos.params$base.per.unit <- 3000
#RCircos.Reset.Plot.Parameters(rcircos.params)
#RCircos.List.Plot.Parameters()


rcircos.cyto
rcircos.cyto$BandColor <- rcircos.cyto$PlotColor
rcircos.cyto$ChrColor <-  rcircos.cyto$PlotColor
RCircos.Reset.Plot.Ideogram(rcircos.cyto)
RCircos.List.Plot.Parameters()


RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()


links$PlotColor <- 1
links[links[,1]=="chr1", "PlotColor"] <- col_vector[1]
links[links[,1]=="chr2", "PlotColor"] <- col_vector[2]
links[links[,1]=="chr3", "PlotColor"] <- col_vector[3]
links[links[,1]=="chr4", "PlotColor"] <- col_vector[4]
links[links[,1]=="chr5", "PlotColor"] <- col_vector[5]
links[links[,1]=="chr6", "PlotColor"] <- col_vector[6]
links[links[,1]=="chr7", "PlotColor"] <- col_vector[7]
links[links[,1]=="chr8", "PlotColor"] <- col_vector[8]
links[links[,1]=="chr9", "PlotColor"] <- col_vector[9]
links[links[,1]=="chr10", "PlotColor"] <- col_vector[10]
links[links[,1]=="chr11", "PlotColor"] <- col_vector[11]
links[links[,1]=="chr12", "PlotColor"] <- col_vector[12]
links[links[,1]=="chr13", "PlotColor"] <- col_vector[13]
links[links[,1]=="chr14", "PlotColor"] <- col_vector[14]
links[links[,1]=="chr15", "PlotColor"] <- col_vector[15]
links[links[,1]=="chr16", "PlotColor"] <- col_vector[16]
links[links[,1]=="chr17", "PlotColor"] <- col_vector[17]
links[links[,1]=="chr18", "PlotColor"] <- col_vector[18]
links[links[,1]=="chr19", "PlotColor"] <- col_vector[19]
links[links[,1]=="chr20", "PlotColor"] <- col_vector[20]
links[links[,1]=="chr21", "PlotColor"] <- col_vector[21]
links[links[,1]=="chr22", "PlotColor"] <- col_vector[22]
links[links[,1]=="chr23", "PlotColor"] <- col_vector[23]
links[links[,1]=="chr24", "PlotColor"] <- col_vector[24]
links[links[,1]=="chr25", "PlotColor"] <- col_vector[25]
links[links[,1]=="chr26", "PlotColor"] <- col_vector[26]
links[links[,1]=="chr27", "PlotColor"] <- col_vector[27]
links[links[,1]=="chr28", "PlotColor"] <- col_vector[28]
links[links[,1]=="chr29", "PlotColor"] <- col_vector[29]

RCircos.Link.Plot(links, track.num = 1)



```


Now, let's check the result with the reads that align only once

```{r}

library(RCircos)
both <- read.csv("both.csv")

RCircos.Get.Plot.Colors(both, col_vector)

both$PlotColor <- col_vector

chr.exclude <- NULL
cyto.info <- both
tracks.inside <- 0
tracks.outside <- 0
RCircos.Set.Core.Components(cyto.info, chr.exclude,tracks.inside, tracks.outside)


rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.params
rcircos.cyto <- RCircos.Get.Plot.Ideogram()
rcircos.cyto
rcircos.position <- RCircos.Get.Plot.Positions()
rcircos.position
RCircos.List.Plot.Parameters()

#rcircos.params <- RCircos.Get.Plot.Parameters()
#rcircos.params$base.per.unit <- 3000
#RCircos.Reset.Plot.Parameters(rcircos.params)
#RCircos.List.Plot.Parameters()


rcircos.cyto
rcircos.cyto$BandColor <- rcircos.cyto$PlotColor
rcircos.cyto$ChrColor <-  rcircos.cyto$PlotColor
RCircos.Reset.Plot.Ideogram(rcircos.cyto)
RCircos.List.Plot.Parameters()


RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()


bowtie2dup <- read.csv("bowtie2dup.csv", sep = "")
bowtie2dup
dim(bowtie2dup)
duplicatedreads <- bowtie2dup$Marker[duplicated(bowtie2dup$Marker)]
noduplicatedreads <- setdiff(bowtie2dup$Marker,duplicatedreads)
noduplicatedreads

bowtie2nodup <-  bowtie2dup[bowtie2dup$Marker %in% noduplicatedreads,]
bowtie2nodup


both

linksnodup <- merge(bowtie2nodup, both, by.x = "Marker", by.y = "Band", sort = F)
linksnodup <- linksnodup[,c(2,3,4,5,6,7)]
linksnodup

RCircos.Link.Plot(linksnodup, track.num, TRUE)






linksnodup$PlotColor <- 1
linksnodup[linksnodup[,1]=="chr1", "PlotColor"] <- col_vector[1]
linksnodup[linksnodup[,1]=="chr2", "PlotColor"] <- col_vector[2]
linksnodup[linksnodup[,1]=="chr3", "PlotColor"] <- col_vector[3]
linksnodup[linksnodup[,1]=="chr4", "PlotColor"] <- col_vector[4]
linksnodup[linksnodup[,1]=="chr5", "PlotColor"] <- col_vector[5]
linksnodup[linksnodup[,1]=="chr6", "PlotColor"] <- col_vector[6]
linksnodup[linksnodup[,1]=="chr7", "PlotColor"] <- col_vector[7]
linksnodup[linksnodup[,1]=="chr8", "PlotColor"] <- col_vector[8]
linksnodup[linksnodup[,1]=="chr9", "PlotColor"] <- col_vector[9]
linksnodup[linksnodup[,1]=="chr10", "PlotColor"] <- col_vector[10]
linksnodup[linksnodup[,1]=="chr11", "PlotColor"] <- col_vector[11]
linksnodup[linksnodup[,1]=="chr12", "PlotColor"] <- col_vector[12]
linksnodup[linksnodup[,1]=="chr13", "PlotColor"] <- col_vector[13]
linksnodup[linksnodup[,1]=="chr14", "PlotColor"] <- col_vector[14]
linksnodup[linksnodup[,1]=="chr15", "PlotColor"] <- col_vector[15]
linksnodup[linksnodup[,1]=="chr16", "PlotColor"] <- col_vector[16]
linksnodup[linksnodup[,1]=="chr17", "PlotColor"] <- col_vector[17]
linksnodup[linksnodup[,1]=="chr18", "PlotColor"] <- col_vector[18]
linksnodup[linksnodup[,1]=="chr19", "PlotColor"] <- col_vector[19]
linksnodup[linksnodup[,1]=="chr20", "PlotColor"] <- col_vector[20]
linksnodup[linksnodup[,1]=="chr21", "PlotColor"] <- col_vector[21]
linksnodup[linksnodup[,1]=="chr22", "PlotColor"] <- col_vector[22]
linksnodup[linksnodup[,1]=="chr23", "PlotColor"] <- col_vector[23]
linksnodup[linksnodup[,1]=="chr24", "PlotColor"] <- col_vector[24]
linksnodup[linksnodup[,1]=="chr25", "PlotColor"] <- col_vector[25]
linksnodup[linksnodup[,1]=="chr26", "PlotColor"] <- col_vector[26]
linksnodup[linksnodup[,1]=="chr27", "PlotColor"] <- col_vector[27]
linksnodup[linksnodup[,1]=="chr28", "PlotColor"] <- col_vector[28]
linksnodup[linksnodup[,1]=="chr29", "PlotColor"] <- col_vector[29]

RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup, track.num = 1)

out.file <- "ComparativeGenomicsLG1nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG1",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG2nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG2",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG3nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG3",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG4nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG4",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG5nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG5",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG6nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG6",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG7nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG7",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG8nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG8",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG9nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG9",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG10nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG10",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG11nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG11",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG12nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG12",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG13nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG13",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG14nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG14",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG15nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG15",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG16nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG16",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG17nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG17",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG18nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG18",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG19nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG19",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG20nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG20",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG21nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG21",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG22nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG22",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG23nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG23",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG24nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG24",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG25nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG25",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG26nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG26",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG27nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG27",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG28nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG28",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG29nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG29",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG30nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG30",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG31nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG31",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG32nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG32",], track.num, TRUE)
dev.off()




out.file <- "ComparativeGenomicschr1nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr1",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr2nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr2",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr3nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr3",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr4nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr4",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr5nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr5",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr6nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr6",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr7nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr7",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr8nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr8",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr9nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr9",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr10nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr10",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr11nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr11",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr12nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr12",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr13nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr13",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr14nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr14",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr15nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr15",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr16nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr16",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr17nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr17",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr18nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr18",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr19nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr19",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr20nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr20",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr21nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr21",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr22nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr22",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr23nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr23",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr24nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr24",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr25nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr25",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr26nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr26",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr27nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr27",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr28nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr28",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr29nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr29",], track.num, TRUE)
dev.off()


```


Defining blocks!

```{r}

chr1 <- linksnodup[linksnodup$Chromosome.x=="chr1",]
chr1 <- chr1[order(chr1$chromStart),]
chr1
#chr1blocks <- chr1[c(2,4,6),c(1,2,4)] 

chr23 <- linksnodup[linksnodup$Chromosome.x=="chr23",]
chr23 <- chr23[order(chr23$chromStart),]
chr23

chr24 <- linksnodup[linksnodup$Chromosome.x=="chr24",]
chr24 <- chr24[order(chr24$chromStart),]
chr24
chr24blocks <- chr24[c(1,8),c(1,2,3,4)] 
chr24blocks
chr24blocks[1,3] <- chr24[8,2]-1
chr24blocks[2,3] <- chr24[18,2]
chr24blocks[1,4] <- "unsorted"
chr24blocks[2,4] <- "conserved"
chr24blocks

chr25 <- linksnodup[linksnodup$Chromosome.x=="chr25",]
chr25 <- chr25[order(chr25$chromStart),]
chr25
chr25blocks <- chr25[c(1,3,7,9),c(1,2,3,4)] 
chr25blocks[1,3] <- chr25[2,3]
chr25blocks[2,3] <- chr25[6,3]
chr25blocks[3,2] <- chr25[6,3] + 1
chr25blocks[3,3] <- chr25[9,2] - 1
chr25blocks[3,4] <- "fisfus"
chr25blocks[4,3] <- chr25[15,3]
newrow <- c("chr25", 325961, 1321736, "fisfus")
#insertRow <- function(existingDF, newrow, r) {
#  existingDF[seq(r+1,nrow(existingDF)+1),] <- existingDF[seq(r,nrow(existingDF)),]
#  existingDF[r,] <- newrow
#  existingDF
#}
chr25blocks <- insertRow(chr25blocks, newrow, 2)
chr25blocks[1,4] <- "maybeconserved"
chr25blocks[3,4] <- "conserved"
chr25blocks[5,4] <- "conserved"
chr25blocks
#2 conserved block and 3rd possible, and two rearrangementes between then

chr26 <- linksnodup[linksnodup$Chromosome.x=="chr26",]
chr26 <- chr26[order(chr26$chromStart),]
chr26
chr26blocks <- chr26[c(1,5),c(1,2,3,4)] 
chr26blocks[1,3] <- chr26[4,2]
chr26blocks[2,2] <- chr26[4,2] + 1
chr26blocks[2,3] <- chr26[8,2]
chr26blocks[2,4] <- "complex"
chr26blocks[1,4] <- "conserved"
chr26blocks
#single conserved block and area with possible multiple rearrangements

chr27 <- linksnodup[linksnodup$Chromosome.x=="chr27",]
chr27 <- chr27[order(chr27$chromStart),]
chr27
chr27blocks <- chr27[c(1,4),c(1,2,3,4)] 
chr27blocks[1,3] <- chr27[3,2]
chr27blocks[1,4] <- "conserved"
chr27blocks[2,3] <- chr27[6,2]
chr27blocks[2,4] <- "conserved"
chr27blocks <- rbind(chr27blocks[1,],c("chr27", 3696901,4879022,"fisfus"),chr27blocks[2,])
chr27blocks
#two conserved blocks detected, 1 rearrangement between both

chr28 <- linksnodup[linksnodup$Chromosome.x=="chr28",]
chr28 <- chr28[order(chr28$chromStart),]
chr28
#single block

chr29 <- linksnodup[linksnodup$Chromosome.x=="chr29",]
chr29 <- chr29[order(chr29$chromStart),]
chr29
#single block


```

