---
title: "CarexComparativeGenomics"
author: "Marcial Escudero"
date: "7/13/2021"
output: html_document
editor_options: 
  chunk_output_type: console
---

Lerning with the tutorial

```{r}
library(RCircos)

data(RCircos.Histogram.Data)
head(RCircos.Histogram.Data)

data(RCircos.Heatmap.Data)
head(RCircos.Heatmap.Data)

data(RCircos.Link.Data)
head(RCircos.Link.Data)

data(UCSC.HG19.Human.CytoBandIdeogram)
head(UCSC.HG19.Human.CytoBandIdeogram)

data(UCSC.Mouse.GRCm38.CytoBandIdeogram)
head(UCSC.Mouse.GRCm38.CytoBandIdeogram)

data(UCSC.Baylor.3.4.Rat.cytoBandIdeogram)
head(UCSC.Baylor.3.4.Rat.cytoBandIdeogram)


chr.exclude <- NULL
cyto.info <- UCSC.HG19.Human.CytoBandIdeogram
cyto.info <- RCircos.Gene.Label.Data
tracks.inside <- 10
tracks.outside <- 0
RCircos.Set.Core.Components(cyto.info, chr.exclude,tracks.inside, tracks.outside)

rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.cyto <- RCircos.Get.Plot.Ideogram()
rcircos.position <- RCircos.Get.Plot.Positions()
RCircos.List.Plot.Parameters()

rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.params$base.per.unit <- 3000
RCircos.Reset.Plot.Parameters(rcircos.params)
RCircos.List.Plot.Parameters()

par(mai=c(0.25, 0.25, 0.25, 0.25))
plot.new()
plot.window(c(-2.5,2.5), c(-2.5, 2.5))

out.file <- "RCircosDemoHumanGenome.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)

RCircos.Set.Plot.Area()

RCircos.Chromosome.Ideogram.Plot()


data(RCircos.Gene.Label.Data)
head(RCircos.Gene.Label.Data)
name.col <- 4
side <- "in"
track.num <- 1
RCircos.Gene.Connector.Plot(RCircos.Gene.Label.Data,track.num, side)
track.num <- 2
RCircos.Gene.Name.Plot(RCircos.Gene.Label.Data,name.col,track.num, side)

data(RCircos.Heatmap.Data)
data.col <- 6
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(RCircos.Heatmap.Data, data.col, track.num, side)
data(RCircos.Scatter.Data)
data.col <- 5
track.num <- 6
side <- "in"
by.fold <- 1
RCircos.Scatter.Plot(RCircos.Scatter.Data, data.col,  track.num, side, by.fold)
data(RCircos.Line.Data)
data.col <- 5
track.num <- 7
side <- "in"
RCircos.Line.Plot(RCircos.Line.Data, data.col,track.num, side)
data(RCircos.Histogram.Data)
data.col <- 4
track.num <- 8
side <- "in"
RCircos.Histogram.Plot(RCircos.Histogram.Data,data.col, track.num, side)
data(RCircos.Tile.Data)
track.num <- 9
side <- "in"
RCircos.Tile.Plot(RCircos.Tile.Data, track.num, side)


data(RCircos.Link.Data)
track.num <- 11
RCircos.Link.Plot(RCircos.Link.Data, track.num, TRUE)
data(RCircos.Ribbon.Data)
RCircos.Ribbon.Plot(ribbon.data=RCircos.Ribbon.Data,track.num=11, by.chromosome=FALSE, twist=FALSE)


dev.off()

```


Opening some Carex data

```{r}

both <- read.csv("both.csv")

both

chr.exclude <- NULL
cyto.info <- both
tracks.inside <- 0
tracks.outside <- 0
RCircos.Set.Core.Components(cyto.info, chr.exclude,tracks.inside, tracks.outside)


rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.cyto <- RCircos.Get.Plot.Ideogram()
rcircos.position <- RCircos.Get.Plot.Positions()
RCircos.List.Plot.Parameters()

 

rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.params$base.per.unit <- 3000
RCircos.Reset.Plot.Parameters(rcircos.params)
RCircos.List.Plot.Parameters()

RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()



```

Setting up the bowtie2 analysis

```{r}

readsdata <- read.csv("batch_3.catalog.tags.tsv", sep = "")

readsdata[,3] <- paste0("M", readsdata[,3])

readsdata <- readsdata[,c(3,9)]

readsdata2 <- readsdata[readsdata[,1]%in%both[,4],]

write.csv(readsdata2, file = "reads.csv")



```


Making the comparative genomics analyses
```{r}

bowtie2 <- read.csv("bowtie2.csv", sep = "")
bowtie2

both

links <- merge(bowtie2, both, by.x = "Marker", by.y = "Band", sort = F)
links <- links[,c(2,3,4,5,6,7)]
links

RCircos.Link.Plot(links, track.num, TRUE)

out.file <- "ComparativeGenomicsLG1.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG1",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG2.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG2",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG3.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG3",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG4.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG4",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG5.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG5",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG6.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG6",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG7.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG7",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG8.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG8",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG9.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG9",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG10.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG10",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG11.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG11",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG12.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG12",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG13.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG13",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG14.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG14",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG15.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG15",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG16.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG16",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG17.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG17",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG18.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG18",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG19.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG19",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG20.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG20",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG21.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG21",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG22.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG22",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG23.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG23",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG24.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG24",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG25.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG25",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG26.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG26",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG27.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG27",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG28.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG28",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG29.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG29",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG30.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG30",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG31.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG31",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG32.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG32",], track.num, TRUE)
dev.off()




out.file <- "ComparativeGenomicschr1.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr1",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr2.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr2",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr3.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr3",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr4.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr4",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr5.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr5",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr6.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr6",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr7.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr7",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr8.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr8",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr9.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr9",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr10.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr10",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr11.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr11",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr12.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr12",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr13.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr13",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr14.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr14",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr15.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr15",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr16.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr16",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr17.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr17",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr18.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr18",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr19.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr19",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr20.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr20",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr21.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr21",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr22.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr22",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr23.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr23",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr24.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr24",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr25.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr25",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr26.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr26",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr27.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr27",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr28.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr28",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr29.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr29",], track.num, TRUE)
dev.off()

```

Let's try to plot something that could be publish

```{r}

library(RColorBrewer)
n <- 30
qual_col_pals = brewer.pal.info[brewer.pal.info$category == 'qual',]
col_vector = unlist(mapply(brewer.pal, qual_col_pals$maxcolors, rownames(qual_col_pals)))
pie(rep(1,n), col=sample(col_vector, n))

col_vector[1:30]
col_vector <- c(col_vector[1:30],rep(col_vector[30],1381))

length(col_vector)
dim (both)

library(RCircos)
both <- read.csv("both.csv")

RCircos.Get.Plot.Colors(both, col_vector)

both$PlotColor <- col_vector

chr.exclude <- NULL
cyto.info <- both
tracks.inside <- 0
tracks.outside <- 0
RCircos.Set.Core.Components(cyto.info, chr.exclude,tracks.inside, tracks.outside)


rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.params
rcircos.cyto <- RCircos.Get.Plot.Ideogram()
rcircos.cyto
rcircos.position <- RCircos.Get.Plot.Positions()
rcircos.position
RCircos.List.Plot.Parameters()

#rcircos.params <- RCircos.Get.Plot.Parameters()
#rcircos.params$base.per.unit <- 3000
#RCircos.Reset.Plot.Parameters(rcircos.params)
#RCircos.List.Plot.Parameters()


rcircos.cyto
rcircos.cyto$BandColor <- rcircos.cyto$PlotColor
rcircos.cyto$ChrColor <-  rcircos.cyto$PlotColor
RCircos.Reset.Plot.Ideogram(rcircos.cyto)
RCircos.List.Plot.Parameters()


RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()


links$PlotColor <- 1
links[links[,1]=="chr1", "PlotColor"] <- col_vector[1]
links[links[,1]=="chr2", "PlotColor"] <- col_vector[2]
links[links[,1]=="chr3", "PlotColor"] <- col_vector[3]
links[links[,1]=="chr4", "PlotColor"] <- col_vector[4]
links[links[,1]=="chr5", "PlotColor"] <- col_vector[5]
links[links[,1]=="chr6", "PlotColor"] <- col_vector[6]
links[links[,1]=="chr7", "PlotColor"] <- col_vector[7]
links[links[,1]=="chr8", "PlotColor"] <- col_vector[8]
links[links[,1]=="chr9", "PlotColor"] <- col_vector[9]
links[links[,1]=="chr10", "PlotColor"] <- col_vector[10]
links[links[,1]=="chr11", "PlotColor"] <- col_vector[11]
links[links[,1]=="chr12", "PlotColor"] <- col_vector[12]
links[links[,1]=="chr13", "PlotColor"] <- col_vector[13]
links[links[,1]=="chr14", "PlotColor"] <- col_vector[14]
links[links[,1]=="chr15", "PlotColor"] <- col_vector[15]
links[links[,1]=="chr16", "PlotColor"] <- col_vector[16]
links[links[,1]=="chr17", "PlotColor"] <- col_vector[17]
links[links[,1]=="chr18", "PlotColor"] <- col_vector[18]
links[links[,1]=="chr19", "PlotColor"] <- col_vector[19]
links[links[,1]=="chr20", "PlotColor"] <- col_vector[20]
links[links[,1]=="chr21", "PlotColor"] <- col_vector[21]
links[links[,1]=="chr22", "PlotColor"] <- col_vector[22]
links[links[,1]=="chr23", "PlotColor"] <- col_vector[23]
links[links[,1]=="chr24", "PlotColor"] <- col_vector[24]
links[links[,1]=="chr25", "PlotColor"] <- col_vector[25]
links[links[,1]=="chr26", "PlotColor"] <- col_vector[26]
links[links[,1]=="chr27", "PlotColor"] <- col_vector[27]
links[links[,1]=="chr28", "PlotColor"] <- col_vector[28]
links[links[,1]=="chr29", "PlotColor"] <- col_vector[29]

RCircos.Link.Plot(links, track.num = 1)



```


Now, let's check the result with the reads that align only once

```{r}

library(RCircos)
both <- read.csv("both.csv")

RCircos.Get.Plot.Colors(both, col_vector)

both$PlotColor <- col_vector

chr.exclude <- NULL
cyto.info <- both
tracks.inside <- 0
tracks.outside <- 0
RCircos.Set.Core.Components(cyto.info, chr.exclude,tracks.inside, tracks.outside)


rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.params
rcircos.cyto <- RCircos.Get.Plot.Ideogram()
rcircos.cyto
rcircos.position <- RCircos.Get.Plot.Positions()
rcircos.position
RCircos.List.Plot.Parameters()

#rcircos.params <- RCircos.Get.Plot.Parameters()
#rcircos.params$base.per.unit <- 3000
#RCircos.Reset.Plot.Parameters(rcircos.params)
#RCircos.List.Plot.Parameters()


rcircos.cyto
rcircos.cyto$BandColor <- rcircos.cyto$PlotColor
rcircos.cyto$ChrColor <-  rcircos.cyto$PlotColor
RCircos.Reset.Plot.Ideogram(rcircos.cyto)
RCircos.List.Plot.Parameters()


RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()


bowtie2dup <- read.csv("bowtie2dup.csv", sep = "")
bowtie2dup
dim(bowtie2dup)
duplicatedreads <- bowtie2dup$Marker[duplicated(bowtie2dup$Marker)]
noduplicatedreads <- setdiff(bowtie2dup$Marker,duplicatedreads)
noduplicatedreads

bowtie2nodup <-  bowtie2dup[bowtie2dup$Marker %in% noduplicatedreads,]
bowtie2nodup


both

linksnodup <- merge(bowtie2nodup, both, by.x = "Marker", by.y = "Band", sort = F)
linksnodup <- linksnodup[,c(2,3,4,5,6,7)]
linksnodup

RCircos.Link.Plot(linksnodup, track.num, TRUE)






linksnodup$PlotColor <- 1
linksnodup[linksnodup[,1]=="chr1", "PlotColor"] <- col_vector[1]
linksnodup[linksnodup[,1]=="chr2", "PlotColor"] <- col_vector[2]
linksnodup[linksnodup[,1]=="chr3", "PlotColor"] <- col_vector[3]
linksnodup[linksnodup[,1]=="chr4", "PlotColor"] <- col_vector[4]
linksnodup[linksnodup[,1]=="chr5", "PlotColor"] <- col_vector[5]
linksnodup[linksnodup[,1]=="chr6", "PlotColor"] <- col_vector[6]
linksnodup[linksnodup[,1]=="chr7", "PlotColor"] <- col_vector[7]
linksnodup[linksnodup[,1]=="chr8", "PlotColor"] <- col_vector[8]
linksnodup[linksnodup[,1]=="chr9", "PlotColor"] <- col_vector[9]
linksnodup[linksnodup[,1]=="chr10", "PlotColor"] <- col_vector[10]
linksnodup[linksnodup[,1]=="chr11", "PlotColor"] <- col_vector[11]
linksnodup[linksnodup[,1]=="chr12", "PlotColor"] <- col_vector[12]
linksnodup[linksnodup[,1]=="chr13", "PlotColor"] <- col_vector[13]
linksnodup[linksnodup[,1]=="chr14", "PlotColor"] <- col_vector[14]
linksnodup[linksnodup[,1]=="chr15", "PlotColor"] <- col_vector[15]
linksnodup[linksnodup[,1]=="chr16", "PlotColor"] <- col_vector[16]
linksnodup[linksnodup[,1]=="chr17", "PlotColor"] <- col_vector[17]
linksnodup[linksnodup[,1]=="chr18", "PlotColor"] <- col_vector[18]
linksnodup[linksnodup[,1]=="chr19", "PlotColor"] <- col_vector[19]
linksnodup[linksnodup[,1]=="chr20", "PlotColor"] <- col_vector[20]
linksnodup[linksnodup[,1]=="chr21", "PlotColor"] <- col_vector[21]
linksnodup[linksnodup[,1]=="chr22", "PlotColor"] <- col_vector[22]
linksnodup[linksnodup[,1]=="chr23", "PlotColor"] <- col_vector[23]
linksnodup[linksnodup[,1]=="chr24", "PlotColor"] <- col_vector[24]
linksnodup[linksnodup[,1]=="chr25", "PlotColor"] <- col_vector[25]
linksnodup[linksnodup[,1]=="chr26", "PlotColor"] <- col_vector[26]
linksnodup[linksnodup[,1]=="chr27", "PlotColor"] <- col_vector[27]
linksnodup[linksnodup[,1]=="chr28", "PlotColor"] <- col_vector[28]
linksnodup[linksnodup[,1]=="chr29", "PlotColor"] <- col_vector[29]

RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup, track.num = 1)

out.file <- "ComparativeGenomicsLG1nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG1",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG2nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG2",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG3nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG3",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG4nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG4",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG5nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG5",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG6nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG6",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG7nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG7",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG8nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG8",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG9nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG9",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG10nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG10",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG11nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG11",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG12nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG12",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG13nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG13",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG14nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG14",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG15nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG15",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG16nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG16",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG17nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG17",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG18nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG18",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG19nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG19",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG20nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG20",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG21nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG21",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG22nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG22",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG23nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG23",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG24nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG24",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG25nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG25",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG26nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG26",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG27nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG27",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG28nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG28",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG29nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG29",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG30nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG30",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG31nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG31",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG32nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG32",], track.num, TRUE)
dev.off()




out.file <- "ComparativeGenomicschr1nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr1",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr2nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr2",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr3nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr3",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr4nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr4",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr5nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr5",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr6nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr6",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr7nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr7",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr8nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr8",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr9nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr9",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr10nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr10",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr11nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr11",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr12nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr12",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr13nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr13",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr14nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr14",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr15nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr15",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr16nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr16",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr17nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr17",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr18nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr18",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr19nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr19",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr20nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr20",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr21nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr21",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr22nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr22",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr23nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr23",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr24nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr24",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr25nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr25",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr26nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr26",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr27nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr27",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr28nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr28",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr29nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr29",], track.num, TRUE)
dev.off()


```


Defining blocks!
And compare with BRC

```{r}

BCR <- read.csv("BCR.csv")
#Somewhat similar sequences (blastn), default
BCR2 <- read.csv("BCR2.csv")
#Somewhat similar sequences (blastn), Word size 7, Match/Mitmatch Score 4,-5, 


chr1 <- linksnodup[linksnodup$Chromosome.x=="chr1",]
chr1 <- chr1[order(chr1$chromStart),]
chr1
chr1blocks <- chr1[c(1,2,5,8),c(1,2,3,4)] 
# the markers in 6th, and 7th position are slightly unsorted but we are not going to consider it.
chr1blocks[1,3] <- chr1[2,2]-1
chr1blocks[2,3] <- chr1[4,2]
chr1blocks[3,3] <- chr1[7,2]
chr1blocks[4,3] <- chr1[10,2]
newrow <- c("chr1", 5952569, 8518522, "fisfus")
chr1blocks <- insertRow(chr1blocks, newrow, 3)
newrow <- c("chr1", 13077361, 16671836, "fisfus")
chr1blocks <- insertRow(chr1blocks, newrow, 5)
chr1blocks[1,4]<- "maybefisfus"
chr1blocks[2,4]<- "conserved"
chr1blocks[4,4]<- "conserved"
chr1blocks[6,4]<- "conserved"
chr1blocks

BCR[BCR$Chromosome=="chr1",]
colnames(chr1blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr1blocks$PatternCode <- c(0.5,0,1,0,1,0) 
chr1blocks$CBR <- c(1,0,1,0,2,1)
chr1blocks

chr2 <- linksnodup[linksnodup$Chromosome.x=="chr2",]
chr2 <- chr2[order(chr2$chromStart),]
chr2
chr2blocks <- chr2[c(1,9,10,14),c(1,2,3,4)]
chr2blocks[1,3] <- chr2[8,2]
chr2blocks[2,2] <- chr2[8,2]+1
chr2blocks[2,3] <- chr2[10,2]-1
chr2blocks[3,3] <- chr2[13,2]
chr2blocks[4,2] <- chr2[13,2]+1
chr2blocks[4,3] <- chr2[16,2]
chr2blocks[1,4] <- "conserved"
chr2blocks[2,4] <- "maybefisfus"
chr2blocks[3,4] <- "conserved"
chr2blocks[4,4] <- "unsortedx1&fisfus"
chr2blocks

BCR[BCR$Chromosome=="chr2",]
colnames(chr2blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr2blocks$PatternCode <- c(0,0.5,0,1.5)
chr2blocks$CBR <- c(0,2,1,0)
#there is one BCR just before the first conserved area
chr2blocks

chr3 <- linksnodup[linksnodup$Chromosome.x=="chr3",]
chr3 <- chr3[order(chr3$chromStart),]
chr3
chr3blocks <- chr3[c(1,3,9,10,13),c(1,2,3,4)]
chr3blocks[1,3] <- chr3[3,2]-1
chr3blocks[2,3] <- chr3[8,2]
chr3blocks[3,2] <- chr3[8,2]+1
chr3blocks[3,3] <- chr3[10,2]-1
chr3blocks[4,3] <- chr3[12,2]
chr3blocks[5,3] <- chr3[14,2]
chr3blocks
newrow <- c("chr3", 13772270, 15865085, "fisfus")
chr3blocks <- insertRow(chr3blocks, newrow, 5)
chr3blocks[1,4] <- "fisfus"
chr3blocks[2,4] <- "conserved"
chr3blocks[3,4] <- "fisfus"
chr3blocks[4,4] <- "conserved"
chr3blocks[6,4] <- "maybeconserved"
chr3blocks

BCR[BCR$Chromosome=="chr3",]
colnames(chr3blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr3blocks$PatternCode <- c(1,0,1,0,1,0) 
chr3blocks$CBR <- c(0,0,1,0,1,0)
#there is one BCR just before the first fisfus, single repeat using BCR2 setting but we are not considering it
chr3blocks

chr4 <- linksnodup[linksnodup$Chromosome.x=="chr4",]
chr4 <- chr4[order(chr4$chromStart),]
chr4
chr4blocks <- chr4[c(1,4,7,8,13,17),c(1,2,3,4)]
chr4blocks[1,3] <- chr4[4,2]-1
chr4blocks[2,3] <- chr4[6,2]
chr4blocks[3,2] <- chr4[6,2]+1
chr4blocks[3,3] <- chr4[8,2]-1
chr4blocks[4,3] <- chr4[12,2]
chr4blocks[5,3] <- chr4[16,2]
chr4blocks[6,2] <- chr4[16,2]+1
chr4blocks[6,3] <- chr4[18,2]-1
newrow <- c("chr4", 12453692, 13155061, "fisfus")
chr4blocks <- insertRow(chr4blocks, newrow, 5)
chr4blocks[1,4] <- "maybefisfusx2"
chr4blocks[2,4] <- "conserved"
chr4blocks[3,4] <- "maybefisfus"
chr4blocks[4,4] <- "conserved"
chr4blocks[6,4] <- "conserved"
chr4blocks[7,4] <- "unsorted&maybefisfus"
chr4blocks

BCR[BCR$Chromosome=="chr4",]
colnames(chr4blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr4blocks$PatternCode <- c(1,0,0.5,0,1,0,1) 
chr4blocks$CBR <- c(1,0,0,1,0,0,1)
chr4blocks


chr5 <- linksnodup[linksnodup$Chromosome.x=="chr5",]
chr5 <- chr5[order(chr5$chromStart),]
chr5
chr5blocks <- chr5[c(1,5,7,10),c(1,2,3,4)]
chr5blocks[1,3] <- chr5[4,2]
chr5blocks[2,2] <- chr5[4,2]+1
chr5blocks[2,3] <- chr5[7,2]-1
chr5blocks[3,3] <- chr5[9,2]
chr5blocks[4,2] <- chr5[9,2]+1
chr5blocks[4,3] <- chr5[10,2]
chr5blocks[1,4] <- "conserved"
chr5blocks[2,4] <- "unsorted&fisfus"
chr5blocks[3,4] <- "conserved"
chr5blocks[4,4] <- "unsorted"
chr5blocks

BCR[BCR$Chromosome=="chr5",]
colnames(chr5blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr5blocks$PatternCode <- c(0,1.5,0,0.5) 
chr5blocks$CBR <- c(0,3,0,0)
#there are three CBR areas but both are in the fisfus of the patterns and not in the change order
chr5blocks


chr6 <- linksnodup[linksnodup$Chromosome.x=="chr6",]
chr6 <- chr6[order(chr6$chromStart),]
chr6
chr6blocks <- chr6[c(1,4,7),c(1,2,3,4)]
chr6blocks[1,3] <- chr6[3,2]
chr6blocks[2,2] <- chr6[3,2]+1
chr6blocks[2,3] <- chr6[7,2]-1
chr6blocks[3,3] <- chr6[11,2]
chr6blocks[1,4] <- "conserved"
chr6blocks[2,4] <- "fisfusx3"
chr6blocks[3,4] <- "conserved"
chr6blocks

BCR2[BCR2$Chromosome=="chr6",]
#There are not BCR this chr, but there is one when relaxing parameters
colnames(chr6blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr6blocks$PatternCode <- c(0,3,0) 
chr6blocks$CBR <- c(0,1,0)
chr6blocks


chr7 <- linksnodup[linksnodup$Chromosome.x=="chr7",]
chr7 <- chr7[order(chr7$chromStart),]
chr7
chr7blocks <- chr7[c(1,8,11),c(1,2,3,4)]
chr7blocks[1,3] <- chr7[8,2]-1
chr7blocks[2,3] <- chr7[10,2]
chr7blocks[3,3] <- chr7[13,2]
newrow <- c("chr7", 9609375, 10386170, "fisfus")
chr7blocks <- insertRow(chr7blocks, newrow, 3)
chr7blocks[1,4] <- "unsortedx2&fisfus"
chr7blocks[2,4] <- "conserved"
chr7blocks[4,4] <- "conserved"
chr7blocks

BCR[BCR$Chromosome=="chr7",]
colnames(chr7blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr7blocks$PatternCode <- c(2,0,1,0) 
chr7blocks$CBR <- c(2,0,0,0)
#There are two BCR, but the second one in after and out of the known conserved
chr7blocks


chr8 <- linksnodup[linksnodup$Chromosome.x=="chr8",]
chr8 <- chr8[order(chr8$chromStart),]
chr8
chr8blocks <- chr8[c(1,6,8,10,11),c(1,2,3,4)]
# the markers in 14th, 15th and 16th position are slightly unsorted but we are not going to consider it.
chr8blocks[1,3] <- chr8[5,2]
chr8blocks[2,2] <- chr8[5,2]+1
chr8blocks[2,3] <- chr8[8,2]-1
chr8blocks[3,3] <- chr8[9,2]
chr8blocks[4,2] <- chr8[9,2]+1
chr8blocks[4,3] <- chr8[11,2]-1
chr8blocks[5,3] <- chr8[17,2]
chr8blocks[1,4] <- "conserved"
chr8blocks[2,4] <- "fisfusx2"
chr8blocks[3,4] <- "maybeconserved"
chr8blocks[4,4] <- "maybefisfus"
chr8blocks[5,4] <- "conserved"
chr8blocks

BCR[BCR$Chromosome=="chr8",]
colnames(chr8blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr8blocks$PatternCode <- c(0,2,0,0.5,0) 
chr8blocks$CBR <- c(0,1,0,1,0)
#there are 1 BCR before the 1st conserved area, and one after the last conserved area
#the 1 BCR in the maybefisfus is just one single
chr8blocks

chr9 <- linksnodup[linksnodup$Chromosome.x=="chr9",]
chr9 <- chr9[order(chr9$chromStart),]
chr9
chr9blocks <- chr9[c(1,2,6),c(1,2,3,4)]
# the markers in 7th and 9th position are slightly unsorted but we are not going to consider it. The same for 10th and 11th.
chr9blocks[1,3] <- chr9[2,2]-1
chr9blocks[2,3] <- chr9[5,2]
chr9blocks[3,3] <- chr9[11,2]
newrow <- c("chr9", 5660034, 12014988, "fisfus")
chr9blocks <- insertRow(chr9blocks, newrow, 3)
chr9blocks[1,4] <- "maybefisfus"
chr9blocks[2,4] <- "conserved"
chr9blocks[4,4] <- "conserved"
chr9blocks

BCR[BCR$Chromosome=="chr9",]
BCR2[BCR2$Chromosome=="chr9",]
#There are not BCR this chr
colnames(chr9blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr9blocks$PatternCode <- c(0.5,0,1,0) 
chr9blocks$CBR <- c(0,0,0,0)
chr9blocks
#We are missing something

chr10 <- linksnodup[linksnodup$Chromosome.x=="chr10",]
chr10 <- chr10[order(chr10$chromStart),]
chr10
chr10blocks <- chr10[c(1,13,18,19,23,24),c(1,2,3,4)]
# the markers in 19th and 20th position are slightly unsorted but we are not going to consider it. The same for 21st and 22nd.
chr10blocks[1,3] <- chr10[12,2]
chr10blocks[2,3] <- chr10[17,2]
chr10blocks[3,2] <- chr10[17,2]+1
chr10blocks[3,3] <- chr10[19,2]-1
chr10blocks[4,3] <- chr10[22,2]
chr10blocks[5,2] <- chr10[22,2]+1
chr10blocks[5,3] <- chr10[24,2]-1
chr10blocks[6,3] <- chr10[26,2]
newrow <- c("chr10", 4244125, 5269404, "fisfus")
chr10blocks <- insertRow(chr10blocks, newrow, 2)
chr10blocks[1,4] <- "conserved"
chr10blocks[3,4] <- "conserved"
chr10blocks[4,4] <- "maybefisfus"
chr10blocks[5,4] <- "conserved"
chr10blocks[6,4] <- "maybefisfus"
chr10blocks[7,4] <- "conserved"
chr10blocks

BCR[BCR$Chromosome=="chr10",]
BCR2[BCR2$Chromosome=="chr10",]
#There are not BCR this chr
colnames(chr10blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr10blocks$PatternCode <- c(0,1,0,0.5,0,0.5,0) 
chr10blocks$CBR <- c(0,0,0,0,0,0,0)
chr10blocks


chr11 <- linksnodup[linksnodup$Chromosome.x=="chr11",]
chr11 <- chr11[order(chr11$chromStart),]
chr11
chr11blocks <- chr11[c(1,2,6,9,11),c(1,2,3,4)]
# the markers in 18th and 19th position are slightly unsorted but we are not going to consider it.
chr11blocks[1,3] <- chr11[2,2]-1
chr11blocks[2,3] <- chr11[5,2]
chr11blocks[3,3] <- chr11[8,2]
chr11blocks[4,3] <- chr11[10,2]
chr11blocks[5,3] <- chr11[23,2]
newrow <- c("chr11", 2606506, 3836219, "fisfus")
chr11blocks <- insertRow(chr11blocks, newrow, 3)
newrow <- c("chr11", 5751244, 6162157, "maybefisfus")
chr11blocks <- insertRow(chr11blocks, newrow, 5)
newrow <- c("chr11", 7481618, 10687339, "maybefisfus")
chr11blocks <- insertRow(chr11blocks, newrow, 7)
chr11blocks[1,4] <- "maybefisfus"
chr11blocks[2,4] <- "conserved"
chr11blocks[4,4] <- "conserved"
chr11blocks[6,4] <- "maybeconserved"
chr11blocks[8,4] <- "conserved"
chr11blocks

BCR[BCR$Chromosome=="chr11",]
colnames(chr11blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr11blocks$PatternCode <- c(0.5,0,1,0,0.5,0,0.5,0) 
chr11blocks$CBR <- c(1,1,0,0,0,0,0,0)
#the first BCR is in the area of some small unsorting that we have not considered
#There are two BCR, but the second one in after and out of the known conserved
chr11blocks


chr12 <- linksnodup[linksnodup$Chromosome.x=="chr12",]
chr12 <- chr12[order(chr12$chromStart),]
chr12
chr12blocks <- chr12[c(1,3,6,12,19,21),c(1,2,3,4)] 
# the markers in 16th and 17th position are slightly unsorted but we are not going to consider it.
chr12blocks[1,3] <- chr12[2,2]
chr12blocks[2,2] <- chr12[2,2]+1
chr12blocks[2,3] <- chr12[6,2]-1
chr12blocks[3,3] <- chr12[11,2]
chr12blocks[4,3] <- chr12[17,2]
chr12blocks[5,3] <- chr12[20,2]
chr12blocks[6,3] <- chr12[23,2]
newrow <- c("chr12", 6078481, 7374679, "fisfus")
chr12blocks <- insertRow(chr12blocks, newrow, 4)
newrow <- c("chr12", 9012769, 10409253, "maybefisfus")
chr12blocks <- insertRow(chr12blocks, newrow, 6)
newrow <- c("chr12", 10917073, 11405499, "maybefisfus")
chr12blocks <- insertRow(chr12blocks, newrow, 8)
chr12blocks[1,4] <- "maybeconserved"
chr12blocks[2,4] <- "maybefisfusx2"
chr12blocks[3,4] <- "conserved"
chr12blocks[5,4] <- "conserved"
chr12blocks[7,4] <- "maybeconserved"
chr12blocks[9,4] <- "conserved"
chr12blocks

BCR[BCR$Chromosome=="chr12",]
BCR2[BCR2$Chromosome=="chr12",]
colnames(chr12blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr12blocks$PatternCode <- c(0,1,0,1,0,0.5,0,0.5,0) 
chr12blocks$CBR <- c(0,0,0,0,0,0,0,0,0)
#the first BCR in after and out of the known conserved, the same with BCR2 settings
chr12blocks
#We are missing something

chr13 <- linksnodup[linksnodup$Chromosome.x=="chr13",]
chr13 <- chr13[order(chr13$chromStart),]
chr13
chr13blocks <- chr13[c(1,8,13,19),c(1,2,3,4)] 
chr13blocks[1,3] <- chr13[8,2]-1
chr13blocks[2,3] <- chr13[12,2]
chr13blocks[3,3] <- chr13[18,2]
chr13blocks[4,2] <- chr13[18,2]+1
chr13blocks[4,3] <- chr13[21,2]
newrow <- c("chr13", 9428904, 10457675, "fisfus")
chr13blocks <- insertRow(chr13blocks, newrow, 3)
chr13blocks[1,4] <- "unsortedx3&mayfisfus"
chr13blocks[2,4] <- "conserved"
chr13blocks[4,4] <- "conserved"
chr13blocks[5,4] <- "fisfus&unsorted"
chr13blocks

BCR[BCR$Chromosome=="chr13",]
colnames(chr13blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr13blocks$PatternCode <- c(2.5,0,1,0,1.5)
chr13blocks$CBR <- c(1,0,0,0,0)
chr13blocks


chr14 <- linksnodup[linksnodup$Chromosome.x=="chr14",]
chr14 <- chr14[order(chr14$chromStart),]
chr14
# the markers in 12th and 13th position are slightly unsorted but we are not going to consider it. The same for the markers in the positions 15th and 16th.
chr14blocks <- chr14[c(1,3,8,15,22),c(1,2,3,4)] 
chr14blocks[1,3]<-chr14[3,2]-1
chr14blocks[2,3]<-chr14[7,2]
chr14blocks[3,3]<-chr14[14,2]
chr14blocks[4,3]<-chr14[21,2]
chr14blocks[5,2]<-chr14[21,2]+1
chr14blocks[5,3]<-chr14[25,2]
newrow <- c("chr14", 5018142, 5279697, "fisfus")
chr14blocks <- insertRow(chr14blocks, newrow, 3)
newrow <- c("chr14", 8444297, 10345369, "fisfus")
chr14blocks  <- insertRow(chr14blocks, newrow, 5)
chr14blocks[1,4] <- "maybefisfusx2"
chr14blocks[2,4] <- "conserved"
chr14blocks[4,4] <- "conserved"
chr14blocks[6,4] <- "conserved"
chr14blocks[7,4] <- "maybefisfusx2&unsorted"
chr14blocks

BCR[BCR$Chromosome=="chr14",]
colnames(chr14blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr14blocks$PatternCode <- c(1,0,1,0,1,0,1.5)
chr14blocks$CBR <- c(1,0,0,1,1,0,0)
#the BCR is before the first maybefisfusx2
chr14blocks

chr15 <- linksnodup[linksnodup$Chromosome.x=="chr15",]
chr15 <- chr15[order(chr15$chromStart),]
chr15
# the markers in third and fourth position are slightly unsorted but we are not going to consider it. The same for the markers in the positions 9th and 10th.
chr15blocks <- chr15[c(1,6,12,13,16),c(1,2,3,4)] 
chr15blocks[1,3] <- chr15[5,2]
chr15blocks[2,3] <- chr15[11,2]
chr15blocks[3,2] <- chr15[11,2]+1
chr15blocks[3,3] <- chr15[13,2]-1
chr15blocks[4,3] <- chr15[15,2]
chr15blocks[5,2] <- chr15[15,2]+1
chr15blocks[5,3] <- chr15[16,2]
newrow <- c("chr15", 4376634, 8917572, "fisfus")
chr15blocks <- insertRow(chr15blocks, newrow, 2)
chr15blocks[1,4] <- "conserved"
chr15blocks[3,4] <- "conserved"
chr15blocks[4,4] <- "maybefisfus"
chr15blocks[5,4] <- "conserved"
chr15blocks[6,4] <- "maybefisfus"
chr15blocks

#There are not BCR this chr
BCR[BCR$Chromosome=="chr15",]
BCR2[BCR2$Chromosome=="chr15",]
colnames(chr15blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr15blocks$PatternCode <- c(0,1,0,0.5,0,0.5)
chr15blocks$CBR <- c(0,0,0,0,0,0)
chr15blocks


chr16 <- linksnodup[linksnodup$Chromosome.x=="chr16",]
chr16 <- chr16[order(chr16$chromStart),]
chr16
# the markers in sixth and seventh position are slightly unsorted but we are not going to consider it
chr16blocks <- chr16[c(1,9,10,13),c(1,2,3,4)] 
chr16blocks[1,3] <- chr16[8,2]
chr16blocks[2,2] <- chr16[8,2]+1
chr16blocks[2,3] <- chr16[10,2]-1
chr16blocks[3,3] <- chr16[12,2]
chr16blocks[4,2] <- chr16[12,2]+1
chr16blocks[4,3] <- chr16[14,2]
chr16blocks[1,4] <- "conserved"
chr16blocks[2,4] <- "maybefisfus"
chr16blocks[3,4] <- "conserved"
chr16blocks[4,4] <- "maybefisfusx2"
chr16blocks

BCR[BCR$Chromosome=="chr16",]
colnames(chr16blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr16blocks$PatternCode <- c(0,0.5,0,1)
chr16blocks$CBR <- c(0,1,1,1)
chr16blocks


chr17 <- linksnodup[linksnodup$Chromosome.x=="chr17",]
chr17 <- chr17[order(chr17$chromStart),]
chr17
chr17blocks <- chr17[c(1,7,10,11,13),c(1,2,3,4)] 
chr17blocks[1,3]<-chr17[7,2]-1
chr17blocks[2,3]<-chr17[9,2]
chr17blocks[3,2]<-chr17[9,2]+1
chr17blocks[3,3]<-chr17[11,2]-1
chr17blocks[4,3]<-chr17[12,2]
chr17blocks[5,2]<-chr17[12,2]+1
chr17blocks[5,3]<-chr17[14,2]
chr17blocks[1,4]<-"maybefisfus&unsortedx2"
chr17blocks[2,4]<-"conserved"
chr17blocks[3,4]<-"unsorted&fisfus&unsorted"
chr17blocks[4,4]<-"maybeconserved"
chr17blocks[5,4]<-"fisfusx1.5"
chr17blocks
#the maybeconserved could be problably gone

BCR[BCR$Chromosome=="chr17",]
colnames(chr17blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr17blocks$PatternCode <- c(1.5,0,2,0,1.5)
chr17blocks$CBR <- c(0,0,0,0,2)
#The second BCR is after the whole thing
chr17blocks


chr18 <- linksnodup[linksnodup$Chromosome.x=="chr18",]
chr18 <- chr18[order(chr18$chromStart),]
chr18
chr18blocks <- chr18[c(1,3,5,8,9,13),c(1,2,3,4)] 
chr18blocks[1,3] <- chr18[2,2]
chr18blocks[2,2] <- chr18[2,2]+1
chr18blocks[2,3] <- chr18[5,2]-1
chr18blocks[3,3] <- chr18[7,2]
chr18blocks[4,2] <- chr18[7,2]+1
chr18blocks[4,3] <- chr18[9,2]-1
chr18blocks[5,3] <- chr18[12,2]
chr18blocks[6,3] <- chr18[20,2]
newrow <- c("chr18", 6038837, 6531250, "fisfus")
chr18blocks <- insertRow(chr18blocks, newrow, 6)
chr18blocks[1,4] <- "maybeconserved"
chr18blocks[2,4] <- "maybefisfus&unsorted"
chr18blocks[3,4] <- "conserved"
chr18blocks[4,4] <- "unsorted&maybefisfus"
chr18blocks[5,4] <- "conserved"
chr18blocks[7,4] <- "conserved"
chr18blocks

BCR[BCR$Chromosome=="chr18",]
colnames(chr18blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr18blocks$PatternCode <- c(0,1,0,1,0,1,0)
chr18blocks$CBR <- c(0,0,0,0,0,0,1)
#There is one BCR before everything
chr18blocks

chr19 <- linksnodup[linksnodup$Chromosome.x=="chr19",]
chr19 <- chr19[order(chr19$chromStart),]
chr19
chr19blocks <- chr19[c(1,4,9,12),c(1,2,3,4)] 
chr19blocks[1,3] <- chr19[3,2]
chr19blocks[2,2] <- chr19[3,2] + 1
chr19blocks[2,3] <- chr19[9,2] -1
chr19blocks[3,3] <- chr19[11,2]
chr19blocks[4,3] <- chr19[15,2]
newrow <- c("chr19", 7006970, 8323676, "fisfus")
chr19blocks <- insertRow(chr19blocks, newrow, 4)
chr19blocks[1,4] <- "conserved"
chr19blocks[2,4] <- "maybefisfus&unsortedx2"
chr19blocks[3,4] <- "conserved"
chr19blocks[5,4] <- "conserved"
chr19blocks

#There are not BCR this chr
BCR[BCR$Chromosome=="chr19",]
BCR2[BCR2$Chromosome=="chr19",]
colnames(chr19blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr19blocks$PatternCode <- c(0,1.5,0,1,0)
chr19blocks$CBR <- c(0,0,0,0,0)
chr19blocks

chr20 <- linksnodup[linksnodup$Chromosome.x=="chr20",]
chr20 <- chr20[order(chr20$chromStart),]
chr20
chr20blocks <- chr20[c(1),c(1,2,3,4)] 
chr20blocks[1,3] <- chr20[5,2]
chr20blocks[1,4] <- "unsorted&maybefisfus"
chr20blocks

#There are not BCR this chr
BCR[BCR$Chromosome=="chr20",]
BCR2[BCR2$Chromosome=="chr20",]
colnames(chr20blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr20blocks$PatternCode <- c(1)
chr20blocks$CBR <- c(0)
chr20blocks

chr21 <- linksnodup[linksnodup$Chromosome.x=="chr21",]
chr21 <- chr21[order(chr21$chromStart),]
chr21
chr21blocks <- chr21[c(1,3,5),c(1,2,3,4)] 
chr21blocks[1,3] <- chr21[3,2]-1 
chr21blocks[2,3] <- chr21[4,2]
chr21blocks[3,3] <- chr21[6,2]
newrow <- c("chr21", 5440874, 6958720, "maybefisfus")
chr21blocks <- insertRow(chr21blocks, newrow, 3)
chr21blocks[1,4] <- "maybefisfusx2"
chr21blocks[2,4] <- "maybeconserved"
chr21blocks[4,4] <- "maybeconserved"
chr21blocks

#There are not BCR this chr
BCR[BCR$Chromosome=="chr21",]
BCR2[BCR2$Chromosome=="chr21",]
colnames(chr21blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr21blocks$PatternCode <- c(1, 0, 0.5, 0)
chr21blocks$CBR <- c(0,0,0,0)
chr21blocks


chr22 <- linksnodup[linksnodup$Chromosome.x=="chr22",]
chr22 <- chr22[order(chr22$chromStart),]
chr22
chr22blocks <- chr22[c(1,4,9,10,13),c(1,2,3,4)] 
chr22blocks[1,3] <- chr22[4,2] - 1
chr22blocks[2,3] <- chr22[8,2]
chr22blocks[3,2] <- chr22[8,2] + 1
chr22blocks[3,3] <- chr22[10,2] - 1
chr22blocks[4,3] <- chr22[12,2]
chr22blocks[5,2] <- chr22[12,2] + 1
chr22blocks[5,3] <- chr22[17,2]
chr22blocks[1,4] <- "unsorted&maybefisfus"
chr22blocks[2,4] <- "conserved"
chr22blocks[3,4] <- "unsorted&fisfus"
chr22blocks[4,4] <- "conserved"
chr22blocks[5,4] <- "fisfusx1.5&unsorted"
chr22blocks

BCR[BCR$Chromosome=="chr22",]
colnames(chr22blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr22blocks$PatternCode <- c(1, 0, 1.5, 0,2)
chr22blocks$CBR <- c(0,0,0,0,1)
chr22blocks


chr23 <- linksnodup[linksnodup$Chromosome.x=="chr23",]
chr23 <- chr23[order(chr23$chromStart),]
chr23
chr23blocks <- chr23[c(1,5,10,16),c(1,2,3,4)] 
chr23blocks[1,3] <- chr23[5,2] - 1 
chr23blocks[2,3] <- chr23[9,2]
chr23blocks[3,3] <- chr23[15,2]
chr23blocks[4,2] <- chr23[15,2]+1
chr23blocks[4,3] <- chr23[19,2]
newrow <- c("chr23", 5567451, 5942385, "fisfus")
chr23blocks <- insertRow(chr23blocks, newrow, 3)
chr23blocks[1,4] <- "unsortedx2&fisfus"
chr23blocks[2,4] <- "conserved"
chr23blocks[4,4] <- "conserved"
chr23blocks[5,4] <- "unsorted&maybefisfus"
chr23blocks

#There are not BCR this chr
BCR[BCR$Chromosome=="chr23",]
BCR2[BCR2$Chromosome=="chr23",]
colnames(chr23blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr23blocks$PatternCode <- c(2, 0, 1, 0,1)
chr23blocks$CBR <- c(0,0,0,1,0)
chr23blocks


chr24 <- linksnodup[linksnodup$Chromosome.x=="chr24",]
chr24 <- chr24[order(chr24$chromStart),]
chr24
chr24blocks <- chr24[c(1,8),c(1,2,3,4)] 
chr24blocks
chr24blocks[1,3] <- chr24[8,2]-1
chr24blocks[2,3] <- chr24[18,2]
chr24blocks[1,4] <- "unsortedx2&fisfus"
chr24blocks[2,4] <- "conserved"
chr24blocks

BCR[BCR$Chromosome=="chr24",]
colnames(chr24blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr24blocks$PatternCode <- c(2, 0)
chr24blocks$CBR <- c(1,0)
#There are only 2 single repeat at the beginning, before everything
chr24blocks

chr25 <- linksnodup[linksnodup$Chromosome.x=="chr25",]
chr25 <- chr25[order(chr25$chromStart),]
chr25
chr25blocks <- chr25[c(1,3,7,9),c(1,2,3,4)] 
chr25blocks[1,3] <- chr25[2,3]
chr25blocks[2,3] <- chr25[6,3]
chr25blocks[3,2] <- chr25[6,3] + 1
chr25blocks[3,3] <- chr25[9,2] - 1
chr25blocks[4,3] <- chr25[15,3]
newrow <- c("chr25", 325961, 1321736, "fisfus")
#insertRow <- function(existingDF, newrow, r) {
#  existingDF[seq(r+1,nrow(existingDF)+1),] <- existingDF[seq(r,nrow(existingDF)),]
#  existingDF[r,] <- newrow
#  existingDF
#}
chr25blocks <- insertRow(chr25blocks, newrow, 2)
chr25blocks[1,4] <- "maybeconserved"
chr25blocks[3,4] <- "conserved"
chr25blocks[4,4] <- "maybefisfus&fisfus"
chr25blocks[5,4] <- "conserved"
chr25blocks
#2 conserved block and 3rd possible, and two rearrangementes between then

#There are not BCR this chr
BCR[BCR$Chromosome=="chr25",]
colnames(chr25blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr25blocks$PatternCode <- c(0, 1, 0, 1.5,0)
#we are going to remove the maybeconserved
chr25blocks$CBR <- c(0,0,0,0,0)
chr25blocks


chr26 <- linksnodup[linksnodup$Chromosome.x=="chr26",]
chr26 <- chr26[order(chr26$chromStart),]
chr26
chr26blocks <- chr26[c(1,5),c(1,2,3,4)] 
chr26blocks[1,3] <- chr26[4,2]
chr26blocks[2,2] <- chr26[4,2] + 1
chr26blocks[2,3] <- chr26[8,2]
chr26blocks[2,4] <- "fisfus&maybefisfusx2"
chr26blocks[1,4] <- "conserved"
chr26blocks
#single conserved block and area with possible multiple rearrangements

#There are not BCR this chr
BCR[BCR$Chromosome=="chr26",]
colnames(chr26blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr26blocks$PatternCode <- c(0, 2)
chr26blocks$CBR <- c(0,0)
chr26blocks



chr27 <- linksnodup[linksnodup$Chromosome.x=="chr27",]
chr27 <- chr27[order(chr27$chromStart),]
chr27
chr27blocks <- chr27[c(1,4),c(1,2,3,4)] 
chr27blocks[1,3] <- chr27[3,2]
chr27blocks[1,4] <- "conserved"
chr27blocks[2,3] <- chr27[6,2]
chr27blocks[2,4] <- "conserved"
chr27blocks <- rbind(chr27blocks[1,],c("chr27", 3696901,4879022,"fisfus"),chr27blocks[2,])
chr27blocks
#two conserved blocks detected, 1 rearrangement between both

#There are not BCR this chr
BCR[BCR$Chromosome=="chr27",]
colnames(chr27blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr27blocks$PatternCode <- c(0, 1, 0)
chr27blocks$CBR <- c(0,0,0)
chr27blocks

chr28 <- linksnodup[linksnodup$Chromosome.x=="chr28",]
chr28 <- chr28[order(chr28$chromStart),]
chr28
chr28blocks <- chr28[c(1,5),c(1,2,3,4)] 
chr28blocks[1,3] <- chr28[4,2]
chr28blocks[2,2] <- chr28[4,2] + 1
chr28blocks[2,3] <- chr28[6,2]
chr28blocks[1,4] <- "conserved"
chr28blocks[2,4] <- "unsorted"
chr28blocks
#single conserved block, a little unsorting

#There are not BCR this chr
BCR[BCR$Chromosome=="chr28",]
colnames(chr28blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr28blocks$PatternCode <- c(0, 0.5)
chr28blocks$CBR <- c(0,0)
chr28blocks

chr29 <- linksnodup[linksnodup$Chromosome.x=="chr29",]
chr29 <- chr29[order(chr29$chromStart),]
chr29
chr29blocks <- chr29[c(1),c(1,2,3,4)] 
chr29blocks[1,3] <- chr29[3,2]
chr29blocks[1,4] <- "conserved"
chr29blocks
#single conserved block

#There are not BCR this chr
BCR[BCR$Chromosome=="chr29",]
colnames(chr29blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr29blocks$PatternCode <- c(0)
chr29blocks$CBR <- c(0)
chr29blocks



```

Let's compare the block and the BCR

```{r}

```

