---
title: "CarexComparativeGenomics"
author: "Marcial Escudero"
date: "7/13/2021"
output: html_document
editor_options: 
  chunk_output_type: console
---

Lerning with the tutorial

```{r}
library(RCircos)

data(RCircos.Histogram.Data)
head(RCircos.Histogram.Data)

data(RCircos.Heatmap.Data)
head(RCircos.Heatmap.Data)

data(RCircos.Link.Data)
head(RCircos.Link.Data)

data(UCSC.HG19.Human.CytoBandIdeogram)
head(UCSC.HG19.Human.CytoBandIdeogram)

data(UCSC.Mouse.GRCm38.CytoBandIdeogram)
head(UCSC.Mouse.GRCm38.CytoBandIdeogram)

data(UCSC.Baylor.3.4.Rat.cytoBandIdeogram)
head(UCSC.Baylor.3.4.Rat.cytoBandIdeogram)


chr.exclude <- NULL
cyto.info <- UCSC.HG19.Human.CytoBandIdeogram
#cyto.info <- RCircos.Gene.Label.Data
tracks.inside <- 10
tracks.outside <- 0
RCircos.Set.Core.Components(cyto.info, chr.exclude,tracks.inside, tracks.outside)

rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.cyto <- RCircos.Get.Plot.Ideogram()
rcircos.position <- RCircos.Get.Plot.Positions()
RCircos.List.Plot.Parameters()

rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.params$base.per.unit <- 3000
RCircos.Reset.Plot.Parameters(rcircos.params)
RCircos.List.Plot.Parameters()

par(mai=c(0.25, 0.25, 0.25, 0.25))
plot.new()
plot.window(c(-2.5,2.5), c(-2.5, 2.5))

out.file <- "RCircosDemoHumanGenome.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)

RCircos.Set.Plot.Area()

RCircos.Chromosome.Ideogram.Plot()


data(RCircos.Gene.Label.Data)
head(RCircos.Gene.Label.Data)
name.col <- 4
side <- "in"
track.num <- 1
RCircos.Gene.Connector.Plot(RCircos.Gene.Label.Data,track.num, side)
track.num <- 2
RCircos.Gene.Name.Plot(RCircos.Gene.Label.Data,name.col,track.num, side)

data(RCircos.Heatmap.Data)
data.col <- 6
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(RCircos.Heatmap.Data, data.col, track.num, side)
data(RCircos.Scatter.Data)
data.col <- 5
track.num <- 6
side <- "in"
by.fold <- 1
RCircos.Scatter.Plot(RCircos.Scatter.Data, data.col,  track.num, side, by.fold)
data(RCircos.Line.Data)
data.col <- 5
track.num <- 7
side <- "in"
RCircos.Line.Plot(RCircos.Line.Data, data.col,track.num, side)
data(RCircos.Histogram.Data)
data.col <- 4
track.num <- 8
side <- "in"
RCircos.Histogram.Plot(RCircos.Histogram.Data,data.col, track.num, side)
data(RCircos.Tile.Data)
track.num <- 9
side <- "in"
RCircos.Tile.Plot(RCircos.Tile.Data, track.num, side)


data(RCircos.Link.Data)
track.num <- 11
RCircos.Link.Plot(RCircos.Link.Data, track.num, TRUE)
data(RCircos.Ribbon.Data)
RCircos.Ribbon.Plot(ribbon.data=RCircos.Ribbon.Data,track.num=11, by.chromosome=FALSE, twist=FALSE)


dev.off()

```


Opening some Carex data

```{r}

both <- read.csv("both.csv")

both

chr.exclude <- NULL
cyto.info <- both
tracks.inside <- 0
tracks.outside <- 0
RCircos.Set.Core.Components(cyto.info, chr.exclude,tracks.inside, tracks.outside)


rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.cyto <- RCircos.Get.Plot.Ideogram()
rcircos.position <- RCircos.Get.Plot.Positions()
RCircos.List.Plot.Parameters()

 

rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.params$base.per.unit <- 3000
RCircos.Reset.Plot.Parameters(rcircos.params)
RCircos.List.Plot.Parameters()

RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()



```

Setting up the bowtie2 analysis

```{r}

readsdata <- read.csv("batch_3.catalog.tags.tsv", sep = "")

readsdata[,3] <- paste0("M", readsdata[,3])

readsdata <- readsdata[,c(3,9)]

readsdata2 <- readsdata[readsdata[,1]%in%both[,4],]

write.csv(readsdata2, file = "reads.csv")



```


Making the comparative genomics analyses
```{r}

bowtie2 <- read.csv("bowtie2.csv", sep = "")
bowtie2

both

links <- merge(bowtie2, both, by.x = "Marker", by.y = "Band", sort = F)
links <- links[,c(2,3,4,5,6,7)]
links

track.num <- 1
RCircos.Link.Plot(links, track.num, TRUE)

out.file <- "ComparativeGenomicsLG1.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG1",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG2.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG2",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG3.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG3",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG4.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG4",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG5.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG5",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG6.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG6",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG7.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG7",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG8.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG8",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG9.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG9",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG10.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG10",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG11.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG11",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG12.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG12",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG13.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG13",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG14.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG14",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG15.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG15",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG16.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG16",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG17.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG17",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG18.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG18",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG19.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG19",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG20.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG20",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG21.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG21",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG22.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG22",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG23.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG23",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG24.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG24",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG25.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG25",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG26.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG26",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG27.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG27",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG28.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG28",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG29.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG29",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG30.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG30",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG31.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG31",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG32.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG32",], track.num, TRUE)
dev.off()




out.file <- "ComparativeGenomicschr1.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr1",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr2.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr2",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr3.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr3",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr4.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr4",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr5.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr5",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr6.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr6",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr7.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr7",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr8.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr8",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr9.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr9",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr10.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr10",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr11.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr11",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr12.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr12",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr13.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr13",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr14.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr14",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr15.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr15",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr16.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr16",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr17.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr17",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr18.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr18",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr19.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr19",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr20.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr20",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr21.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr21",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr22.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr22",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr23.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr23",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr24.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr24",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr25.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr25",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr26.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr26",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr27.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr27",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr28.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr28",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr29.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr29",], track.num, TRUE)
dev.off()

```

Let's try to plot something that could be publish

```{r}

library(RColorBrewer)
n <- 30
qual_col_pals = brewer.pal.info[brewer.pal.info$category == 'qual',]
col_vector = unlist(mapply(brewer.pal, qual_col_pals$maxcolors, rownames(qual_col_pals)))
pie(rep(1,n), col=sample(col_vector, n))

col_vector[1:30]
col_vector <- c(col_vector[1:30],rep(col_vector[30],1381))

length(col_vector)
dim (both)

library(RCircos)
both <- read.csv("both.csv")

RCircos.Get.Plot.Colors(both, col_vector)

both$PlotColor <- col_vector

chr.exclude <- NULL
cyto.info <- both
tracks.inside <- 0
tracks.outside <- 0
RCircos.Set.Core.Components(cyto.info, chr.exclude,tracks.inside, tracks.outside)


rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.params
rcircos.cyto <- RCircos.Get.Plot.Ideogram()
rcircos.cyto
rcircos.position <- RCircos.Get.Plot.Positions()
rcircos.position
RCircos.List.Plot.Parameters()

#rcircos.params <- RCircos.Get.Plot.Parameters()
#rcircos.params$base.per.unit <- 3000
#RCircos.Reset.Plot.Parameters(rcircos.params)
#RCircos.List.Plot.Parameters()


rcircos.cyto
rcircos.cyto$BandColor <- rcircos.cyto$PlotColor
rcircos.cyto$ChrColor <-  rcircos.cyto$PlotColor
RCircos.Reset.Plot.Ideogram(rcircos.cyto)
RCircos.List.Plot.Parameters()


RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()


links$PlotColor <- 1
links[links[,1]=="chr1", "PlotColor"] <- col_vector[1]
links[links[,1]=="chr2", "PlotColor"] <- col_vector[2]
links[links[,1]=="chr3", "PlotColor"] <- col_vector[3]
links[links[,1]=="chr4", "PlotColor"] <- col_vector[4]
links[links[,1]=="chr5", "PlotColor"] <- col_vector[5]
links[links[,1]=="chr6", "PlotColor"] <- col_vector[6]
links[links[,1]=="chr7", "PlotColor"] <- col_vector[7]
links[links[,1]=="chr8", "PlotColor"] <- col_vector[8]
links[links[,1]=="chr9", "PlotColor"] <- col_vector[9]
links[links[,1]=="chr10", "PlotColor"] <- col_vector[10]
links[links[,1]=="chr11", "PlotColor"] <- col_vector[11]
links[links[,1]=="chr12", "PlotColor"] <- col_vector[12]
links[links[,1]=="chr13", "PlotColor"] <- col_vector[13]
links[links[,1]=="chr14", "PlotColor"] <- col_vector[14]
links[links[,1]=="chr15", "PlotColor"] <- col_vector[15]
links[links[,1]=="chr16", "PlotColor"] <- col_vector[16]
links[links[,1]=="chr17", "PlotColor"] <- col_vector[17]
links[links[,1]=="chr18", "PlotColor"] <- col_vector[18]
links[links[,1]=="chr19", "PlotColor"] <- col_vector[19]
links[links[,1]=="chr20", "PlotColor"] <- col_vector[20]
links[links[,1]=="chr21", "PlotColor"] <- col_vector[21]
links[links[,1]=="chr22", "PlotColor"] <- col_vector[22]
links[links[,1]=="chr23", "PlotColor"] <- col_vector[23]
links[links[,1]=="chr24", "PlotColor"] <- col_vector[24]
links[links[,1]=="chr25", "PlotColor"] <- col_vector[25]
links[links[,1]=="chr26", "PlotColor"] <- col_vector[26]
links[links[,1]=="chr27", "PlotColor"] <- col_vector[27]
links[links[,1]=="chr28", "PlotColor"] <- col_vector[28]
links[links[,1]=="chr29", "PlotColor"] <- col_vector[29]

RCircos.Link.Plot(links, track.num = 1)



```


Now, let's check the results with the reads that align only once

```{r}

library(RCircos)
both <- read.csv("both.csv")

RCircos.Get.Plot.Colors(both, col_vector)

both$PlotColor <- col_vector

chr.exclude <- NULL
cyto.info <- both
tracks.inside <- 0
tracks.outside <- 0
RCircos.Set.Core.Components(cyto.info, chr.exclude,tracks.inside, tracks.outside)


rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.params
rcircos.cyto <- RCircos.Get.Plot.Ideogram()
rcircos.cyto
rcircos.position <- RCircos.Get.Plot.Positions()
rcircos.position
RCircos.List.Plot.Parameters()

#rcircos.params <- RCircos.Get.Plot.Parameters()
#rcircos.params$base.per.unit <- 3000
#RCircos.Reset.Plot.Parameters(rcircos.params)
#RCircos.List.Plot.Parameters()


rcircos.cyto
rcircos.cyto$BandColor <- rcircos.cyto$PlotColor
rcircos.cyto$ChrColor <-  rcircos.cyto$PlotColor
RCircos.Reset.Plot.Ideogram(rcircos.cyto)
RCircos.List.Plot.Parameters()


RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()


bowtie2dup <- read.csv("bowtie2dup.csv", sep = "")
bowtie2dup
dim(bowtie2dup)
duplicatedreads <- bowtie2dup$Marker[duplicated(bowtie2dup$Marker)]
noduplicatedreads <- setdiff(bowtie2dup$Marker,duplicatedreads)
noduplicatedreads

bowtie2nodup <-  bowtie2dup[bowtie2dup$Marker %in% noduplicatedreads,]
bowtie2nodup


both

linksnodup <- merge(bowtie2nodup, both, by.x = "Marker", by.y = "Band", sort = F)
linksnodup <- linksnodup[,c(2,3,4,5,6,7)]
linksnodup

RCircos.Link.Plot(linksnodup, track.num, TRUE)






linksnodup$PlotColor <- 1
linksnodup[linksnodup[,1]=="chr1", "PlotColor"] <- col_vector[1]
linksnodup[linksnodup[,1]=="chr2", "PlotColor"] <- col_vector[2]
linksnodup[linksnodup[,1]=="chr3", "PlotColor"] <- col_vector[3]
linksnodup[linksnodup[,1]=="chr4", "PlotColor"] <- col_vector[4]
linksnodup[linksnodup[,1]=="chr5", "PlotColor"] <- col_vector[5]
linksnodup[linksnodup[,1]=="chr6", "PlotColor"] <- col_vector[6]
linksnodup[linksnodup[,1]=="chr7", "PlotColor"] <- col_vector[7]
linksnodup[linksnodup[,1]=="chr8", "PlotColor"] <- col_vector[8]
linksnodup[linksnodup[,1]=="chr9", "PlotColor"] <- col_vector[9]
linksnodup[linksnodup[,1]=="chr10", "PlotColor"] <- col_vector[10]
linksnodup[linksnodup[,1]=="chr11", "PlotColor"] <- col_vector[11]
linksnodup[linksnodup[,1]=="chr12", "PlotColor"] <- col_vector[12]
linksnodup[linksnodup[,1]=="chr13", "PlotColor"] <- col_vector[13]
linksnodup[linksnodup[,1]=="chr14", "PlotColor"] <- col_vector[14]
linksnodup[linksnodup[,1]=="chr15", "PlotColor"] <- col_vector[15]
linksnodup[linksnodup[,1]=="chr16", "PlotColor"] <- col_vector[16]
linksnodup[linksnodup[,1]=="chr17", "PlotColor"] <- col_vector[17]
linksnodup[linksnodup[,1]=="chr18", "PlotColor"] <- col_vector[18]
linksnodup[linksnodup[,1]=="chr19", "PlotColor"] <- col_vector[19]
linksnodup[linksnodup[,1]=="chr20", "PlotColor"] <- col_vector[20]
linksnodup[linksnodup[,1]=="chr21", "PlotColor"] <- col_vector[21]
linksnodup[linksnodup[,1]=="chr22", "PlotColor"] <- col_vector[22]
linksnodup[linksnodup[,1]=="chr23", "PlotColor"] <- col_vector[23]
linksnodup[linksnodup[,1]=="chr24", "PlotColor"] <- col_vector[24]
linksnodup[linksnodup[,1]=="chr25", "PlotColor"] <- col_vector[25]
linksnodup[linksnodup[,1]=="chr26", "PlotColor"] <- col_vector[26]
linksnodup[linksnodup[,1]=="chr27", "PlotColor"] <- col_vector[27]
linksnodup[linksnodup[,1]=="chr28", "PlotColor"] <- col_vector[28]
linksnodup[linksnodup[,1]=="chr29", "PlotColor"] <- col_vector[29]

RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup, track.num = 1)

out.file <- "ComparativeGenomicsLG1nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG1",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG2nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG2",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG3nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG3",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG4nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG4",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG5nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG5",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG6nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG6",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG7nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG7",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG8nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG8",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG9nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG9",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG10nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG10",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG11nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG11",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG12nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG12",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG13nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG13",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG14nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG14",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG15nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG15",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG16nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG16",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG17nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG17",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG18nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG18",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG19nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG19",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG20nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG20",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG21nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG21",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG22nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG22",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG23nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG23",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG24nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG24",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG25nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG25",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG26nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG26",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG27nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG27",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG28nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG28",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG29nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG29",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG30nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG30",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG31nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG31",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG32nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG32",], track.num, TRUE)
dev.off()




out.file <- "ComparativeGenomicschr1nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr1",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr2nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr2",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr3nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr3",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr4nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr4",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr5nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr5",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr6nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr6",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr7nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr7",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr8nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr8",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr9nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr9",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr10nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr10",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr11nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr11",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr12nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr12",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr13nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr13",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr14nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr14",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr15nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr15",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr16nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr16",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr17nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr17",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr18nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr18",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr19nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr19",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr20nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr20",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr21nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr21",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr22nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr22",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr23nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr23",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr24nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr24",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr25nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr25",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr26nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr26",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr27nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr27",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr28nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr28",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr29nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr29",], track.num, TRUE)
dev.off()


```


Defining the Carex little-dalei chromosomes

```{r}
chr1 <- linksnodup[linksnodup$Chromosome.x=="chr1",]
chr1 <- chr1[order(chr1$chromStart),]
chr1

chr2 <- linksnodup[linksnodup$Chromosome.x=="chr2",]
chr2 <- chr2[order(chr2$chromStart),]
chr2

chr3 <- linksnodup[linksnodup$Chromosome.x=="chr3",]
chr3 <- chr3[order(chr3$chromStart),]
chr3

chr4 <- linksnodup[linksnodup$Chromosome.x=="chr4",]
chr4 <- chr4[order(chr4$chromStart),]
chr4

chr5 <- linksnodup[linksnodup$Chromosome.x=="chr5",]
chr5 <- chr5[order(chr5$chromStart),]
chr5

chr6 <- linksnodup[linksnodup$Chromosome.x=="chr6",]
chr6 <- chr6[order(chr6$chromStart),]
chr6

chr7 <- linksnodup[linksnodup$Chromosome.x=="chr7",]
chr7 <- chr7[order(chr7$chromStart),]
chr7

chr8 <- linksnodup[linksnodup$Chromosome.x=="chr8",]
chr8 <- chr8[order(chr8$chromStart),]
chr8

chr9 <- linksnodup[linksnodup$Chromosome.x=="chr9",]
chr9 <- chr9[order(chr9$chromStart),]
chr9

chr10 <- linksnodup[linksnodup$Chromosome.x=="chr10",]
chr10 <- chr10[order(chr10$chromStart),]
chr10

chr11 <- linksnodup[linksnodup$Chromosome.x=="chr11",]
chr11 <- chr11[order(chr11$chromStart),]
chr11

chr12 <- linksnodup[linksnodup$Chromosome.x=="chr12",]
chr12 <- chr12[order(chr12$chromStart),]
chr12

chr13 <- linksnodup[linksnodup$Chromosome.x=="chr13",]
chr13 <- chr13[order(chr13$chromStart),]
chr13

chr14 <- linksnodup[linksnodup$Chromosome.x=="chr14",]
chr14 <- chr14[order(chr14$chromStart),]
chr14

chr15 <- linksnodup[linksnodup$Chromosome.x=="chr15",]
chr15 <- chr15[order(chr15$chromStart),]
chr15

chr16 <- linksnodup[linksnodup$Chromosome.x=="chr16",]
chr16 <- chr16[order(chr16$chromStart),]
chr16

chr17 <- linksnodup[linksnodup$Chromosome.x=="chr17",]
chr17 <- chr17[order(chr17$chromStart),]
chr17

chr18 <- linksnodup[linksnodup$Chromosome.x=="chr18",]
chr18 <- chr18[order(chr18$chromStart),]
chr18

chr19 <- linksnodup[linksnodup$Chromosome.x=="chr19",]
chr19 <- chr19[order(chr19$chromStart),]
chr19

chr20 <- linksnodup[linksnodup$Chromosome.x=="chr20",]
chr20 <- chr20[order(chr20$chromStart),]
chr20

chr21 <- linksnodup[linksnodup$Chromosome.x=="chr21",]
chr21 <- chr21[order(chr21$chromStart),]
chr21

chr22 <- linksnodup[linksnodup$Chromosome.x=="chr22",]
chr22 <- chr22[order(chr22$chromStart),]
chr22

chr23 <- linksnodup[linksnodup$Chromosome.x=="chr23",]
chr23 <- chr23[order(chr23$chromStart),]
chr23

chr24 <- linksnodup[linksnodup$Chromosome.x=="chr24",]
chr24 <- chr24[order(chr24$chromStart),]
chr24

chr25 <- linksnodup[linksnodup$Chromosome.x=="chr25",]
chr25 <- chr25[order(chr25$chromStart),]
chr25

chr26 <- linksnodup[linksnodup$Chromosome.x=="chr26",]
chr26 <- chr26[order(chr26$chromStart),]
chr26

chr27 <- linksnodup[linksnodup$Chromosome.x=="chr27",]
chr27 <- chr27[order(chr27$chromStart),]
chr27

chr28 <- linksnodup[linksnodup$Chromosome.x=="chr28",]
chr28 <- chr28[order(chr28$chromStart),]
chr28

chr29 <- linksnodup[linksnodup$Chromosome.x=="chr29",]
chr29 <- chr29[order(chr29$chromStart),]
chr29



```

Defining blocks

```{r}

source("genomeblocks.R")

outblock1 <- genomeblocks(chr1)
outblock1
outblock2 <- genomeblocks(chr2)
outblock2
outblock3 <- genomeblocks(chr3)
outblock3
outblock4 <- genomeblocks(chr4)
outblock4
outblock5 <- genomeblocks(chr5)
outblock5
outblock6 <- genomeblocks(chr6)
outblock6
outblock7 <- genomeblocks(chr7)
outblock7
outblock8 <- genomeblocks(chr8)
outblock8
outblock9 <- genomeblocks(chr9)
outblock9
outblock10 <- genomeblocks(chr10)
outblock10
outblock11 <- genomeblocks(chr11)
outblock11
outblock12 <- genomeblocks(chr12)
outblock12
outblock13 <- genomeblocks(chr13)
outblock13
outblock14 <- genomeblocks(chr14)
outblock14
outblock15 <- genomeblocks(chr15)
outblock15
outblock16 <- genomeblocks(chr16)
outblock16
outblock17 <- genomeblocks(chr17)
outblock17
outblock18 <- genomeblocks(chr18)
outblock18
outblock19 <- genomeblocks(chr19)
outblock19
outblock20 <- genomeblocks(chr20)
outblock20
outblock21 <- genomeblocks(chr21)
outblock21
outblock22 <- genomeblocks(chr22)
outblock22
outblock23 <- genomeblocks(chr23)
outblock23
outblock24 <- genomeblocks(chr24)
outblock24
outblock25 <- genomeblocks(chr25)
outblock25
outblock26 <- genomeblocks(chr26)
outblock26
outblock27 <- genomeblocks(chr27)
outblock27
outblock28 <- genomeblocks(chr28)
outblock28
outblock29 <- genomeblocks(chr29)
outblock29


```

Let's search for BCR in the blocks and interblocks with the BCR provide by the authors of the paper with Carex little-dalei genome. These analyses were done with Blastn.

```{r}
BCR <- read.csv("BCR.csv")
#Somewhat similar sequences (blastn), default

BCR1 <- BCR[BCR$Chromosome=="chr1",]
BCR1
BCR2 <- BCR[BCR$Chromosome=="chr2",]
BCR2
BCR3 <- BCR[BCR$Chromosome=="chr3",]
BCR3
BCR4 <- BCR[BCR$Chromosome=="chr4",]
BCR4
BCR5 <- BCR[BCR$Chromosome=="chr5",]
BCR5
BCR6 <- BCR[BCR$Chromosome=="chr6",]
BCR6
BCR7 <- BCR[BCR$Chromosome=="chr7",]
BCR7
BCR8 <- BCR[BCR$Chromosome=="chr8",]
BCR8
BCR9 <- BCR[BCR$Chromosome=="chr9",]
BCR9
BCR10 <- BCR[BCR$Chromosome=="chr10",]
BCR10
BCR11 <- BCR[BCR$Chromosome=="chr11",]
BCR11
BCR12 <- BCR[BCR$Chromosome=="chr12",]
BCR12
BCR13 <- BCR[BCR$Chromosome=="chr13",]
BCR13
BCR14 <- BCR[BCR$Chromosome=="chr14",]
BCR14
BCR15 <- BCR[BCR$Chromosome=="chr15",]
BCR15
BCR16 <- BCR[BCR$Chromosome=="chr16",]
BCR16
BCR17 <- BCR[BCR$Chromosome=="chr17",]
BCR17
BCR18 <- BCR[BCR$Chromosome=="chr18",]
BCR18
BCR19 <- BCR[BCR$Chromosome=="chr19",]
BCR19
BCR20 <- BCR[BCR$Chromosome=="chr20",]
BCR20
BCR21 <- BCR[BCR$Chromosome=="chr21",]
BCR21
BCR22 <- BCR[BCR$Chromosome=="chr22",]
BCR22
BCR23 <- BCR[BCR$Chromosome=="chr23",]
BCR23
BCR24 <- BCR[BCR$Chromosome=="chr24",]
BCR24
BCR25 <- BCR[BCR$Chromosome=="chr25",]
BCR25
BCR26 <- BCR[BCR$Chromosome=="chr26",]
BCR26
BCR27 <- BCR[BCR$Chromosome=="chr27",]
BCR27
BCR28 <- BCR[BCR$Chromosome=="chr28",]
BCR28
BCR29 <- BCR[BCR$Chromosome=="chr29",]
BCR29

outblock1

source("genomeblocksBCR.R") 

genomeblocksBCR1 <- genomeblocksBCR(outblock1, BCR1)
genomeblocksBCR1
genomeblocksBCR2 <- genomeblocksBCR(outblock2, BCR2)
genomeblocksBCR2
genomeblocksBCR3 <- genomeblocksBCR(outblock3, BCR3)
genomeblocksBCR3
genomeblocksBCR4 <- genomeblocksBCR(outblock4, BCR4)
genomeblocksBCR4
genomeblocksBCR5 <- genomeblocksBCR(outblock5, BCR5)
genomeblocksBCR5
genomeblocksBCR6 <- genomeblocksBCR(outblock6, BCR6)
genomeblocksBCR6
genomeblocksBCR7 <- genomeblocksBCR(outblock7, BCR7)
genomeblocksBCR7
genomeblocksBCR8 <- genomeblocksBCR(outblock8, BCR8)
genomeblocksBCR8
genomeblocksBCR9 <- genomeblocksBCR(outblock9, BCR9)
genomeblocksBCR9
genomeblocksBCR10 <- genomeblocksBCR(outblock10, BCR10)
genomeblocksBCR10
genomeblocksBCR11 <- genomeblocksBCR(outblock11, BCR11)
genomeblocksBCR11
genomeblocksBCR12 <- genomeblocksBCR(outblock12, BCR12)
genomeblocksBCR12
genomeblocksBCR13 <- genomeblocksBCR(outblock13, BCR13)
genomeblocksBCR13
genomeblocksBCR14 <- genomeblocksBCR(outblock14, BCR14)
genomeblocksBCR14
genomeblocksBCR15 <- genomeblocksBCR(outblock15, BCR15)
genomeblocksBCR15
genomeblocksBCR16 <- genomeblocksBCR(outblock16, BCR16)
genomeblocksBCR16
genomeblocksBCR17 <- genomeblocksBCR(outblock17, BCR17)
genomeblocksBCR17
genomeblocksBCR18 <- genomeblocksBCR(outblock18, BCR18)
genomeblocksBCR18
genomeblocksBCR19 <- genomeblocksBCR(outblock19, BCR19)
genomeblocksBCR19
genomeblocksBCR20 <- genomeblocksBCR(outblock20, BCR20)
genomeblocksBCR20
genomeblocksBCR21 <- genomeblocksBCR(outblock21, BCR21)
genomeblocksBCR21
genomeblocksBCR22 <- genomeblocksBCR(outblock22, BCR22)
genomeblocksBCR22
genomeblocksBCR23 <- genomeblocksBCR(outblock23, BCR23)
genomeblocksBCR23
genomeblocksBCR24 <- genomeblocksBCR(outblock24, BCR24)
genomeblocksBCR24
genomeblocksBCR25 <- genomeblocksBCR(outblock25, BCR25)
genomeblocksBCR25
genomeblocksBCR26 <- genomeblocksBCR(outblock26, BCR26)
genomeblocksBCR26
genomeblocksBCR27 <- genomeblocksBCR(outblock27, BCR27)
genomeblocksBCR27
genomeblocksBCR28 <- genomeblocksBCR(outblock28, BCR28)
genomeblocksBCR28
genomeblocksBCR29 <- genomeblocksBCR(outblock29, BCR29)
genomeblocksBCR29

genomeblocksBCRall <- rbind(genomeblocksBCR1,genomeblocksBCR2,genomeblocksBCR3,genomeblocksBCR4,genomeblocksBCR5,genomeblocksBCR6,genomeblocksBCR7,genomeblocksBCR8,genomeblocksBCR9,genomeblocksBCR10,genomeblocksBCR11,genomeblocksBCR12,genomeblocksBCR13,genomeblocksBCR14,genomeblocksBCR15,genomeblocksBCR16,genomeblocksBCR17,genomeblocksBCR18,genomeblocksBCR19,genomeblocksBCR20,genomeblocksBCR21,genomeblocksBCR22,genomeblocksBCR23,genomeblocksBCR24,genomeblocksBCR25,genomeblocksBCR26,genomeblocksBCR27,genomeblocksBCR28,genomeblocksBCR29)

genomeblocksBCRall
boxplot(as.numeric(genomeblocksBCRall[,"BCR"]) ~ genomeblocksBCRall[,"type"])

anova(lm(as.numeric(genomeblocksBCRall[,"BCR"]) ~ as.character(genomeblocksBCRall[,"type"])))
summary(aov(as.numeric(genomeblocksBCRall[,"BCR"]) ~ as.character(genomeblocksBCRall[,"type"])))


#MORE RELAXED SEARCH FOR BCR
BCR2 <- read.csv("BCR2.csv")
#Somewhat similar sequences (blastn), Word size 7, Match/Mitmatch Score 4,-5, 

BCR2_1 <- BCR2[BCR2$Chromosome=="chr1",]
BCR2_1
BCR2_2 <- BCR2[BCR2$Chromosome=="chr2",]
BCR2_2
BCR2_3 <- BCR2[BCR2$Chromosome=="chr3",]
BCR2_3
BCR2_4 <- BCR2[BCR2$Chromosome=="chr4",]
BCR2_4
BCR2_5 <- BCR2[BCR2$Chromosome=="chr5",]
BCR2_5
BCR2_6 <- BCR2[BCR2$Chromosome=="chr6",]
BCR2_6
BCR2_7 <- BCR2[BCR2$Chromosome=="chr7",]
BCR2_7
BCR2_8 <- BCR2[BCR2$Chromosome=="chr8",]
BCR2_8
BCR2_9 <- BCR2[BCR2$Chromosome=="chr9",]
BCR2_9
BCR2_10 <- BCR2[BCR2$Chromosome=="chr10",]
BCR2_10
BCR2_11 <- BCR2[BCR2$Chromosome=="chr11",]
BCR2_11
BCR2_12 <- BCR2[BCR2$Chromosome=="chr12",]
BCR2_12
BCR2_13 <- BCR2[BCR2$Chromosome=="chr13",]
BCR2_13
BCR2_14 <- BCR2[BCR2$Chromosome=="chr14",]
BCR2_14
BCR2_15 <- BCR2[BCR2$Chromosome=="chr15",]
BCR2_15
BCR2_16 <- BCR2[BCR2$Chromosome=="chr16",]
BCR2_16
BCR2_17 <- BCR2[BCR2$Chromosome=="chr17",]
BCR2_17
BCR2_18 <- BCR2[BCR2$Chromosome=="chr18",]
BCR2_18
BCR2_19 <- BCR2[BCR2$Chromosome=="chr19",]
BCR2_19
BCR2_20 <- BCR2[BCR2$Chromosome=="chr20",]
BCR2_20
BCR2_21 <- BCR2[BCR2$Chromosome=="chr21",]
BCR2_21
BCR2_22 <- BCR2[BCR2$Chromosome=="chr22",]
BCR2_22
BCR2_23 <- BCR2[BCR2$Chromosome=="chr23",]
BCR2_23
BCR2_24 <- BCR2[BCR2$Chromosome=="chr24",]
BCR2_24
BCR2_25 <- BCR2[BCR2$Chromosome=="chr25",]
BCR2_25
BCR2_26 <- BCR2[BCR2$Chromosome=="chr26",]
BCR2_26
BCR2_27 <- BCR2[BCR2$Chromosome=="chr27",]
BCR2_27
BCR2_28 <- BCR2[BCR2$Chromosome=="chr28",]
BCR2_28
BCR2_29 <- BCR2[BCR2$Chromosome=="chr29",]
BCR2_29


genomeblocksBCR2_1 <- genomeblocksBCR(outblock1, BCR2_1)
genomeblocksBCR2_1
genomeblocksBCR2_2 <- genomeblocksBCR(outblock2, BCR2_2)
genomeblocksBCR2_2
genomeblocksBCR2_3 <- genomeblocksBCR(outblock3, BCR2_3)
genomeblocksBCR2_3
genomeblocksBCR2_4 <- genomeblocksBCR(outblock4, BCR2_4)
genomeblocksBCR2_4
genomeblocksBCR2_5 <- genomeblocksBCR(outblock5, BCR2_5)
genomeblocksBCR2_5
genomeblocksBCR2_6 <- genomeblocksBCR(outblock6, BCR2_6)
genomeblocksBCR2_6
genomeblocksBCR2_7 <- genomeblocksBCR(outblock7, BCR2_7)
genomeblocksBCR2_7
genomeblocksBCR2_8 <- genomeblocksBCR(outblock8, BCR2_8)
genomeblocksBCR2_8
genomeblocksBCR2_9 <- genomeblocksBCR(outblock9, BCR2_9)
genomeblocksBCR2_9
genomeblocksBCR2_10 <- genomeblocksBCR(outblock10, BCR2_10)
genomeblocksBCR2_10
genomeblocksBCR2_11 <- genomeblocksBCR(outblock11, BCR2_11)
genomeblocksBCR2_11
genomeblocksBCR2_12 <- genomeblocksBCR(outblock12, BCR2_12)
genomeblocksBCR2_12
genomeblocksBCR2_13 <- genomeblocksBCR(outblock13, BCR2_13)
genomeblocksBCR2_13
genomeblocksBCR2_14 <- genomeblocksBCR(outblock14, BCR2_14)
genomeblocksBCR2_14
genomeblocksBCR2_15 <- genomeblocksBCR(outblock15, BCR2_15)
genomeblocksBCR2_15
genomeblocksBCR2_16 <- genomeblocksBCR(outblock16, BCR2_16)
genomeblocksBCR2_16
genomeblocksBCR2_17 <- genomeblocksBCR(outblock17, BCR2_17)
genomeblocksBCR2_17
genomeblocksBCR2_18 <- genomeblocksBCR(outblock18, BCR2_18)
genomeblocksBCR2_18
genomeblocksBCR2_19 <- genomeblocksBCR(outblock19, BCR2_19)
genomeblocksBCR2_19
genomeblocksBCR2_20 <- genomeblocksBCR(outblock20, BCR2_20)
genomeblocksBCR2_20
genomeblocksBCR2_21 <- genomeblocksBCR(outblock21, BCR2_21)
genomeblocksBCR2_21
genomeblocksBCR2_22 <- genomeblocksBCR(outblock22, BCR2_22)
genomeblocksBCR2_22
genomeblocksBCR2_23 <- genomeblocksBCR(outblock23, BCR2_23)
genomeblocksBCR2_23
genomeblocksBCR2_24 <- genomeblocksBCR(outblock24, BCR2_24)
genomeblocksBCR2_24
genomeblocksBCR2_25 <- genomeblocksBCR(outblock25, BCR2_25)
genomeblocksBCR2_25
genomeblocksBCR2_26 <- genomeblocksBCR(outblock26, BCR2_26)
genomeblocksBCR2_26
genomeblocksBCR2_27 <- genomeblocksBCR(outblock27, BCR2_27)
genomeblocksBCR2_27
genomeblocksBCR2_28 <- genomeblocksBCR(outblock28, BCR2_28)
genomeblocksBCR2_28
genomeblocksBCR2_29 <- genomeblocksBCR(outblock29, BCR2_29)
genomeblocksBCR2_29

genomeblocksBCR2_all <- rbind(genomeblocksBCR2_1,genomeblocksBCR2_2,genomeblocksBCR2_3,genomeblocksBCR2_4,genomeblocksBCR2_5,genomeblocksBCR2_6,genomeblocksBCR2_7,genomeblocksBCR2_8,genomeblocksBCR2_9,genomeblocksBCR2_10,genomeblocksBCR2_11,genomeblocksBCR2_12,genomeblocksBCR2_13,genomeblocksBCR2_14,genomeblocksBCR2_15,genomeblocksBCR2_16,genomeblocksBCR2_17,genomeblocksBCR2_18,genomeblocksBCR2_19,genomeblocksBCR2_20,genomeblocksBCR2_21,genomeblocksBCR2_22,genomeblocksBCR2_23,genomeblocksBCR2_24,genomeblocksBCR2_25,genomeblocksBCR2_26,genomeblocksBCR2_27,genomeblocksBCR2_28,genomeblocksBCR2_29)

genomeblocksBCR2_all
boxplot(as.numeric(genomeblocksBCR2_all[,"BCR"]) ~ genomeblocksBCR2_all[,"type"])

anova(lm(as.numeric(genomeblocksBCR2_all[,"BCR"]) ~ as.character(genomeblocksBCR2_all[,"type"])))
summary(aov(as.numeric(genomeblocksBCR2_all[,"BCR"]) ~ as.character(genomeblocksBCR2_all[,"type"])))
```


LET'S USE ANDR CENTROMERIC REPEAT
We will use this one in the paper.

```{r}
BCR3 <- read.csv("BCR3.csv")
#Somewhat similar sequences (blastn), default

BCR3_1 <- BCR3[BCR3$Chromosome=="chr1",]
BCR3_1
BCR3_2 <- BCR3[BCR3$Chromosome=="chr2",]
BCR3_2
BCR3_3 <- BCR3[BCR3$Chromosome=="chr3",]
BCR3_3
BCR3_4 <- BCR3[BCR3$Chromosome=="chr4",]
BCR3_4
BCR3_5 <- BCR3[BCR3$Chromosome=="chr5",]
BCR3_5
BCR3_6 <- BCR3[BCR3$Chromosome=="chr6",]
BCR3_6
BCR3_7 <- BCR3[BCR3$Chromosome=="chr7",]
BCR3_7
BCR3_8 <- BCR3[BCR3$Chromosome=="chr8",]
BCR3_8
BCR3_9 <- BCR3[BCR3$Chromosome=="chr9",]
BCR3_9
BCR3_10 <- BCR3[BCR3$Chromosome=="chr10",]
BCR3_10
BCR3_11 <- BCR3[BCR3$Chromosome=="chr11",]
BCR3_11
BCR3_12 <- BCR3[BCR3$Chromosome=="chr12",]
BCR3_12
BCR3_13 <- BCR3[BCR3$Chromosome=="chr13",]
BCR3_13
BCR3_14 <- BCR3[BCR3$Chromosome=="chr14",]
BCR3_14
BCR3_15 <- BCR3[BCR3$Chromosome=="chr15",]
BCR3_15
BCR3_16 <- BCR3[BCR3$Chromosome=="chr16",]
BCR3_16
BCR3_17 <- BCR3[BCR3$Chromosome=="chr17",]
BCR3_17
BCR3_18 <- BCR3[BCR3$Chromosome=="chr18",]
BCR3_18
BCR3_19 <- BCR3[BCR3$Chromosome=="chr19",]
BCR3_19
BCR3_20 <- BCR3[BCR3$Chromosome=="chr20",]
BCR3_20
BCR3_21 <- BCR3[BCR3$Chromosome=="chr21",]
BCR3_21
BCR3_22 <- BCR3[BCR3$Chromosome=="chr22",]
BCR3_22
BCR3_23 <- BCR3[BCR3$Chromosome=="chr23",]
BCR3_23
BCR3_24 <- BCR3[BCR3$Chromosome=="chr24",]
BCR3_24
BCR3_25 <- BCR3[BCR3$Chromosome=="chr25",]
BCR3_25
BCR3_26 <- BCR3[BCR3$Chromosome=="chr26",]
BCR3_26
BCR3_27 <- BCR3[BCR3$Chromosome=="chr27",]
BCR3_27
BCR3_28 <- BCR3[BCR3$Chromosome=="chr28",]
BCR3_28
BCR3_29 <- BCR3[BCR3$Chromosome=="chr29",]
BCR3_29


genomeblocksBCR3_1 <- genomeblocksBCR(outblock1, BCR3_1)
genomeblocksBCR3_1
genomeblocksBCR3_2 <- genomeblocksBCR(outblock2, BCR3_2)
genomeblocksBCR3_2
genomeblocksBCR3_3 <- genomeblocksBCR(outblock3, BCR3_3)
genomeblocksBCR3_3
genomeblocksBCR3_4 <- genomeblocksBCR(outblock4, BCR3_4)
genomeblocksBCR3_4
genomeblocksBCR3_5 <- genomeblocksBCR(outblock5, BCR3_5)
genomeblocksBCR3_5
genomeblocksBCR3_6 <- genomeblocksBCR(outblock6, BCR3_6)
genomeblocksBCR3_6
genomeblocksBCR3_7 <- genomeblocksBCR(outblock7, BCR3_7)
genomeblocksBCR3_7
genomeblocksBCR3_8 <- genomeblocksBCR(outblock8, BCR3_8)
genomeblocksBCR3_8
genomeblocksBCR3_9 <- genomeblocksBCR(outblock9, BCR3_9)
genomeblocksBCR3_9
genomeblocksBCR3_10 <- genomeblocksBCR(outblock10, BCR3_10)
genomeblocksBCR3_10
genomeblocksBCR3_11 <- genomeblocksBCR(outblock11, BCR3_11)
genomeblocksBCR3_11
genomeblocksBCR3_12 <- genomeblocksBCR(outblock12, BCR3_12)
genomeblocksBCR3_12
genomeblocksBCR3_13 <- genomeblocksBCR(outblock13, BCR3_13)
genomeblocksBCR3_13
genomeblocksBCR3_14 <- genomeblocksBCR(outblock14, BCR3_14)
genomeblocksBCR3_14
genomeblocksBCR3_15 <- genomeblocksBCR(outblock15, BCR3_15)
genomeblocksBCR3_15
genomeblocksBCR3_16 <- genomeblocksBCR(outblock16, BCR3_16)
genomeblocksBCR3_16
genomeblocksBCR3_17 <- genomeblocksBCR(outblock17, BCR3_17)
genomeblocksBCR3_17
genomeblocksBCR3_18 <- genomeblocksBCR(outblock18, BCR3_18)
genomeblocksBCR3_18
genomeblocksBCR3_19 <- genomeblocksBCR(outblock19, BCR3_19)
genomeblocksBCR3_19
genomeblocksBCR3_20 <- genomeblocksBCR(outblock20, BCR3_20)
genomeblocksBCR3_20
genomeblocksBCR3_21 <- genomeblocksBCR(outblock21, BCR3_21)
genomeblocksBCR3_21
genomeblocksBCR3_22 <- genomeblocksBCR(outblock22, BCR3_22)
genomeblocksBCR3_22
genomeblocksBCR3_23 <- genomeblocksBCR(outblock23, BCR3_23)
genomeblocksBCR3_23
genomeblocksBCR3_24 <- genomeblocksBCR(outblock24, BCR3_24)
genomeblocksBCR3_24
genomeblocksBCR3_25 <- genomeblocksBCR(outblock25, BCR3_25)
genomeblocksBCR3_25
genomeblocksBCR3_26 <- genomeblocksBCR(outblock26, BCR3_26)
genomeblocksBCR3_26
genomeblocksBCR3_27 <- genomeblocksBCR(outblock27, BCR3_27)
genomeblocksBCR3_27
genomeblocksBCR3_28 <- genomeblocksBCR(outblock28, BCR3_28)
genomeblocksBCR3_28
genomeblocksBCR3_29 <- genomeblocksBCR(outblock29, BCR3_29)
genomeblocksBCR3_29

genomeblocksBCR3_all <- rbind(genomeblocksBCR3_1,genomeblocksBCR3_2,genomeblocksBCR3_3,genomeblocksBCR3_4,genomeblocksBCR3_5,genomeblocksBCR3_6,genomeblocksBCR3_7,genomeblocksBCR3_8,genomeblocksBCR3_9,genomeblocksBCR3_10,genomeblocksBCR3_11,genomeblocksBCR3_12,genomeblocksBCR3_13,genomeblocksBCR3_14,genomeblocksBCR3_15,genomeblocksBCR3_16,genomeblocksBCR3_17,genomeblocksBCR3_18,genomeblocksBCR3_19,genomeblocksBCR3_20,genomeblocksBCR3_21,genomeblocksBCR3_22,genomeblocksBCR3_23,genomeblocksBCR3_24,genomeblocksBCR3_25,genomeblocksBCR3_26,genomeblocksBCR3_27,genomeblocksBCR3_28,genomeblocksBCR3_29)

genomeblocksBCR3_all
boxplot(as.numeric(genomeblocksBCR3_all[,"BCR"]) ~ genomeblocksBCR3_all[,"type"])

anova(lm(as.numeric(genomeblocksBCR3_all[,"BCR"]) ~ as.character(genomeblocksBCR3_all[,"type"])))
summary(aov(as.numeric(genomeblocksBCR3_all[,"BCR"]) ~ as.character(genomeblocksBCR3_all[,"type"])))


BCR4 <- read.csv("BCR4.csv")
#megablast, default

BCR4_1 <- BCR4[BCR4$Chromosome=="chr1",]
BCR4_1
BCR4_2 <- BCR4[BCR4$Chromosome=="chr2",]
BCR4_2
BCR4_3 <- BCR4[BCR4$Chromosome=="chr3",]
BCR4_3
BCR4_4 <- BCR4[BCR4$Chromosome=="chr4",]
BCR4_4
BCR4_5 <- BCR4[BCR4$Chromosome=="chr5",]
BCR4_5
BCR4_6 <- BCR4[BCR4$Chromosome=="chr6",]
BCR4_6
BCR4_7 <- BCR4[BCR4$Chromosome=="chr7",]
BCR4_7
BCR4_8 <- BCR4[BCR4$Chromosome=="chr8",]
BCR4_8
BCR4_9 <- BCR4[BCR4$Chromosome=="chr9",]
BCR4_9
BCR4_10 <- BCR4[BCR4$Chromosome=="chr10",]
BCR4_10
BCR4_11 <- BCR4[BCR4$Chromosome=="chr11",]
BCR4_11
BCR4_12 <- BCR4[BCR4$Chromosome=="chr12",]
BCR4_12
BCR4_13 <- BCR4[BCR4$Chromosome=="chr13",]
BCR4_13
BCR4_14 <- BCR4[BCR4$Chromosome=="chr14",]
BCR4_14
BCR4_15 <- BCR4[BCR4$Chromosome=="chr15",]
BCR4_15
BCR4_16 <- BCR4[BCR4$Chromosome=="chr16",]
BCR4_16
BCR4_17 <- BCR4[BCR4$Chromosome=="chr17",]
BCR4_17
BCR4_18 <- BCR4[BCR4$Chromosome=="chr18",]
BCR4_18
BCR4_19 <- BCR4[BCR4$Chromosome=="chr19",]
BCR4_19
BCR4_20 <- BCR4[BCR4$Chromosome=="chr20",]
BCR4_20
BCR4_21 <- BCR4[BCR4$Chromosome=="chr21",]
BCR4_21
BCR4_22 <- BCR4[BCR4$Chromosome=="chr22",]
BCR4_22
BCR4_23 <- BCR4[BCR4$Chromosome=="chr23",]
BCR4_23
BCR4_24 <- BCR4[BCR4$Chromosome=="chr24",]
BCR4_24
BCR4_25 <- BCR4[BCR4$Chromosome=="chr25",]
BCR4_25
BCR4_26 <- BCR4[BCR4$Chromosome=="chr26",]
BCR4_26
BCR4_27 <- BCR4[BCR4$Chromosome=="chr27",]
BCR4_27
BCR4_28 <- BCR4[BCR4$Chromosome=="chr28",]
BCR4_28
BCR4_29 <- BCR4[BCR4$Chromosome=="chr29",]
BCR4_29


genomeblocksBCR4_1 <- genomeblocksBCR(outblock1, BCR4_1)
genomeblocksBCR4_1
genomeblocksBCR4_2 <- genomeblocksBCR(outblock2, BCR4_2)
genomeblocksBCR4_2
genomeblocksBCR4_3 <- genomeblocksBCR(outblock3, BCR4_3)
genomeblocksBCR4_3
genomeblocksBCR4_4 <- genomeblocksBCR(outblock4, BCR4_4)
genomeblocksBCR4_4
genomeblocksBCR4_5 <- genomeblocksBCR(outblock5, BCR4_5)
genomeblocksBCR4_5
genomeblocksBCR4_6 <- genomeblocksBCR(outblock6, BCR4_6)
genomeblocksBCR4_6
genomeblocksBCR4_7 <- genomeblocksBCR(outblock7, BCR4_7)
genomeblocksBCR4_7
genomeblocksBCR4_8 <- genomeblocksBCR(outblock8, BCR4_8)
genomeblocksBCR4_8
genomeblocksBCR4_9 <- genomeblocksBCR(outblock9, BCR4_9)
genomeblocksBCR4_9
genomeblocksBCR4_10 <- genomeblocksBCR(outblock10, BCR4_10)
genomeblocksBCR4_10
genomeblocksBCR4_11 <- genomeblocksBCR(outblock11, BCR4_11)
genomeblocksBCR4_11
genomeblocksBCR4_12 <- genomeblocksBCR(outblock12, BCR4_12)
genomeblocksBCR4_12
genomeblocksBCR4_13 <- genomeblocksBCR(outblock13, BCR4_13)
genomeblocksBCR4_13
genomeblocksBCR4_14 <- genomeblocksBCR(outblock14, BCR4_14)
genomeblocksBCR4_14
genomeblocksBCR4_15 <- genomeblocksBCR(outblock15, BCR4_15)
genomeblocksBCR4_15
genomeblocksBCR4_16 <- genomeblocksBCR(outblock16, BCR4_16)
genomeblocksBCR4_16
genomeblocksBCR4_17 <- genomeblocksBCR(outblock17, BCR4_17)
genomeblocksBCR4_17
genomeblocksBCR4_18 <- genomeblocksBCR(outblock18, BCR4_18)
genomeblocksBCR4_18
genomeblocksBCR4_19 <- genomeblocksBCR(outblock19, BCR4_19)
genomeblocksBCR4_19
genomeblocksBCR4_20 <- genomeblocksBCR(outblock20, BCR4_20)
genomeblocksBCR4_20
genomeblocksBCR4_21 <- genomeblocksBCR(outblock21, BCR4_21)
genomeblocksBCR4_21
genomeblocksBCR4_22 <- genomeblocksBCR(outblock22, BCR4_22)
genomeblocksBCR4_22
genomeblocksBCR4_23 <- genomeblocksBCR(outblock23, BCR4_23)
genomeblocksBCR4_23
genomeblocksBCR4_24 <- genomeblocksBCR(outblock24, BCR4_24)
genomeblocksBCR4_24
genomeblocksBCR4_25 <- genomeblocksBCR(outblock25, BCR4_25)
genomeblocksBCR4_25
genomeblocksBCR4_26 <- genomeblocksBCR(outblock26, BCR4_26)
genomeblocksBCR4_26
genomeblocksBCR4_27 <- genomeblocksBCR(outblock27, BCR4_27)
genomeblocksBCR4_27
genomeblocksBCR4_28 <- genomeblocksBCR(outblock28, BCR4_28)
genomeblocksBCR4_28
genomeblocksBCR4_29 <- genomeblocksBCR(outblock29, BCR4_29)
genomeblocksBCR4_29

genomeblocksBCR4_all <- rbind(genomeblocksBCR4_1,genomeblocksBCR4_2,genomeblocksBCR4_3,genomeblocksBCR4_4,genomeblocksBCR4_5,genomeblocksBCR4_6,genomeblocksBCR4_7,genomeblocksBCR4_8,genomeblocksBCR4_9,genomeblocksBCR4_10,genomeblocksBCR4_11,genomeblocksBCR4_12,genomeblocksBCR4_13,genomeblocksBCR4_14,genomeblocksBCR4_15,genomeblocksBCR4_16,genomeblocksBCR4_17,genomeblocksBCR4_18,genomeblocksBCR4_19,genomeblocksBCR4_20,genomeblocksBCR4_21,genomeblocksBCR4_22,genomeblocksBCR4_23,genomeblocksBCR4_24,genomeblocksBCR4_25,genomeblocksBCR4_26,genomeblocksBCR4_27,genomeblocksBCR4_28,genomeblocksBCR4_29)

length(genomeblocksBCR4_all[,4] == "conserved")

genomeblocksBCR4_all
boxplot(as.numeric(genomeblocksBCR4_all[,"BCR"]) ~ genomeblocksBCR4_all[,"type"])

anova(lm(as.numeric(genomeblocksBCR4_all[,"BCR"]) ~ as.character(genomeblocksBCR4_all[,"type"])))
summary(aov(as.numeric(genomeblocksBCR4_all[,"BCR"]) ~ as.character(genomeblocksBCR4_all[,"type"])))
```


We are going to use the hits that have duplicates

```{r}
#We are using this in the publication

chr1d <- links[links$Chromosome.x=="chr1",]
chr1d <- chr1[order(chr1$chromStart),]
chr1d

chr2d <- links[links$Chromosome.x=="chr2",]
chr2d <- chr2[order(chr2$chromStart),]
chr2d

chr3d <- links[links$Chromosome.x=="chr3",]
chr3d <- chr3[order(chr3$chromStart),]
chr3d

chr4d <- links[links$Chromosome.x=="chr4",]
chr4d <- chr4[order(chr4$chromStart),]
chr4d

chr5d <- links[links$Chromosome.x=="chr5",]
chr5d <- chr5[order(chr5$chromStart),]
chr5d

chr6d <- links[links$Chromosome.x=="chr6",]
chr6d <- chr6[order(chr6$chromStart),]
chr6d

chr7d <- links[links$Chromosome.x=="chr7",]
chr7d <- chr7[order(chr7$chromStart),]
chr7d

chr8d <- links[links$Chromosome.x=="chr8",]
chr8d <- chr8[order(chr8$chromStart),]
chr8d

chr9d <- links[links$Chromosome.x=="chr9",]
chr9d <- chr9[order(chr9$chromStart),]
chr9d

chr10d <- links[links$Chromosome.x=="chr10",]
chr10d <- chr10[order(chr10$chromStart),]
chr10d

chr11d <- links[links$Chromosome.x=="chr11",]
chr11d <- chr11[order(chr11$chromStart),]
chr11d

chr12d <- links[links$Chromosome.x=="chr12",]
chr12d <- chr12[order(chr12$chromStart),]
chr12d

chr13d <- links[links$Chromosome.x=="chr13",]
chr13d <- chr13[order(chr13$chromStart),]
chr13d

chr14d <- links[links$Chromosome.x=="chr14",]
chr14d <- chr14[order(chr14$chromStart),]
chr14d

chr15d <- links[links$Chromosome.x=="chr15",]
chr15d <- chr15[order(chr15$chromStart),]
chr15d

chr16d <- links[links$Chromosome.x=="chr16",]
chr16d <- chr16[order(chr16$chromStart),]
chr16d

chr17d <- links[links$Chromosome.x=="chr17",]
chr17d <- chr17[order(chr17$chromStart),]
chr17d

chr18d <- links[links$Chromosome.x=="chr18",]
chr18d <- chr18[order(chr18$chromStart),]
chr18d

chr19d <- links[links$Chromosome.x=="chr19",]
chr19d <- chr19[order(chr19$chromStart),]
chr19d

chr20d <- links[links$Chromosome.x=="chr20",]
chr20d <- chr20[order(chr20$chromStart),]
chr20d

chr21d <- links[links$Chromosome.x=="chr21",]
chr21d <- chr21[order(chr21$chromStart),]
chr21d

chr22d <- links[links$Chromosome.x=="chr22",]
chr22d <- chr22[order(chr22$chromStart),]
chr22d

chr23d <- links[links$Chromosome.x=="chr23",]
chr23d <- chr23[order(chr23$chromStart),]
chr23d

chr24d <- links[links$Chromosome.x=="chr24",]
chr24d <- chr24[order(chr24$chromStart),]
chr24d

chr25d <- links[links$Chromosome.x=="chr25",]
chr25d <- chr25[order(chr25$chromStart),]
chr25d

chr26d <- links[links$Chromosome.x=="chr26",]
chr26d <- chr26[order(chr26$chromStart),]
chr26d

chr27d <- links[links$Chromosome.x=="chr27",]
chr27d <- chr27[order(chr27$chromStart),]
chr27d

chr28d <- links[links$Chromosome.x=="chr28",]
chr28d <- chr28[order(chr28$chromStart),]
chr28d

chr29d <- links[links$Chromosome.x=="chr29",]
chr29d <- chr29[order(chr29$chromStart),]
chr29d

outblock1d <- genomeblocks(chr1d)
outblock1d
outblock2d <- genomeblocks(chr2d)
outblock2d
outblock3d <- genomeblocks(chr3d)
outblock3d
outblock4d <- genomeblocks(chr4d)
outblock4d
outblock5d <- genomeblocks(chr5d)
outblock5d
outblock6d <- genomeblocks(chr6d)
outblock6d
outblock7d <- genomeblocks(chr7d)
outblock7d
outblock8d <- genomeblocks(chr8d)
outblock8d
outblock9d <- genomeblocks(chr9d)
outblock9d
outblock10d <- genomeblocks(chr10d)
outblock10d
outblock11d <- genomeblocks(chr11d)
outblock11d
outblock12d <- genomeblocks(chr12d)
outblock12d
outblock13d <- genomeblocks(chr13d)
outblock13d
outblock14d <- genomeblocks(chr14d)
outblock14d
outblock15d <- genomeblocks(chr15d)
outblock15d
outblock16d <- genomeblocks(chr16d)
outblock16d
outblock17d <- genomeblocks(chr17d)
outblock17d
outblock18d <- genomeblocks(chr18d)
outblock18d
outblock19d <- genomeblocks(chr19d)
outblock19d
outblock20d <- genomeblocks(chr20d)
outblock20d
outblock21d <- genomeblocks(chr21d)
outblock21d
outblock22d <- genomeblocks(chr22d)
outblock22d
outblock23d <- genomeblocks(chr23d)
outblock23d
outblock24d <- genomeblocks(chr24d)
outblock24d
outblock25d <- genomeblocks(chr25d)
outblock25d
outblock26d <- genomeblocks(chr26d)
outblock26d
outblock27d <- genomeblocks(chr27d)
outblock27d
outblock28d <- genomeblocks(chr28d)
outblock28d
outblock29d <- genomeblocks(chr29d)
outblock29d

#We are using this in the publication
#Using Andre BCR and somewhat similar search
genomeblocksBCR3_1d <- genomeblocksBCR(outblock1d, BCR3_1)
genomeblocksBCR3_1d
genomeblocksBCR3_2d <- genomeblocksBCR(outblock2d, BCR3_2)
genomeblocksBCR3_2d
genomeblocksBCR3_3d <- genomeblocksBCR(outblock3d, BCR3_3)
genomeblocksBCR3_3d
genomeblocksBCR3_4d <- genomeblocksBCR(outblock4d, BCR3_4)
genomeblocksBCR3_4d
genomeblocksBCR3_5d <- genomeblocksBCR(outblock5d, BCR3_5)
genomeblocksBCR3_5d
genomeblocksBCR3_6d <- genomeblocksBCR(outblock6d, BCR3_6)
genomeblocksBCR3_6d
genomeblocksBCR3_7d <- genomeblocksBCR(outblock7d, BCR3_7)
genomeblocksBCR3_7d
genomeblocksBCR3_8d <- genomeblocksBCR(outblock8d, BCR3_8)
genomeblocksBCR3_8d
genomeblocksBCR3_9d <- genomeblocksBCR(outblock9d, BCR3_9)
genomeblocksBCR3_9d
genomeblocksBCR3_10d <- genomeblocksBCR(outblock10d, BCR3_10)
genomeblocksBCR3_10d
genomeblocksBCR3_11d <- genomeblocksBCR(outblock11d, BCR3_11)
genomeblocksBCR3_11d
genomeblocksBCR3_12d <- genomeblocksBCR(outblock12d, BCR3_12)
genomeblocksBCR3_12d
genomeblocksBCR3_13d <- genomeblocksBCR(outblock13d, BCR3_13)
genomeblocksBCR3_13d
genomeblocksBCR3_14d <- genomeblocksBCR(outblock14d, BCR3_14)
genomeblocksBCR3_14d
genomeblocksBCR3_15d <- genomeblocksBCR(outblock15d, BCR3_15)
genomeblocksBCR3_15d
genomeblocksBCR3_16d <- genomeblocksBCR(outblock16d, BCR3_16)
genomeblocksBCR3_16d
genomeblocksBCR3_17d <- genomeblocksBCR(outblock17d, BCR3_17)
genomeblocksBCR3_17d
genomeblocksBCR3_18d <- genomeblocksBCR(outblock18d, BCR3_18)
genomeblocksBCR3_18d
genomeblocksBCR3_19d <- genomeblocksBCR(outblock19d, BCR3_19)
genomeblocksBCR3_19d
genomeblocksBCR3_20d <- genomeblocksBCR(outblock20d, BCR3_20)
genomeblocksBCR3_20d
genomeblocksBCR3_21d <- genomeblocksBCR(outblock21d, BCR3_21)
genomeblocksBCR3_21d
genomeblocksBCR3_22d <- genomeblocksBCR(outblock22d, BCR3_22)
genomeblocksBCR3_22d
genomeblocksBCR3_23d <- genomeblocksBCR(outblock23d, BCR3_23)
genomeblocksBCR3_23d
genomeblocksBCR3_24d <- genomeblocksBCR(outblock24d, BCR3_24)
genomeblocksBCR3_24d
genomeblocksBCR3_25d <- genomeblocksBCR(outblock25d, BCR3_25)
genomeblocksBCR3_25d
genomeblocksBCR3_26d <- genomeblocksBCR(outblock26d, BCR3_26)
genomeblocksBCR3_26d
genomeblocksBCR3_27d <- genomeblocksBCR(outblock27d, BCR3_27)
genomeblocksBCR3_27d
genomeblocksBCR3_28d <- genomeblocksBCR(outblock28d, BCR3_28)
genomeblocksBCR3_28d
genomeblocksBCR3_29d <- genomeblocksBCR(outblock29d, BCR3_29)
genomeblocksBCR3_29d

genomeblocksBCR3_alld <- rbind(genomeblocksBCR3_1d,genomeblocksBCR3_2d,genomeblocksBCR3_3d,genomeblocksBCR3_4d,genomeblocksBCR3_5d,genomeblocksBCR3_6d,genomeblocksBCR3_7d,genomeblocksBCR3_8d,genomeblocksBCR3_9d,genomeblocksBCR3_10d,genomeblocksBCR3_11d,genomeblocksBCR3_12d,genomeblocksBCR3_13d,genomeblocksBCR3_14d,genomeblocksBCR3_15d,genomeblocksBCR3_16d,genomeblocksBCR3_17d,genomeblocksBCR3_18d,genomeblocksBCR3_19d,genomeblocksBCR3_20d,genomeblocksBCR3_21d,genomeblocksBCR3_22d,genomeblocksBCR3_23d,genomeblocksBCR3_24d,genomeblocksBCR3_25d,genomeblocksBCR3_26d,genomeblocksBCR3_27d,genomeblocksBCR3_28d,genomeblocksBCR3_29d)

length(genomeblocksBCR3_alld[,4][genomeblocksBCR3_alld[,4] == "conserved"])
length(genomeblocksBCR3_alld[,4][genomeblocksBCR3_alld[,4] == "fisfus"])
length(genomeblocksBCR3_alld[,4][genomeblocksBCR3_alld[,4] == "singlemarker"])

mean(as.numeric(genomeblocksBCR3_alld[,5][genomeblocksBCR3_alld[,4] == "conserved"]))
mean(as.numeric(genomeblocksBCR3_alld[,5][genomeblocksBCR3_alld[,4] == "fisfus"]))
mean(as.numeric(genomeblocksBCR3_alld[,5][genomeblocksBCR3_alld[,4] == "singlemarker"]))

sd(as.numeric(genomeblocksBCR3_alld[,5][genomeblocksBCR3_alld[,4] == "conserved"]))
sd(as.numeric(genomeblocksBCR3_alld[,5][genomeblocksBCR3_alld[,4] == "fisfus"]))
sd(as.numeric(genomeblocksBCR3_alld[,5][genomeblocksBCR3_alld[,4] == "singlemarker"]))

genomeblocksBCR3_alld
genomeblocksBCR3_alld[genomeblocksBCR3_alld[,4] != "singlemarker", ]



```


Now we will see the patterns with CG content

```{r}

source("genomeblocksGC.R")

#y<-chr2GC
#x<-genomeblocksBCR3_2d
#i <- 3
#genomeblocksGC(x,y)

chr1GC <- read.csv("./GCcontent/chr1GCcontent.csv")
genomeblocksBCR3_1dGC <- genomeblocksGC(genomeblocksBCR3_1d,chr1GC)
genomeblocksBCR3_1dGC

chr2GC <- read.csv("./GCcontent/chr2GCcontent.csv")
genomeblocksBCR3_2dGC <- genomeblocksGC(genomeblocksBCR3_2d,chr2GC)
genomeblocksBCR3_2dGC

chr3GC <- read.csv("./GCcontent/chr3GCcontent.csv")
genomeblocksBCR3_3dGC <- genomeblocksGC(genomeblocksBCR3_3d,chr3GC)
genomeblocksBCR3_3dGC

chr4GC <- read.csv("./GCcontent/chr4GCcontent.csv")
genomeblocksBCR3_4dGC <- genomeblocksGC(genomeblocksBCR3_4d,chr4GC)
genomeblocksBCR3_4dGC

chr5GC <- read.csv("./GCcontent/chr5GCcontent.csv")
genomeblocksBCR3_5dGC <- genomeblocksGC(genomeblocksBCR3_5d,chr5GC)
genomeblocksBCR3_5dGC

chr6GC <- read.csv("./GCcontent/chr6GCcontent.csv")
genomeblocksBCR3_6dGC <- genomeblocksGC(genomeblocksBCR3_6d,chr6GC)
genomeblocksBCR3_6dGC

chr7GC <- read.csv("./GCcontent/chr7GCcontent.csv")
genomeblocksBCR3_7dGC <- genomeblocksGC(genomeblocksBCR3_7d,chr7GC)
genomeblocksBCR3_7dGC

chr8GC <- read.csv("./GCcontent/chr8GCcontent.csv")
genomeblocksBCR3_8dGC <- genomeblocksGC(genomeblocksBCR3_8d,chr8GC)
genomeblocksBCR3_8dGC

chr9GC <- read.csv("./GCcontent/chr9GCcontent.csv")
genomeblocksBCR3_9dGC <- genomeblocksGC(genomeblocksBCR3_9d,chr9GC)
genomeblocksBCR3_9dGC

chr10GC <- read.csv("./GCcontent/chr10GCcontent.csv")
genomeblocksBCR3_10dGC <- genomeblocksGC(genomeblocksBCR3_10d,chr10GC)
genomeblocksBCR3_10dGC

chr11GC <- read.csv("./GCcontent/chr11GCcontent.csv")
genomeblocksBCR3_11dGC <- genomeblocksGC(genomeblocksBCR3_11d,chr11GC)
genomeblocksBCR3_11dGC

chr12GC <- read.csv("./GCcontent/chr12GCcontent.csv")
genomeblocksBCR3_12dGC <- genomeblocksGC(genomeblocksBCR3_12d,chr12GC)
genomeblocksBCR3_12dGC

chr13GC <- read.csv("./GCcontent/chr13GCcontent.csv")
genomeblocksBCR3_13dGC <- genomeblocksGC(genomeblocksBCR3_13d,chr13GC)
genomeblocksBCR3_13dGC

chr14GC <- read.csv("./GCcontent/chr14GCcontent.csv")
genomeblocksBCR3_14dGC <- genomeblocksGC(genomeblocksBCR3_14d,chr14GC)
genomeblocksBCR3_14dGC

chr15GC <- read.csv("./GCcontent/chr15GCcontent.csv")
genomeblocksBCR3_15dGC <- genomeblocksGC(genomeblocksBCR3_15d,chr15GC)
genomeblocksBCR3_15dGC

chr16GC <- read.csv("./GCcontent/chr16GCcontent.csv")
genomeblocksBCR3_16dGC <- genomeblocksGC(genomeblocksBCR3_16d,chr16GC)
genomeblocksBCR3_16dGC

chr17GC <- read.csv("./GCcontent/chr17GCcontent.csv")
genomeblocksBCR3_17dGC <- genomeblocksGC(genomeblocksBCR3_17d,chr17GC)
genomeblocksBCR3_17dGC

chr18GC <- read.csv("./GCcontent/chr18GCcontent.csv")
genomeblocksBCR3_18dGC <- genomeblocksGC(genomeblocksBCR3_18d,chr18GC)
genomeblocksBCR3_18dGC

chr19GC <- read.csv("./GCcontent/chr19GCcontent.csv")
genomeblocksBCR3_19dGC <- genomeblocksGC(genomeblocksBCR3_19d,chr19GC)
genomeblocksBCR3_19dGC

chr20GC <- read.csv("./GCcontent/chr20GCcontent.csv")
genomeblocksBCR3_20dGC <- genomeblocksGC(genomeblocksBCR3_20d,chr20GC)
genomeblocksBCR3_20dGC

chr21GC <- read.csv("./GCcontent/chr21GCcontent.csv")
genomeblocksBCR3_21dGC <- genomeblocksGC(genomeblocksBCR3_21d,chr21GC)
genomeblocksBCR3_21dGC

chr22GC <- read.csv("./GCcontent/chr22GCcontent.csv")
genomeblocksBCR3_22dGC <- genomeblocksGC(genomeblocksBCR3_22d,chr22GC)
genomeblocksBCR3_22dGC

chr23GC <- read.csv("./GCcontent/chr23GCcontent.csv")
genomeblocksBCR3_23dGC <- genomeblocksGC(genomeblocksBCR3_23d,chr23GC)
genomeblocksBCR3_23dGC

chr24GC <- read.csv("./GCcontent/chr24GCcontent.csv")
genomeblocksBCR3_24dGC <- genomeblocksGC(genomeblocksBCR3_24d,chr24GC)
genomeblocksBCR3_24dGC

chr25GC <- read.csv("./GCcontent/chr25GCcontent.csv")
genomeblocksBCR3_25dGC <- genomeblocksGC(genomeblocksBCR3_25d,chr25GC)
genomeblocksBCR3_25dGC

chr26GC <- read.csv("./GCcontent/chr26GCcontent.csv")
genomeblocksBCR3_26dGC <- genomeblocksGC(genomeblocksBCR3_26d,chr26GC)
genomeblocksBCR3_26dGC

chr27GC <- read.csv("./GCcontent/chr27GCcontent.csv")
genomeblocksBCR3_27dGC <- genomeblocksGC(genomeblocksBCR3_27d,chr27GC)
genomeblocksBCR3_27dGC

chr28GC <- read.csv("./GCcontent/chr28GCcontent.csv")
genomeblocksBCR3_28dGC <- genomeblocksGC(genomeblocksBCR3_28d,chr28GC)
genomeblocksBCR3_28dGC

chr29GC <- read.csv("./GCcontent/chr29GCcontent.csv")
genomeblocksBCR3_29dGC <- genomeblocksGC(genomeblocksBCR3_29d,chr29GC)
genomeblocksBCR3_29dGC


chr1GC <- cbind("chr1",chr1GC)
colnames(chr1GC)<- c("chr", "Start..bp.", "Stop..bp.","Length..bp.","GC.content....")
chr2GC <- cbind("chr2",chr2GC) 
colnames(chr2GC)<- c("chr", "Start..bp.", "Stop..bp.","Length..bp.","GC.content....")
chr3GC <- cbind("chr3",chr3GC)
colnames(chr3GC)<- c("chr", "Start..bp.", "Stop..bp.","Length..bp.","GC.content....")
chr4GC <- cbind("chr4",chr4GC)
colnames(chr4GC)<- c("chr", "Start..bp.", "Stop..bp.","Length..bp.","GC.content....")
chr5GC <- cbind("chr5",chr5GC)
colnames(chr5GC)<- c("chr", "Start..bp.", "Stop..bp.","Length..bp.","GC.content....")
chr6GC <- cbind("chr6",chr6GC)
colnames(chr6GC)<- c("chr", "Start..bp.", "Stop..bp.","Length..bp.","GC.content....")
chr7GC <- cbind("chr7",chr7GC)
colnames(chr7GC)<- c("chr", "Start..bp.", "Stop..bp.","Length..bp.","GC.content....")
chr8GC <- cbind("chr8",chr8GC)
colnames(chr8GC)<- c("chr", "Start..bp.", "Stop..bp.","Length..bp.","GC.content....")
chr9GC <- cbind("chr9",chr9GC)
colnames(chr9GC)<- c("chr", "Start..bp.", "Stop..bp.","Length..bp.","GC.content....")
chr10GC <- cbind("chr10",chr10GC)
colnames(chr10GC)<- c("chr", "Start..bp.", "Stop..bp.","Length..bp.","GC.content....")
chr11GC <- cbind("chr11",chr11GC)
colnames(chr11GC)<- c("chr", "Start..bp.", "Stop..bp.","Length..bp.","GC.content....")
chr12GC <- cbind("chr12",chr12GC)
colnames(chr12GC)<- c("chr", "Start..bp.", "Stop..bp.","Length..bp.","GC.content....")
chr13GC <- cbind("chr13",chr13GC)
colnames(chr13GC)<- c("chr", "Start..bp.", "Stop..bp.","Length..bp.","GC.content....")
chr14GC <- cbind("chr14",chr14GC)
colnames(chr14GC)<- c("chr", "Start..bp.", "Stop..bp.","Length..bp.","GC.content....")
chr15GC <- cbind("chr15",chr15GC)
colnames(chr15GC)<- c("chr", "Start..bp.", "Stop..bp.","Length..bp.","GC.content....")
chr16GC <- cbind("chr16",chr16GC)
colnames(chr16GC)<- c("chr", "Start..bp.", "Stop..bp.","Length..bp.","GC.content....")
chr17GC <- cbind("chr17",chr17GC)
colnames(chr17GC)<- c("chr", "Start..bp.", "Stop..bp.","Length..bp.","GC.content....")
chr18GC <- cbind("chr18",chr18GC)
colnames(chr18GC)<- c("chr", "Start..bp.", "Stop..bp.","Length..bp.","GC.content....")
chr19GC <- cbind("chr19",chr19GC)
colnames(chr19GC)<- c("chr", "Start..bp.", "Stop..bp.","Length..bp.","GC.content....")
chr20GC <- cbind("chr20",chr20GC)
colnames(chr20GC)<- c("chr", "Start..bp.", "Stop..bp.","Length..bp.","GC.content....")
chr21GC <- cbind("chr21",chr21GC)
colnames(chr21GC)<- c("chr", "Start..bp.", "Stop..bp.","Length..bp.","GC.content....")
chr22GC <- cbind("chr22",chr22GC)
colnames(chr22GC)<- c("chr", "Start..bp.", "Stop..bp.","Length..bp.","GC.content....")
chr23GC <- cbind("chr23",chr23GC)
colnames(chr23GC)<- c("chr", "Start..bp.", "Stop..bp.","Length..bp.","GC.content....")
chr24GC <- cbind("chr24",chr24GC)
colnames(chr24GC)<- c("chr", "Start..bp.", "Stop..bp.","Length..bp.","GC.content....")
chr25GC <- cbind("chr25",chr25GC)
colnames(chr25GC)<- c("chr", "Start..bp.", "Stop..bp.","Length..bp.","GC.content....")
chr26GC <- cbind("chr26",chr26GC)
colnames(chr26GC)<- c("chr", "Start..bp.", "Stop..bp.","Length..bp.","GC.content....")
chr27GC <- cbind("chr27",chr27GC)
colnames(chr27GC)<- c("chr", "Start..bp.", "Stop..bp.","Length..bp.","GC.content....")
chr28GC <- cbind("chr28",chr28GC)
colnames(chr28GC)<- c("chr", "Start..bp.", "Stop..bp.","Length..bp.","GC.content....")
chr29GC <- cbind("chr29",chr29GC) 
colnames(chr29GC)<- c("chr", "Start..bp.", "Stop..bp.","Length..bp.","GC.content....")

chrGCall <- rbind(chr1GC, chr2GC, chr3GC, chr4GC, chr5GC, chr6GC, chr7GC, chr8GC, chr9GC, chr10GC, chr11GC, chr12GC, chr13GC, chr14GC, chr15GC, chr16GC, chr17GC, chr18GC, chr19GC, chr20GC, chr21GC, chr22GC, chr23GC, chr24GC, chr25GC, chr26GC, chr27GC, chr28GC, chr29GC)

write.csv(chrGCall, file = "chrGCall.csv")

```

Now we are going to check whats going on with genes

```{r}

#Using the genome with duplicates

Genes <- read.csv("Kobresia_littledalei_genes.bed", sep = "\t")
Genes 

Genes <- Genes[,1:3]
Genes[Genes[,1]=="CM022079",1] <- "chr1"
Genes[Genes[,1]=="CM022080",1] <- "chr2"
Genes[Genes[,1]=="CM022081",1] <- "chr3"
Genes[Genes[,1]=="CM022082",1] <- "chr4"
Genes[Genes[,1]=="CM022083",1] <- "chr5"
Genes[Genes[,1]=="CM022084",1] <- "chr6"
Genes[Genes[,1]=="CM022085",1] <- "chr7"
Genes[Genes[,1]=="CM022086",1] <- "chr8"
Genes[Genes[,1]=="CM022087",1] <- "chr9"
Genes[Genes[,1]=="CM022088",1] <- "chr10"
Genes[Genes[,1]=="CM022089",1] <- "chr11"
Genes[Genes[,1]=="CM022090",1] <- "chr12"
Genes[Genes[,1]=="CM022091",1] <- "chr13"
Genes[Genes[,1]=="CM022092",1] <- "chr14"
Genes[Genes[,1]=="CM022093",1] <- "chr15"
Genes[Genes[,1]=="CM022094",1] <- "chr16"
Genes[Genes[,1]=="CM022095",1] <- "chr17"
Genes[Genes[,1]=="CM022096",1] <- "chr18"
Genes[Genes[,1]=="CM022097",1] <- "chr19"
Genes[Genes[,1]=="CM022098",1] <- "chr20"
Genes[Genes[,1]=="CM022099",1] <- "chr21"
Genes[Genes[,1]=="CM022100",1] <- "chr22"
Genes[Genes[,1]=="CM022101",1] <- "chr23"
Genes[Genes[,1]=="CM022102",1] <- "chr24"
Genes[Genes[,1]=="CM022103",1] <- "chr25"
Genes[Genes[,1]=="CM022104",1] <- "chr26"
Genes[Genes[,1]=="CM022105",1] <- "chr27"
Genes[Genes[,1]=="CM022106",1] <- "chr28"
Genes[Genes[,1]=="CM022107",1] <- "chr29"
colnames(Genes) <- c("Chromosome", "Start", "End")


Genes1 <- Genes[Genes$Chromosome=="chr1",]
Genes1
Genes2 <- Genes[Genes$Chromosome=="chr2",]
Genes2
Genes3 <- Genes[Genes$Chromosome=="chr3",]
Genes3
Genes4 <- Genes[Genes$Chromosome=="chr4",]
Genes4
Genes5 <- Genes[Genes$Chromosome=="chr5",]
Genes5
Genes6 <- Genes[Genes$Chromosome=="chr6",]
Genes6
Genes7 <- Genes[Genes$Chromosome=="chr7",]
Genes7
Genes8 <- Genes[Genes$Chromosome=="chr8",]
Genes8
Genes9 <- Genes[Genes$Chromosome=="chr9",]
Genes9
Genes10 <- Genes[Genes$Chromosome=="chr10",]
Genes10
Genes11 <- Genes[Genes$Chromosome=="chr11",]
Genes11
Genes12 <- Genes[Genes$Chromosome=="chr12",]
Genes12
Genes13 <- Genes[Genes$Chromosome=="chr13",]
Genes13
Genes14 <- Genes[Genes$Chromosome=="chr14",]
Genes14
Genes15 <- Genes[Genes$Chromosome=="chr15",]
Genes15
Genes16 <- Genes[Genes$Chromosome=="chr16",]
Genes16
Genes17 <- Genes[Genes$Chromosome=="chr17",]
Genes17
Genes18 <- Genes[Genes$Chromosome=="chr18",]
Genes18
Genes19 <- Genes[Genes$Chromosome=="chr19",]
Genes19
Genes20 <- Genes[Genes$Chromosome=="chr20",]
Genes20
Genes21 <- Genes[Genes$Chromosome=="chr21",]
Genes21
Genes22 <- Genes[Genes$Chromosome=="chr22",]
Genes22
Genes23 <- Genes[Genes$Chromosome=="chr23",]
Genes23
Genes24 <- Genes[Genes$Chromosome=="chr24",]
Genes24
Genes25 <- Genes[Genes$Chromosome=="chr25",]
Genes25
Genes26 <- Genes[Genes$Chromosome=="chr26",]
Genes26
Genes27 <- Genes[Genes$Chromosome=="chr27",]
Genes27
Genes28 <- Genes[Genes$Chromosome=="chr28",]
Genes28
Genes29 <- Genes[Genes$Chromosome=="chr29",]
Genes29


#We can use the same function than with the BCR, something very similar
source("genomeblocksGenes.R")
genomeblocksGenes1d <- genomeblocksGenes(genomeblocksBCR3_1dGC, Genes1)
genomeblocksGenes1d
genomeblocksGenes2d <- genomeblocksGenes(genomeblocksBCR3_2dGC, Genes2)
genomeblocksGenes2d
genomeblocksGenes3d <- genomeblocksGenes(genomeblocksBCR3_3dGC, Genes3)
genomeblocksGenes3d
genomeblocksGenes4d <- genomeblocksGenes(genomeblocksBCR3_4dGC, Genes4)
genomeblocksGenes4d
genomeblocksGenes5d <- genomeblocksGenes(genomeblocksBCR3_5dGC, Genes5)
genomeblocksGenes5d
genomeblocksGenes6d <- genomeblocksGenes(genomeblocksBCR3_6dGC, Genes6)
genomeblocksGenes6d
genomeblocksGenes7d <- genomeblocksGenes(genomeblocksBCR3_7dGC, Genes7)
genomeblocksGenes7d
genomeblocksGenes8d <- genomeblocksGenes(genomeblocksBCR3_8dGC, Genes8)
genomeblocksGenes8d
genomeblocksGenes9d <- genomeblocksGenes(genomeblocksBCR3_9dGC, Genes9)
genomeblocksGenes9d
genomeblocksGenes10d <- genomeblocksGenes(genomeblocksBCR3_10dGC, Genes10)
genomeblocksGenes10d
genomeblocksGenes11d <- genomeblocksGenes(genomeblocksBCR3_11dGC, Genes11)
genomeblocksGenes11d
genomeblocksGenes12d <- genomeblocksGenes(genomeblocksBCR3_12dGC, Genes12)
genomeblocksGenes12d
genomeblocksGenes13d <- genomeblocksGenes(genomeblocksBCR3_13dGC, Genes13)
genomeblocksGenes13d
genomeblocksGenes14d <- genomeblocksGenes(genomeblocksBCR3_14dGC, Genes14)
genomeblocksGenes14d
genomeblocksGenes15d <- genomeblocksGenes(genomeblocksBCR3_15dGC, Genes15)
genomeblocksGenes15d
genomeblocksGenes16d <- genomeblocksGenes(genomeblocksBCR3_16dGC, Genes16)
genomeblocksGenes16d
genomeblocksGenes17d <- genomeblocksGenes(genomeblocksBCR3_17dGC, Genes17)
genomeblocksGenes17d
genomeblocksGenes18d <- genomeblocksGenes(genomeblocksBCR3_18dGC, Genes18)
genomeblocksGenes18d
genomeblocksGenes19d <- genomeblocksGenes(genomeblocksBCR3_19dGC, Genes19)
genomeblocksGenes19d
genomeblocksGenes20d <- genomeblocksGenes(genomeblocksBCR3_20dGC, Genes20)
genomeblocksGenes20d
genomeblocksGenes21d <- genomeblocksGenes(genomeblocksBCR3_21dGC, Genes21)
genomeblocksGenes21d
genomeblocksGenes22d <- genomeblocksGenes(genomeblocksBCR3_22dGC, Genes22)
genomeblocksGenes22d
genomeblocksGenes23d <- genomeblocksGenes(genomeblocksBCR3_23dGC, Genes23)
genomeblocksGenes23d
genomeblocksGenes24d <- genomeblocksGenes(genomeblocksBCR3_24dGC, Genes24)
genomeblocksGenes24d
genomeblocksGenes25d <- genomeblocksGenes(genomeblocksBCR3_25dGC, Genes25)
genomeblocksGenes25d
genomeblocksGenes26d <- genomeblocksGenes(genomeblocksBCR3_26dGC, Genes26)
genomeblocksGenes26d
genomeblocksGenes27d <- genomeblocksGenes(genomeblocksBCR3_27dGC, Genes27)
genomeblocksGenes27d
genomeblocksGenes28d <- genomeblocksGenes(genomeblocksBCR3_28dGC, Genes28)
genomeblocksGenes28d
genomeblocksGenes29d <- genomeblocksGenes(genomeblocksBCR3_29dGC, Genes29)
genomeblocksGenes29d

genomeblocksGenesalld <- rbind(cbind(1,genomeblocksGenes1d),cbind(2,genomeblocksGenes2d),cbind(3,genomeblocksGenes3d),cbind(4,genomeblocksGenes4d),cbind(5,genomeblocksGenes5d),cbind(6,genomeblocksGenes6d),cbind(7,genomeblocksGenes7d),cbind(8,genomeblocksGenes8d),cbind(9,genomeblocksGenes9d),cbind(10,genomeblocksGenes10d),cbind(11,genomeblocksGenes11d),cbind(12,genomeblocksGenes12d),cbind(13,genomeblocksGenes13d),cbind(14,genomeblocksGenes14d),cbind(15,genomeblocksGenes15d),cbind(16,genomeblocksGenes16d),cbind(17,genomeblocksGenes17d),cbind(18,genomeblocksGenes18d),cbind(9,genomeblocksGenes19d),cbind(20,genomeblocksGenes20d),cbind(21,genomeblocksGenes21d),cbind(22,genomeblocksGenes22d),cbind(23,genomeblocksGenes23d),cbind(24,genomeblocksGenes24d),cbind(25,genomeblocksGenes25d),cbind(26,genomeblocksGenes26d),cbind(27,genomeblocksGenes27d),cbind(28,genomeblocksGenes28d),cbind(29,genomeblocksGenes29d))

length(genomeblocksGenesalld[,8][genomeblocksGenesalld[,5] == "conserved"])
length(genomeblocksGenesalld[,8][genomeblocksGenesalld[,5] == "fisfus"])
length(genomeblocksGenesalld[,8][genomeblocksGenesalld[,5] == "singlemarker"])

mean(as.numeric(genomeblocksGenesalld[,8][genomeblocksGenesalld[,5] == "conserved"]))
mean(as.numeric(genomeblocksGenesalld[,8][genomeblocksGenesalld[,5] == "fisfus"]))
mean(as.numeric(genomeblocksGenesalld[,8][genomeblocksGenesalld[,5] == "singlemarker"]))

sd(as.numeric(genomeblocksGenesalld[,8][genomeblocksGenesalld[,5] == "conserved"]))
sd(as.numeric(genomeblocksGenesalld[,8][genomeblocksGenesalld[,5] == "fisfus"]))
sd(as.numeric(genomeblocksGenesalld[,8][genomeblocksGenesalld[,5] == "singlemarker"]))


genomeblocksGenesalld
genomeblocksGenesalld
genomeblocksGenesalld[genomeblocksGenesalld[,5] != "singlemarker", ]

glm_genomeblocks_alld <- genomeblocksGenesalld[genomeblocksGenesalld[,5] != "singlemarker", -c(2:4)]
glm_genomeblocks_alld[,2][glm_genomeblocks_alld[,2]=="conserved"] <- 0
glm_genomeblocks_alld[,2][glm_genomeblocks_alld[,2]=="fisfus"] <- 1

library(lme4)
glm_genomeblocks_alld<- data.frame (glm_genomeblocks_alld)
colnames(glm_genomeblocks_alld) <- c("chromosome", "type", "BCR",  "GC", "Genes")
fittedModel <- glmer (as.numeric(type) ~ as.numeric(BCR) + as.numeric(GC) + as.numeric(Genes) + (1|chromosome), data = glm_genomeblocks_alld, family = binomial(link = "logit"))
summary(fittedModel)

fittedModel2 <- glm (as.numeric(type) ~ as.numeric(BCR) + as.numeric(GC) + as.numeric(Genes), family = binomial(link = "logit"), data = glm_genomeblocks_alld)
summary(fittedModel2)

fittedModel3 <- glm (as.numeric(type) ~ as.numeric(BCR) + as.numeric(GC) + as.numeric(Genes), family = gaussian(link = "identity"), data = glm_genomeblocks_alld)
summary(fittedModel3)

fittedModel4 <- glm (as.numeric(type) ~ as.numeric(BCR) + as.numeric(GC) + as.numeric(Genes), family = poisson(link = "log"), data = glm_genomeblocks_alld)
summary(fittedModel4)

library(DHARMa)

simulationOutput <- simulateResiduals(fittedModel = fittedModel, plot = F)
plot(simulationOutput)

simulationOutput2 <- simulateResiduals(fittedModel = fittedModel2, plot = F)
plot(simulationOutput2)

simulationOutput3 <- simulateResiduals(fittedModel = fittedModel3, plot = F)
plot(simulationOutput3)

simulationOutput4 <- simulateResiduals(fittedModel = fittedModel4, plot = F)
plot(simulationOutput4)

```


Simulating the rearrangements!
We have done here some proofs to set up the definitive analyses.
Note: This is not included in the paper.

```{r}

source("fisfusmodeling.R") 

genome = cbind(c("LG1", "LG1","LG1", "LG1", "LG1", "LG2", "LG2", "LG2", "LG2", "LG2", "LG3", "LG3", "LG3", "LG3", "LG3"),c(0,1000000, 3000000, 9000000, 10000000, 0,1000000, 3000000, 9000000, 10000000, 0,1000000, 3000000, 9000000, 10000000), c(1,1000001, 3000001, 9000001, 10000001, 1,1000001, 3000001, 9000001, 10000001, 0,1000001, 3000001, 9000001, 10000001), c(NA, "M1","M2","M3",NA,NA,"M4","M5","M6",NA,NA,"M7","M8","M9",NA ))
colnames(genome) <- c("Chromosome","chromStart","chromEnd","Marker")
genome <- as.data.frame(genome)
genome

rearrangements = 10
pfis = 0.5
fisfusmodelling(genome, rearrangements,pfis)

# Carex scoparia and Carex littledalei got separated ca. 23 mya 
# rate of fission based on ChromoSSE 3.4 fissions per million year and 5 fussion per million year in anagenetic
# rate of fission based on ChromoSSE 0.6 fissions per million year and 0.4 fussion per million year in anagenetic
# number of rearrangements are ca. 9.5 events per million year x 46 my (23 + 23 million years)
# pfis = 4/9.4

rearrangements = 435
pfis = 0.5

genome <- rbind(
c("chr1",0,1,NA),  
bowtie2[bowtie2[,1]=="chr1",][order(bowtie2[bowtie2[,1]=="chr1",2]),], 
c(both[1,1],both[1,3]-1,both[1,3],NA), 
c("chr2",0,1,NA),  
bowtie2[bowtie2[,1]=="chr2",][order(bowtie2[bowtie2[,1]=="chr2",2]),], 
c(both[2,1],both[2,3]-1,both[2,3],NA),
c("chr3",0,1,NA),  
bowtie2[bowtie2[,1]=="chr3",][order(bowtie2[bowtie2[,1]=="chr3",2]),], 
c(both[3,1],both[3,3]-1,both[3,3],NA),
c("chr4",0,1,NA),  
bowtie2[bowtie2[,1]=="chr4",][order(bowtie2[bowtie2[,1]=="chr4",2]),], 
c(both[4,1],both[4,3]-1,both[4,3],NA),
c("chr5",0,1,NA),  
bowtie2[bowtie2[,1]=="chr5",][order(bowtie2[bowtie2[,1]=="chr5",2]),], 
c(both[5,1],both[5,3]-1,both[5,3],NA),
c("chr6",0,1,NA),  
bowtie2[bowtie2[,1]=="chr6",][order(bowtie2[bowtie2[,1]=="chr6",2]),], 
c(both[6,1],both[6,3]-1,both[6,3],NA),
c("chr7",0,1,NA),  
bowtie2[bowtie2[,1]=="chr7",][order(bowtie2[bowtie2[,1]=="chr7",2]),], 
c(both[7,1],both[7,3]-1,both[7,3],NA),
c("chr8",0,1,NA),  
bowtie2[bowtie2[,1]=="chr8",][order(bowtie2[bowtie2[,1]=="chr8",2]),], 
c(both[8,1],both[8,3]-1,both[8,3],NA),
c("chr9",0,1,NA),  
bowtie2[bowtie2[,1]=="chr9",][order(bowtie2[bowtie2[,1]=="chr9",2]),], 
c(both[9,1],both[9,3]-1,both[9,3],NA),
c("chr10",0,1,NA),  
bowtie2[bowtie2[,1]=="chr10",][order(bowtie2[bowtie2[,1]=="chr10",2]),], 
c(both[10,1],both[10,3]-1,both[10,3],NA),
c("chr11",0,1,NA),  
bowtie2[bowtie2[,1]=="chr11",][order(bowtie2[bowtie2[,1]=="chr11",2]),], 
c(both[11,1],both[11,3]-1,both[11,3],NA),
c("chr12",0,1,NA),  
bowtie2[bowtie2[,1]=="chr12",][order(bowtie2[bowtie2[,1]=="chr12",2]),], 
c(both[12,1],both[12,3]-1,both[12,3],NA),
c("chr13",0,1,NA),  
bowtie2[bowtie2[,1]=="chr13",][order(bowtie2[bowtie2[,1]=="chr13",2]),], 
c(both[13,1],both[13,3]-1,both[13,3],NA),
c("chr14",0,1,NA),  
bowtie2[bowtie2[,1]=="chr14",][order(bowtie2[bowtie2[,1]=="chr14",2]),], 
c(both[14,1],both[14,3]-1,both[14,3],NA),
c("chr15",0,1,NA),  
bowtie2[bowtie2[,1]=="chr15",][order(bowtie2[bowtie2[,1]=="chr15",2]),], 
c(both[15,1],both[15,3]-1,both[15,3],NA),
c("chr16",0,1,NA),  
bowtie2[bowtie2[,1]=="chr16",][order(bowtie2[bowtie2[,1]=="chr16",2]),], 
c(both[16,1],both[16,3]-1,both[16,3],NA),
c("chr17",0,1,NA),  
bowtie2[bowtie2[,1]=="chr17",][order(bowtie2[bowtie2[,1]=="chr17",2]),], 
c(both[17,1],both[17,3]-1,both[17,3],NA),
c("chr18",0,1,NA),  
bowtie2[bowtie2[,1]=="chr18",][order(bowtie2[bowtie2[,1]=="chr18",2]),], 
c(both[18,1],both[18,3]-1,both[18,3],NA),
c("chr19",0,1,NA),  
bowtie2[bowtie2[,1]=="chr19",][order(bowtie2[bowtie2[,1]=="chr19",2]),], 
c(both[19,1],both[19,3]-1,both[19,3],NA),
c("chr20",0,1,NA),  
bowtie2[bowtie2[,1]=="chr20",][order(bowtie2[bowtie2[,1]=="chr20",2]),], 
c(both[20,1],both[20,3]-1,both[20,3],NA),
c("chr21",0,1,NA),  
bowtie2[bowtie2[,1]=="chr21",][order(bowtie2[bowtie2[,1]=="chr21",2]),], 
c(both[21,1],both[21,3]-1,both[21,3],NA),
c("chr22",0,1,NA),  
bowtie2[bowtie2[,1]=="chr22",][order(bowtie2[bowtie2[,1]=="chr22",2]),], 
c(both[22,1],both[22,3]-1,both[22,3],NA),
c("chr23",0,1,NA),  
bowtie2[bowtie2[,1]=="chr23",][order(bowtie2[bowtie2[,1]=="chr23",2]),], 
c(both[23,1],both[23,3]-1,both[23,3],NA),
c("chr24",0,1,NA),  
bowtie2[bowtie2[,1]=="chr24",][order(bowtie2[bowtie2[,1]=="chr24",2]),], 
c(both[24,1],both[24,3]-1,both[24,3],NA),
c("chr25",0,1,NA),  
bowtie2[bowtie2[,1]=="chr25",][order(bowtie2[bowtie2[,1]=="chr25",2]),], 
c(both[25,1],both[25,3]-1,both[25,3],NA),
c("chr26",0,1,NA),  
bowtie2[bowtie2[,1]=="chr26",][order(bowtie2[bowtie2[,1]=="chr26",2]),], 
c(both[26,1],both[26,3]-1,both[26,3],NA),
c("chr27",0,1,NA),  
bowtie2[bowtie2[,1]=="chr27",][order(bowtie2[bowtie2[,1]=="chr27",2]),], 
c(both[27,1],both[27,3]-1,both[27,3],NA),
c("chr28",0,1,NA),  
bowtie2[bowtie2[,1]=="chr28",][order(bowtie2[bowtie2[,1]=="chr28",2]),], 
c(both[28,1],both[28,3]-1,both[28,3],NA),
c("chr29",0,1,NA),  
bowtie2[bowtie2[,1]=="chr29",][order(bowtie2[bowtie2[,1]=="chr29",2]),], 
c(both[29,1],both[29,3]-1,both[29,3],NA)
)

littledaleimodel <- fisfusmodelling(genome, rearrangements,pfis)

littledaleimodel436 <- littledaleimodel[[436]]

littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[1],1] <- "LG1"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[2],1] <- "LG2"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[3],1] <- "LG3"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[4],1] <- "LG4"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[5],1] <- "LG5"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[6],1] <- "LG6"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[7],1] <- "LG7"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[8],1] <- "LG8"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[9],1] <- "LG9"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[10],1] <- "LG10"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[11],1] <- "LG11"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[12],1] <- "LG12"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[13],1] <- "LG13"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[14],1] <- "LG14"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[15],1] <- "LG15"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[16],1] <- "LG16"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[17],1] <- "LG17"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[18],1] <- "LG18"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[19],1] <- "LG19"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[20],1] <- "LG20"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[21],1] <- "LG21"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[22],1] <- "LG22"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[23],1] <- "LG23"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[24],1] <- "LG24"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[25],1] <- "LG25"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[26],1] <- "LG26"
#littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[27],1] <- "LG27"
#littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[28],1] <- "LG28"
#littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[29],1] <- "LG29"
#littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[30],1] <- "LG30"
#littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[31],1] <- "LG31"
#littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[32],1] <- "LG32"
#littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[33],1] <- "LG33"
#littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[34],1] <- "LG34"
#littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[35],1] <- "LG35"
#littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[36],1] <- "LG36"
#littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[37],1] <- "LG37"
#littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[38],1] <- "LG38"
#littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[39],1] <- "LG39"
#littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[40],1] <- "LG40"



both[1:29,]
maximum <- vector()
for (i in 1:length(unique(littledaleimodel436[,1])))
maximum[i] <- max(littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[i],3])  
littledaleimodel436genome <- cbind(unique(littledaleimodel436[,1]),0,maximum,NA,NA)
colnames(littledaleimodel436genome) <- colnames(both[1:5])
genomesim <- rbind(both[1:29,1:5],littledaleimodel436genome) 

genomesim <- as.data.frame(genomesim)

library(RCircos)
chr.exclude <- NULL
cyto.info <- genomesim
tracks.inside <- 0
tracks.outside <- 0
RCircos.Set.Core.Components(cyto.info, chr.exclude,tracks.inside, tracks.outside)


rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.cyto <- RCircos.Get.Plot.Ideogram()
rcircos.position <- RCircos.Get.Plot.Positions()
RCircos.List.Plot.Parameters()

 


RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()

genome <- genome[complete.cases(genome),]
littledaleimodel436 <- littledaleimodel436[complete.cases(littledaleimodel436),]

linkssim <- merge(genome, littledaleimodel436, by = "Marker", sort = F, no.dups = TRUE)
length(genome[,4])
length(unique(genome[,4]))
dim(linkssim)
linkssim <- unique(linkssim)
linkssim <- linkssim[,c(2,3,4,5,6,7)]
linkssim
linkssim <- as.data.frame(linkssim)
RCircos.Link.Plot(linkssim,track.num=1,TRUE)
#There is an error here. Maybe consider to use other package

chr1dsim <- linkssim[linkssim$Chromosome.x=="chr1",]
chr1dsim
chr2dsim <- linkssim[linkssim$Chromosome.x=="chr2",]
chr2dsim
chr3dsim <- linkssim[linkssim$Chromosome.x=="chr3",]
chr3dsim
chr4dsim <- linkssim[linkssim$Chromosome.x=="chr4",]
chr4dsim
chr5dsim <- linkssim[linkssim$Chromosome.x=="chr5",]
chr5dsim
chr6dsim <- linkssim[linkssim$Chromosome.x=="chr6",]
chr6dsim
chr7dsim <- linkssim[linkssim$Chromosome.x=="chr7",]
chr7dsim
chr8dsim <- linkssim[linkssim$Chromosome.x=="chr8",]
chr8dsim
chr9dsim <- linkssim[linkssim$Chromosome.x=="chr9",]
chr9dsim
chr10dsim <- linkssim[linkssim$Chromosome.x=="chr10",]
chr10dsim
chr11dsim <- linkssim[linkssim$Chromosome.x=="chr11",]
chr11dsim
chr12dsim <- linkssim[linkssim$Chromosome.x=="chr12",]
chr12dsim
chr13dsim <- linkssim[linkssim$Chromosome.x=="chr13",]
chr13dsim
chr14dsim <- linkssim[linkssim$Chromosome.x=="chr14",]
chr14dsim
chr15dsim <- linkssim[linkssim$Chromosome.x=="chr15",]
chr15dsim
chr16dsim <- linkssim[linkssim$Chromosome.x=="chr16",]
chr16dsim
chr17dsim <- linkssim[linkssim$Chromosome.x=="chr17",]
chr17dsim
chr18dsim <- linkssim[linkssim$Chromosome.x=="chr18",]
chr18dsim
chr19dsim <- linkssim[linkssim$Chromosome.x=="chr19",]
chr19dsim
chr20dsim <- linkssim[linkssim$Chromosome.x=="chr20",]
chr20dsim
chr21dsim <- linkssim[linkssim$Chromosome.x=="chr21",]
chr21dsim
chr22dsim <- linkssim[linkssim$Chromosome.x=="chr22",]
chr22dsim
chr23dsim <- linkssim[linkssim$Chromosome.x=="chr23",]
chr23dsim
chr24dsim <- linkssim[linkssim$Chromosome.x=="chr24",]
chr24dsim
chr25dsim <- linkssim[linkssim$Chromosome.x=="chr25",]
chr25dsim
chr26dsim <- linkssim[linkssim$Chromosome.x=="chr26",]
chr26dsim
chr27dsim <- linkssim[linkssim$Chromosome.x=="chr27",]
chr27dsim
chr28dsim <- linkssim[linkssim$Chromosome.x=="chr28",]
chr28dsim
chr29dsim <- linkssim[linkssim$Chromosome.x=="chr29",]
chr29dsim

source("genomeblocks.R")
outblock1dsim <- genomeblocks(chr1dsim)
outblock1dsim
outblock2dsim <- genomeblocks(chr2dsim)
outblock2dsim
outblock3dsim <- genomeblocks(chr3dsim)
outblock3dsim
outblock4dsim <- genomeblocks(chr4dsim)
outblock4dsim
outblock5dsim <- genomeblocks(chr5dsim)
outblock5dsim
outblock6dsim <- genomeblocks(chr6dsim)
outblock6dsim
outblock7dsim <- genomeblocks(chr7dsim)
outblock7dsim
outblock8dsim <- genomeblocks(chr8dsim)
outblock8dsim
outblock9dsim <- genomeblocks(chr9dsim)
outblock9dsim
outblock10dsim <- genomeblocks(chr10dsim)
outblock10dsim
outblock11dsim <- genomeblocks(chr11dsim)
outblock11dsim
outblock12dsim <- genomeblocks(chr12dsim)
outblock12dsim
outblock13dsim <- genomeblocks(chr13dsim)
outblock13dsim
outblock14dsim <- genomeblocks(chr14dsim)
outblock14dsim
outblock15dsim <- genomeblocks(chr15dsim)
outblock15dsim
outblock16dsim <- genomeblocks(chr16dsim)
outblock16dsim
outblock17dsim <- genomeblocks(chr17dsim)
outblock17dsim
outblock18dsim <- genomeblocks(chr18dsim)
outblock18dsim
outblock19dsim <- genomeblocks(chr19dsim)
outblock19dsim
outblock20dsim <- genomeblocks(chr20dsim)
outblock20dsim
outblock21dsim <- genomeblocks(chr21dsim)
outblock21dsim
outblock22dsim <- genomeblocks(chr22dsim)
outblock22dsim
outblock23dsim <- genomeblocks(chr23dsim)
outblock23dsim
outblock24dsim <- genomeblocks(chr24dsim)
outblock24dsim
outblock25dsim <- genomeblocks(chr25dsim)
outblock25dsim
outblock26dsim <- genomeblocks(chr26dsim)
outblock26dsim
outblock27dsim <- genomeblocks(chr27dsim)
outblock27dsim
outblock28dsim <- genomeblocks(chr28dsim)
outblock28dsim
outblock29dsim <- genomeblocks(chr29dsim)
outblock29dsim

outblockdsim <- rbind(outblock1dsim,outblock2dsim,outblock3dsim,outblock4dsim,outblock5dsim,outblock6dsim,outblock7dsim,outblock8dsim,outblock9dsim,outblock10dsim,outblock11dsim,outblock12dsim,outblock13dsim,outblock14dsim,outblock15dsim,outblock16dsim,outblock17dsim,outblock18dsim,outblock19dsim,outblock20dsim,outblock21dsim,outblock22dsim,outblock23dsim,outblock24dsim,outblock25dsim,outblock26dsim,outblock27dsim,outblock28dsim,outblock29dsim)


dim(outblockdsim[outblockdsim[,4]=="conserved",])
dim(outblockdsim[outblockdsim[,4]=="singlemarker",])
dim(outblockdsim[outblockdsim[,4]=="fisfus",])
sum(as.numeric(outblockdsim[outblockdsim[,4]=="conserved",3]) - as.numeric(outblockdsim[outblockdsim[,4]=="conserved",2]))
mean(as.numeric(outblockdsim[outblockdsim[,4]=="conserved",3]) - as.numeric(outblockdsim[outblockdsim[,4]=="conserved",2]))

#Datos reales
outblockd <- rbind(outblock1d,outblock2d,outblock3d,outblock4d,outblock5d,outblock6d,outblock7d,outblock8d,outblock9d,outblock10d,outblock11d,outblock12d,outblock13d,outblock14d,outblock15d,outblock16d,outblock17d,outblock18d,outblock19d,outblock20d,outblock21d,outblock22d,outblock23d,outblock24d,outblock25d,outblock26d,outblock27d,outblock28d,outblock29d)

dim(outblockd[outblockd[,4]=="conserved",])
dim(outblockd[outblockd[,4]=="singlemarker",])
dim(outblockd[outblockd[,4]=="fisfus",])
sum(as.numeric(outblockd[outblockd[,4]=="conserved",3]) - as.numeric(outblockd[outblockd[,4]=="conserved",2]))
mean(as.numeric(outblockd[outblockd[,4]=="conserved",3]) - as.numeric(outblockd[outblockd[,4]=="conserved",2]))


```

Now with permutations
This is included in the paper
```{r}
source("fisfusmodelingperm.R")
# Carex scoparia and Carex littledalei got separated ca. 23 mya 
#Cladogenetic
#Fission: mean = 0.567, 95% CI = 0.232 - 0.892
#Fusion: mean = 0.475, 95% CI = 0.17 - 0.81
#
#Anagenetic
#Fission: mean = 6.381, 95% CI = 5.144 - 7.597
#Fusion: mean = 6.66, 95% CI = 5.627 - 7.75
# number of rearrangements are ca. 14 events per million year x 46 my (23 + 23 million years)
# pfis = 6.948/14.083

rearrangements = 644
pfis = 0.5
perm = 99
#make sure the genome object is right
source("genome.R")

littledaleimodelperm <- fisfusmodellingperm(genome, rearrangements,pfis,perm)
#littledaleimodelperm <- littledaleimodelpermcopy
#littledaleimodelpermcopy <- littledaleimodelperm

littledaleimodelperm[[1]]
littledaleimodelperm[[1000]]

for (i in 1:perm) { 
for (j in 1:length(unique(littledaleimodelperm[[i+1]][,1])))  
littledaleimodelperm[[i+1]][littledaleimodelperm[[i+1]][,1] == unique(littledaleimodelperm[[i+1]][,1])[j],1] <- paste("LG",j, sep="")
}

littledaleimodelperm[[2]]

#IMPORTANT, MAKE SURE YOU RESTORE THE ORIGINAL GENOME!
source("genome.R")
genome <- genome[complete.cases(genome),]

for (i in 1:(perm + 1))
littledaleimodelperm[[i]] <- littledaleimodelperm[[i]][complete.cases(littledaleimodelperm[[i]]),]
littledaleimodelperm[[2]]

linkssimperm <- list()
for (i in 1:perm)
linkssimperm[[i]] <- merge(genome, littledaleimodelperm[[i+1]], by = "Marker", sort = F, no.dups = TRUE)
linkssimperm[[67]]

length(genome[,4])
length(unique(genome[,4]))
dim(linkssimperm[[2]])

for (i in 1:perm)
linkssimperm[[i]] <- linkssimperm[[i]][,c(2,3,4,5,6,7)]

linkssimperm[[2]]

chr1dsimperm <- list()
for (i in 1:perm)
chr1dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr1",]
chr1dsimperm[[2]]

chr2dsimperm <- list()
for (i in 1:perm)
chr2dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr2",]
chr3dsimperm <- list()
for (i in 1:perm)
chr3dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr3",]
chr4dsimperm <- list()
for (i in 1:perm)
chr4dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr4",]
chr5dsimperm <- list()
for (i in 1:perm)
chr5dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr5",]
chr6dsimperm <- list()
for (i in 1:perm)
chr6dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr6",]
chr7dsimperm <- list()
for (i in 1:perm)
chr7dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr7",]
chr8dsimperm <- list()
for (i in 1:perm)
chr8dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr8",]
chr9dsimperm <- list()
for (i in 1:perm)
chr9dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr9",]
chr10dsimperm <- list()
for (i in 1:perm)
chr10dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr10",]
chr11dsimperm <- list()
for (i in 1:perm)
chr11dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr11",]
chr12dsimperm <- list()
for (i in 1:perm)
chr12dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr12",]
chr13dsimperm <- list()
for (i in 1:perm)
chr13dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr13",]
chr14dsimperm <- list()
for (i in 1:perm)
chr14dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr14",]
chr15dsimperm <- list()
for (i in 1:perm)
chr15dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr15",]
chr16dsimperm <- list()
for (i in 1:perm)
chr16dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr16",]
chr17dsimperm <- list()
for (i in 1:perm)
chr17dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr17",]
chr18dsimperm <- list()
for (i in 1:perm)
chr18dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr18",]
chr19dsimperm <- list()
for (i in 1:perm)
chr19dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr19",]
chr20dsimperm <- list()
for (i in 1:perm)
chr20dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr20",]
chr21dsimperm <- list()
for (i in 1:perm)
chr21dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr21",]
chr22dsimperm <- list()
for (i in 1:perm)
chr22dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr22",]
chr23dsimperm <- list()
for (i in 1:perm)
chr23dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr23",]
chr24dsimperm <- list()
for (i in 1:perm)
chr24dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr24",]
chr25dsimperm <- list()
for (i in 1:perm)
chr25dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr25",]
chr26dsimperm <- list()
for (i in 1:perm)
chr26dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr26",]
chr27dsimperm <- list()
for (i in 1:perm)
chr27dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr27",]
chr28dsimperm <- list()
for (i in 1:perm)
chr28dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr28",]
chr29dsimperm <- list()
for (i in 1:perm)
chr29dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr29",]

source("genomeblocks.R")
outblock1dsimperm <- list()
for (i in 1:perm)
outblock1dsimperm[[i]] <- genomeblocks(chr1dsimperm[[i]])
outblock2dsimperm <- list()
for (i in 1:perm)
outblock2dsimperm[[i]] <- genomeblocks(chr2dsimperm[[i]])
outblock3dsimperm <- list()
for (i in 1:perm)
outblock3dsimperm[[i]] <- genomeblocks(chr3dsimperm[[i]])
outblock4dsimperm <- list()
for (i in 1:perm)
outblock4dsimperm[[i]] <- genomeblocks(chr4dsimperm[[i]])
outblock5dsimperm <- list()
for (i in 1:perm)
outblock5dsimperm[[i]] <- genomeblocks(chr5dsimperm[[i]])
outblock6dsimperm <- list()
for (i in 1:perm)
outblock6dsimperm[[i]] <- genomeblocks(chr6dsimperm[[i]])
outblock7dsimperm <- list()
for (i in 1:perm)
outblock7dsimperm[[i]] <- genomeblocks(chr7dsimperm[[i]])
outblock8dsimperm <- list()
for (i in 1:perm)
outblock8dsimperm[[i]] <- genomeblocks(chr8dsimperm[[i]])
outblock9dsimperm <- list()
for (i in 1:perm)
outblock9dsimperm[[i]] <- genomeblocks(chr9dsimperm[[i]])
outblock10dsimperm <- list()
for (i in 1:perm)
outblock10dsimperm[[i]] <- genomeblocks(chr10dsimperm[[i]])
outblock11dsimperm <- list()
for (i in 1:perm)
outblock11dsimperm[[i]] <- genomeblocks(chr11dsimperm[[i]])
outblock12dsimperm <- list()
for (i in 1:perm)
outblock12dsimperm[[i]] <- genomeblocks(chr12dsimperm[[i]])
outblock13dsimperm <- list()
for (i in 1:perm)
outblock13dsimperm[[i]] <- genomeblocks(chr13dsimperm[[i]])
outblock14dsimperm <- list()
for (i in 1:perm)
outblock14dsimperm[[i]] <- genomeblocks(chr14dsimperm[[i]])
outblock15dsimperm <- list()
for (i in 1:perm)
outblock15dsimperm[[i]] <- genomeblocks(chr15dsimperm[[i]])
outblock16dsimperm <- list()
for (i in 1:perm)
outblock16dsimperm[[i]] <- genomeblocks(chr16dsimperm[[i]])
outblock17dsimperm <- list()
for (i in 1:perm)
outblock17dsimperm[[i]] <- genomeblocks(chr17dsimperm[[i]])
outblock18dsimperm <- list()
for (i in 1:perm)
outblock18dsimperm[[i]] <- genomeblocks(chr18dsimperm[[i]])
outblock19dsimperm <- list()
for (i in 1:perm)
outblock19dsimperm[[i]] <- genomeblocks(chr19dsimperm[[i]])
outblock20dsimperm <- list()
for (i in 1:perm)
outblock20dsimperm[[i]] <- genomeblocks(chr20dsimperm[[i]])
outblock21dsimperm <- list()
for (i in 1:perm)
outblock21dsimperm[[i]] <- genomeblocks(chr21dsimperm[[i]])
outblock22dsimperm <- list()
for (i in 1:perm)
outblock22dsimperm[[i]] <- genomeblocks(chr22dsimperm[[i]])
outblock23dsimperm <- list()
for (i in 1:perm)
outblock23dsimperm[[i]] <- genomeblocks(chr23dsimperm[[i]])
outblock24dsimperm <- list()
for (i in 1:perm)
outblock24dsimperm[[i]] <- genomeblocks(chr24dsimperm[[i]])
outblock25dsimperm <- list()
for (i in 1:perm)
outblock25dsimperm[[i]] <- genomeblocks(chr25dsimperm[[i]])
outblock26dsimperm <- list()
for (i in 1:perm)
outblock26dsimperm[[i]] <- genomeblocks(chr26dsimperm[[i]])
outblock27dsimperm <- list()
for (i in 1:perm)
outblock27dsimperm[[i]] <- genomeblocks(chr27dsimperm[[i]])
outblock28dsimperm <- list()
for (i in 1:perm)
outblock28dsimperm[[i]] <- genomeblocks(chr28dsimperm[[i]])
outblock29dsimperm <- list()
for (i in 1:perm)
outblock29dsimperm[[i]] <- genomeblocks(chr29dsimperm[[i]])


outblockdsimperm <- list()
for (i in 1:perm)
outblockdsimperm[[i]] <- rbind(outblock1dsimperm[[i]],outblock2dsimperm[[i]],outblock3dsimperm[[i]],outblock4dsimperm[[i]],outblock5dsimperm[[i]],outblock6dsimperm[[i]],outblock7dsimperm[[i]],outblock8dsimperm[[i]],outblock9dsimperm[[i]],outblock10dsimperm[[i]],outblock11dsimperm[[i]],outblock12dsimperm[[i]],outblock13dsimperm[[i]],outblock14dsimperm[[i]],outblock15dsimperm[[i]],outblock16dsimperm[[i]],outblock17dsimperm[[i]],outblock18dsimperm[[i]],outblock19dsimperm[[i]],outblock20dsimperm[[i]],outblock21dsimperm[[i]],outblock22dsimperm[[i]],outblock23dsimperm[[i]],outblock24dsimperm[[i]],outblock25dsimperm[[i]],outblock26dsimperm[[i]],outblock27dsimperm[[i]],outblock28dsimperm[[i]],outblock29dsimperm[[i]])

numberofconserved <- vector()
for (i in 1:perm)
numberofconserved[i] <- dim(outblockdsimperm[[i]][outblockdsimperm[[i]][,4]=="conserved",])[1]
numberofconserved
numberofsinglemarker <- vector()
for (i in 1:perm)
numberofsinglemarker[i] <- dim(outblockdsimperm[[i]][outblockdsimperm[[i]][,4]=="singlemarker",])[1]
numberofsinglemarker
numberoffisfus <- vector()
for (i in 1:perm)
numberoffisfus[i] <- dim(outblockdsimperm[[i]][outblockdsimperm[[i]][,4]=="fisfus",])[1]
numberoffisfus
sumconserved <- vector()
for (i in 1:perm)
sumconserved[i] <- sum(as.numeric(outblockdsimperm[[i]][outblockdsimperm[[i]][,4]=="conserved",3]) - as.numeric(outblockdsimperm[[i]][outblockdsimperm[[i]][,4]=="conserved",2]))
sumconserved
meanconserved <- vector()
for (i in 1:perm)
meanconserved[i] <- mean(as.numeric(outblockdsimperm[[i]][outblockdsimperm[[i]][,4]=="conserved",3]) - as.numeric(outblockdsimperm[[i]][outblockdsimperm[[i]][,4]=="conserved",2]))
meanconserved

#Datos reales
genomeblocksBCR3_alld

dim(genomeblocksBCR3_alld[genomeblocksBCR3_alld[,4]=="conserved",])[1]
dim(genomeblocksBCR3_alld[genomeblocksBCR3_alld[,4]=="singlemarker",])[1]
dim(genomeblocksBCR3_alld[genomeblocksBCR3_alld[,4]=="fisfus",])[1]

#make some histograms!!!!
hist(numberoffisfus, main = "number of rearrangements")
abline(v=dim(genomeblocksBCR3_alld[genomeblocksBCR3_alld[,4]=="fisfus",])[1])

sort(numberoffisfus)

sort(as.numeric(genomeblocksBCR3_alld[genomeblocksBCR3_alld[,4]=="conserved",][,3]) - as.numeric(genomeblocksBCR3_alld[genomeblocksBCR3_alld[,4]=="conserved",][,2]))
mean(as.numeric(genomeblocksBCR3_alld[genomeblocksBCR3_alld[,4]=="conserved",][,3]) - as.numeric(genomeblocksBCR3_alld[genomeblocksBCR3_alld[,4]=="conserved",][,2]))
sd(as.numeric(genomeblocksBCR3_alld[genomeblocksBCR3_alld[,4]=="conserved",][,3]) - as.numeric(genomeblocksBCR3_alld[genomeblocksBCR3_alld[,4]=="conserved",][,2]))

hist(as.numeric(genomeblocksBCR3_alld[genomeblocksBCR3_alld[,4]=="conserved",][,3]) - as.numeric(genomeblocksBCR3_alld[genomeblocksBCR3_alld[,4]=="conserved",][,2]))

```


Analyses of the TEs analyzed by Andr Marques
```{r}
#TE sent by Andr Marques
TE <- read.table("Kobresia_littledalei_TE.bed", sep = "\t")
TE <- TE[,1:3]
TE[TE[,1]=="CM022079",1] <- "chr1"
TE[TE[,1]=="CM022080",1] <- "chr2"
TE[TE[,1]=="CM022081",1] <- "chr3"
TE[TE[,1]=="CM022082",1] <- "chr4"
TE[TE[,1]=="CM022083",1] <- "chr5"
TE[TE[,1]=="CM022084",1] <- "chr6"
TE[TE[,1]=="CM022085",1] <- "chr7"
TE[TE[,1]=="CM022086",1] <- "chr8"
TE[TE[,1]=="CM022087",1] <- "chr9"
TE[TE[,1]=="CM022088",1] <- "chr10"
TE[TE[,1]=="CM022089",1] <- "chr11"
TE[TE[,1]=="CM022090",1] <- "chr12"
TE[TE[,1]=="CM022091",1] <- "chr13"
TE[TE[,1]=="CM022092",1] <- "chr14"
TE[TE[,1]=="CM022093",1] <- "chr15"
TE[TE[,1]=="CM022094",1] <- "chr16"
TE[TE[,1]=="CM022095",1] <- "chr17"
TE[TE[,1]=="CM022096",1] <- "chr18"
TE[TE[,1]=="CM022097",1] <- "chr19"
TE[TE[,1]=="CM022098",1] <- "chr20"
TE[TE[,1]=="CM022099",1] <- "chr21"
TE[TE[,1]=="CM022100",1] <- "chr22"
TE[TE[,1]=="CM022101",1] <- "chr23"
TE[TE[,1]=="CM022102",1] <- "chr24"
TE[TE[,1]=="CM022103",1] <- "chr25"
TE[TE[,1]=="CM022104",1] <- "chr26"
TE[TE[,1]=="CM022105",1] <- "chr27"
TE[TE[,1]=="CM022106",1] <- "chr28"
TE[TE[,1]=="CM022107",1] <- "chr29"
colnames(TE) <- c("Chromosome", "Start", "End")

TE_1 <- TE[TE$Chromosome=="chr1",]
TE_1
TE_2 <- TE[TE$Chromosome=="chr2",]
TE_2
TE_3 <- TE[TE$Chromosome=="chr3",]
TE_3
TE_4 <- TE[TE$Chromosome=="chr4",]
TE_4
TE_5 <- TE[TE$Chromosome=="chr5",]
TE_5
TE_6 <- TE[TE$Chromosome=="chr6",]
TE_6
TE_7 <- TE[TE$Chromosome=="chr7",]
TE_7
TE_8 <- TE[TE$Chromosome=="chr8",]
TE_8
TE_9 <- TE[TE$Chromosome=="chr9",]
TE_9
TE_10 <- TE[TE$Chromosome=="chr10",]
TE_10
TE_11 <- TE[TE$Chromosome=="chr11",]
TE_11
TE_12 <- TE[TE$Chromosome=="chr12",]
TE_12
TE_13 <- TE[TE$Chromosome=="chr13",]
TE_13
TE_14 <- TE[TE$Chromosome=="chr14",]
TE_14
TE_15 <- TE[TE$Chromosome=="chr15",]
TE_15
TE_16 <- TE[TE$Chromosome=="chr16",]
TE_16
TE_17 <- TE[TE$Chromosome=="chr17",]
TE_17
TE_18 <- TE[TE$Chromosome=="chr18",]
TE_18
TE_19 <- TE[TE$Chromosome=="chr19",]
TE_19
TE_20 <- TE[TE$Chromosome=="chr20",]
TE_20
TE_21 <- TE[TE$Chromosome=="chr21",]
TE_21
TE_22 <- TE[TE$Chromosome=="chr22",]
TE_22
TE_23 <- TE[TE$Chromosome=="chr23",]
TE_23
TE_24 <- TE[TE$Chromosome=="chr24",]
TE_24
TE_25 <- TE[TE$Chromosome=="chr25",]
TE_25
TE_26 <- TE[TE$Chromosome=="chr26",]
TE_26
TE_27 <- TE[TE$Chromosome=="chr27",]
TE_27
TE_28 <- TE[TE$Chromosome=="chr28",]
TE_28
TE_29 <- TE[TE$Chromosome=="chr29",]
TE_29

genomeblocksTE_1d <- genomeblocksBCR(outblock1d, TE_1)
genomeblocksTE_1d
genomeblocksTE_2d <- genomeblocksBCR(outblock2d, TE_2)
genomeblocksTE_2d
genomeblocksTE_3d <- genomeblocksBCR(outblock3d, TE_3)
genomeblocksTE_3d
genomeblocksTE_4d <- genomeblocksBCR(outblock4d, TE_4)
genomeblocksTE_4d
genomeblocksTE_5d <- genomeblocksBCR(outblock5d, TE_5)
genomeblocksTE_5d
genomeblocksTE_6d <- genomeblocksBCR(outblock6d, TE_6)
genomeblocksTE_6d
genomeblocksTE_7d <- genomeblocksBCR(outblock7d, TE_7)
genomeblocksTE_7d
genomeblocksTE_8d <- genomeblocksBCR(outblock8d, TE_8)
genomeblocksTE_8d
genomeblocksTE_9d <- genomeblocksBCR(outblock9d, TE_9)
genomeblocksTE_9d
genomeblocksTE_10d <- genomeblocksBCR(outblock10d, TE_10)
genomeblocksTE_10d
genomeblocksTE_11d <- genomeblocksBCR(outblock11d, TE_11)
genomeblocksTE_11d
genomeblocksTE_12d <- genomeblocksBCR(outblock12d, TE_12)
genomeblocksTE_12d
genomeblocksTE_13d <- genomeblocksBCR(outblock13d, TE_13)
genomeblocksTE_13d
genomeblocksTE_14d <- genomeblocksBCR(outblock14d, TE_14)
genomeblocksTE_14d
genomeblocksTE_15d <- genomeblocksBCR(outblock15d, TE_15)
genomeblocksTE_15d
genomeblocksTE_16d <- genomeblocksBCR(outblock16d, TE_16)
genomeblocksTE_16d
genomeblocksTE_17d <- genomeblocksBCR(outblock17d, TE_17)
genomeblocksTE_17d
genomeblocksTE_18d <- genomeblocksBCR(outblock18d, TE_18)
genomeblocksTE_18d
genomeblocksTE_19d <- genomeblocksBCR(outblock19d, TE_19)
genomeblocksTE_19d
genomeblocksTE_20d <- genomeblocksBCR(outblock20d, TE_20)
genomeblocksTE_20d
genomeblocksTE_21d <- genomeblocksBCR(outblock21d, TE_21)
genomeblocksTE_21d
genomeblocksTE_22d <- genomeblocksBCR(outblock22d, TE_22)
genomeblocksTE_22d
genomeblocksTE_23d <- genomeblocksBCR(outblock23d, TE_23)
genomeblocksTE_23d
genomeblocksTE_24d <- genomeblocksBCR(outblock24d, TE_24)
genomeblocksTE_24d
genomeblocksTE_25d <- genomeblocksBCR(outblock25d, TE_25)
genomeblocksTE_25d
genomeblocksTE_26d <- genomeblocksBCR(outblock26d, TE_26)
genomeblocksTE_26d
genomeblocksTE_27d <- genomeblocksBCR(outblock27d, TE_27)
genomeblocksTE_27d
genomeblocksTE_28d <- genomeblocksBCR(outblock28d, TE_28)
genomeblocksTE_28d
genomeblocksTE_29d <- genomeblocksBCR(outblock29d, TE_29)
genomeblocksTE_29d

genomeblocksTE_alld <- rbind(genomeblocksTE_1d,genomeblocksTE_2d,genomeblocksTE_3d,genomeblocksTE_4d,genomeblocksTE_5d,genomeblocksTE_6d,genomeblocksTE_7d,genomeblocksTE_8d,genomeblocksTE_9d,genomeblocksTE_10d,genomeblocksTE_11d,genomeblocksTE_12d,genomeblocksTE_13d,genomeblocksTE_14d,genomeblocksTE_15d,genomeblocksTE_16d,genomeblocksTE_17d,genomeblocksTE_18d,genomeblocksTE_19d,genomeblocksTE_20d,genomeblocksTE_21d,genomeblocksTE_22d,genomeblocksTE_23d,genomeblocksTE_24d,genomeblocksTE_25d,genomeblocksTE_26d,genomeblocksTE_27d,genomeblocksTE_28d,genomeblocksTE_29d)

length(genomeblocksTE_alld[,4][genomeblocksTE_alld[,4] == "conserved"])
length(genomeblocksTE_alld[,4][genomeblocksTE_alld[,4] == "fisfus"])
length(genomeblocksTE_alld[,4][genomeblocksTE_alld[,4] == "singlemarker"])

mean(as.numeric(genomeblocksTE_alld[,5][genomeblocksTE_alld[,4] == "conserved"]))
mean(as.numeric(genomeblocksTE_alld[,5][genomeblocksTE_alld[,4] == "fisfus"]))
mean(as.numeric(genomeblocksTE_alld[,5][genomeblocksTE_alld[,4] == "singlemarker"]))

sd(as.numeric(genomeblocksTE_alld[,5][genomeblocksTE_alld[,4] == "conserved"]))
sd(as.numeric(genomeblocksTE_alld[,5][genomeblocksTE_alld[,4] == "fisfus"]))
sd(as.numeric(genomeblocksTE_alld[,5][genomeblocksTE_alld[,4] == "singlemarker"]))

dim(genomeblocksTE_alld)
dim(genomeblocksTE_alld[genomeblocksTE_alld[,4] != "singlemarker", ])

dim(genomeblocksGenesalld)
dim(genomeblocksGenesalld[genomeblocksGenesalld[,5] != "singlemarker", ])

glm_genomeblocks_alld <- genomeblocksGenesalld[genomeblocksGenesalld[,5] != "singlemarker", -c(2:4)]

glm_genomeblocks_alld <- cbind(glm_genomeblocks_alld,genomeblocksTE_alld[genomeblocksTE_alld[,4] != "singlemarker", 5])

glm_genomeblocks_alld[,2][glm_genomeblocks_alld[,2]=="conserved"] <- 0
glm_genomeblocks_alld[,2][glm_genomeblocks_alld[,2]=="fisfus"] <- 1

library(lme4)
glm_genomeblocks_alld<- data.frame (glm_genomeblocks_alld)
colnames(glm_genomeblocks_alld) <- c("chromosome", "type", "BCR",  "GC", "Genes", "TE")
fittedModel <- glmer (as.numeric(type) ~ as.numeric(BCR) + as.numeric(GC) + as.numeric(Genes) + as.numeric(TE) + (1|chromosome), data = glm_genomeblocks_alld, family = binomial(link = "logit"))
summary(fittedModel)

fittedModel2 <- glm (as.numeric(type) ~ as.numeric(BCR) + as.numeric(GC) + as.numeric(Genes) + as.numeric(TE), family = binomial(link = "logit"), data = glm_genomeblocks_alld)
summary(fittedModel2)

fittedModel3 <- glm (as.numeric(type) ~ as.numeric(BCR) + as.numeric(GC) + as.numeric(Genes) + as.numeric(TE), family = gaussian(link = "identity"), data = glm_genomeblocks_alld)
summary(fittedModel3)

fittedModel4 <- glm (as.numeric(type) ~ as.numeric(BCR) + as.numeric(GC) + as.numeric(Genes)+ as.numeric(Genes), family = poisson(link = "log"), data = glm_genomeblocks_alld)
summary(fittedModel4)

library(DHARMa)

simulationOutput <- simulateResiduals(fittedModel = fittedModel, plot = F)
plot(simulationOutput)

simulationOutput2 <- simulateResiduals(fittedModel = fittedModel2, plot = F)
plot(simulationOutput2)

simulationOutput3 <- simulateResiduals(fittedModel = fittedModel3, plot = F)
plot(simulationOutput3)

simulationOutput4 <- simulateResiduals(fittedModel = fittedModel4, plot = F)
plot(simulationOutput4)

```

Adding the analysis of SatDNA

```{r}
satDNA <- read.table("Carex_littledalei_satDNA.bed", sep = "\t")
satDNA <- satDNA[,1:3]
satDNA[satDNA[,1]=="CM022079",1] <- "chr1"
satDNA[satDNA[,1]=="CM022080",1] <- "chr2"
satDNA[satDNA[,1]=="CM022081",1] <- "chr3"
satDNA[satDNA[,1]=="CM022082",1] <- "chr4"
satDNA[satDNA[,1]=="CM022083",1] <- "chr5"
satDNA[satDNA[,1]=="CM022084",1] <- "chr6"
satDNA[satDNA[,1]=="CM022085",1] <- "chr7"
satDNA[satDNA[,1]=="CM022086",1] <- "chr8"
satDNA[satDNA[,1]=="CM022087",1] <- "chr9"
satDNA[satDNA[,1]=="CM022088",1] <- "chr10"
satDNA[satDNA[,1]=="CM022089",1] <- "chr11"
satDNA[satDNA[,1]=="CM022090",1] <- "chr12"
satDNA[satDNA[,1]=="CM022091",1] <- "chr13"
satDNA[satDNA[,1]=="CM022092",1] <- "chr14"
satDNA[satDNA[,1]=="CM022093",1] <- "chr15"
satDNA[satDNA[,1]=="CM022094",1] <- "chr16"
satDNA[satDNA[,1]=="CM022095",1] <- "chr17"
satDNA[satDNA[,1]=="CM022096",1] <- "chr18"
satDNA[satDNA[,1]=="CM022097",1] <- "chr19"
satDNA[satDNA[,1]=="CM022098",1] <- "chr20"
satDNA[satDNA[,1]=="CM022099",1] <- "chr21"
satDNA[satDNA[,1]=="CM022100",1] <- "chr22"
satDNA[satDNA[,1]=="CM022101",1] <- "chr23"
satDNA[satDNA[,1]=="CM022102",1] <- "chr24"
satDNA[satDNA[,1]=="CM022103",1] <- "chr25"
satDNA[satDNA[,1]=="CM022104",1] <- "chr26"
satDNA[satDNA[,1]=="CM022105",1] <- "chr27"
satDNA[satDNA[,1]=="CM022106",1] <- "chr28"
satDNA[satDNA[,1]=="CM022107",1] <- "chr29"
colnames(satDNA) <- c("Chromosome", "Start", "End")

satDNA_1 <- satDNA[satDNA$Chromosome=="chr1",]
satDNA_1
satDNA_2 <- satDNA[satDNA$Chromosome=="chr2",]
satDNA_2
satDNA_3 <- satDNA[satDNA$Chromosome=="chr3",]
satDNA_3
satDNA_4 <- satDNA[satDNA$Chromosome=="chr4",]
satDNA_4
satDNA_5 <- satDNA[satDNA$Chromosome=="chr5",]
satDNA_5
satDNA_6 <- satDNA[satDNA$Chromosome=="chr6",]
satDNA_6
satDNA_7 <- satDNA[satDNA$Chromosome=="chr7",]
satDNA_7
satDNA_8 <- satDNA[satDNA$Chromosome=="chr8",]
satDNA_8
satDNA_9 <- satDNA[satDNA$Chromosome=="chr9",]
satDNA_9
satDNA_10 <- satDNA[satDNA$Chromosome=="chr10",]
satDNA_10
satDNA_11 <- satDNA[satDNA$Chromosome=="chr11",]
satDNA_11
satDNA_12 <- satDNA[satDNA$Chromosome=="chr12",]
satDNA_12
satDNA_13 <- satDNA[satDNA$Chromosome=="chr13",]
satDNA_13
satDNA_14 <- satDNA[satDNA$Chromosome=="chr14",]
satDNA_14
satDNA_15 <- satDNA[satDNA$Chromosome=="chr15",]
satDNA_15
satDNA_16 <- satDNA[satDNA$Chromosome=="chr16",]
satDNA_16
satDNA_17 <- satDNA[satDNA$Chromosome=="chr17",]
satDNA_17
satDNA_18 <- satDNA[satDNA$Chromosome=="chr18",]
satDNA_18
satDNA_19 <- satDNA[satDNA$Chromosome=="chr19",]
satDNA_19
satDNA_20 <- satDNA[satDNA$Chromosome=="chr20",]
satDNA_20
satDNA_21 <- satDNA[satDNA$Chromosome=="chr21",]
satDNA_21
satDNA_22 <- satDNA[satDNA$Chromosome=="chr22",]
satDNA_22
satDNA_23 <- satDNA[satDNA$Chromosome=="chr23",]
satDNA_23
satDNA_24 <- satDNA[satDNA$Chromosome=="chr24",]
satDNA_24
satDNA_25 <- satDNA[satDNA$Chromosome=="chr25",]
satDNA_25
satDNA_26 <- satDNA[satDNA$Chromosome=="chr26",]
satDNA_26
satDNA_27 <- satDNA[satDNA$Chromosome=="chr27",]
satDNA_27
satDNA_28 <- satDNA[satDNA$Chromosome=="chr28",]
satDNA_28
satDNA_29 <- satDNA[satDNA$Chromosome=="chr29",]
satDNA_29

genomeblockssatDNA_1d <- genomeblocksBCR(outblock1d, satDNA_1)
genomeblockssatDNA_1d
genomeblockssatDNA_2d <- genomeblocksBCR(outblock2d, satDNA_2)
genomeblockssatDNA_2d
genomeblockssatDNA_3d <- genomeblocksBCR(outblock3d, satDNA_3)
genomeblockssatDNA_3d
genomeblockssatDNA_4d <- genomeblocksBCR(outblock4d, satDNA_4)
genomeblockssatDNA_4d
genomeblockssatDNA_5d <- genomeblocksBCR(outblock5d, satDNA_5)
genomeblockssatDNA_5d
genomeblockssatDNA_6d <- genomeblocksBCR(outblock6d, satDNA_6)
genomeblockssatDNA_6d
genomeblockssatDNA_7d <- genomeblocksBCR(outblock7d, satDNA_7)
genomeblockssatDNA_7d
genomeblockssatDNA_8d <- genomeblocksBCR(outblock8d, satDNA_8)
genomeblockssatDNA_8d
genomeblockssatDNA_9d <- genomeblocksBCR(outblock9d, satDNA_9)
genomeblockssatDNA_9d
genomeblockssatDNA_10d <- genomeblocksBCR(outblock10d, satDNA_10)
genomeblockssatDNA_10d
genomeblockssatDNA_11d <- genomeblocksBCR(outblock11d, satDNA_11)
genomeblockssatDNA_11d
genomeblockssatDNA_12d <- genomeblocksBCR(outblock12d, satDNA_12)
genomeblockssatDNA_12d
genomeblockssatDNA_13d <- genomeblocksBCR(outblock13d, satDNA_13)
genomeblockssatDNA_13d
genomeblockssatDNA_14d <- genomeblocksBCR(outblock14d, satDNA_14)
genomeblockssatDNA_14d
genomeblockssatDNA_15d <- genomeblocksBCR(outblock15d, satDNA_15)
genomeblockssatDNA_15d
genomeblockssatDNA_16d <- genomeblocksBCR(outblock16d, satDNA_16)
genomeblockssatDNA_16d
genomeblockssatDNA_17d <- genomeblocksBCR(outblock17d, satDNA_17)
genomeblockssatDNA_17d
genomeblockssatDNA_18d <- genomeblocksBCR(outblock18d, satDNA_18)
genomeblockssatDNA_18d
genomeblockssatDNA_19d <- genomeblocksBCR(outblock19d, satDNA_19)
genomeblockssatDNA_19d
genomeblockssatDNA_20d <- genomeblocksBCR(outblock20d, satDNA_20)
genomeblockssatDNA_20d
genomeblockssatDNA_21d <- genomeblocksBCR(outblock21d, satDNA_21)
genomeblockssatDNA_21d
genomeblockssatDNA_22d <- genomeblocksBCR(outblock22d, satDNA_22)
genomeblockssatDNA_22d
genomeblockssatDNA_23d <- genomeblocksBCR(outblock23d, satDNA_23)
genomeblockssatDNA_23d
genomeblockssatDNA_24d <- genomeblocksBCR(outblock24d, satDNA_24)
genomeblockssatDNA_24d
genomeblockssatDNA_25d <- genomeblocksBCR(outblock25d, satDNA_25)
genomeblockssatDNA_25d
genomeblockssatDNA_26d <- genomeblocksBCR(outblock26d, satDNA_26)
genomeblockssatDNA_26d
genomeblockssatDNA_27d <- genomeblocksBCR(outblock27d, satDNA_27)
genomeblockssatDNA_27d
genomeblockssatDNA_28d <- genomeblocksBCR(outblock28d, satDNA_28)
genomeblockssatDNA_28d
genomeblockssatDNA_29d <- genomeblocksBCR(outblock29d, satDNA_29)
genomeblockssatDNA_29d

genomeblockssatDNA_alld <- rbind(genomeblockssatDNA_1d,genomeblockssatDNA_2d,genomeblockssatDNA_3d,genomeblockssatDNA_4d,genomeblockssatDNA_5d,genomeblockssatDNA_6d,genomeblockssatDNA_7d,genomeblockssatDNA_8d,genomeblockssatDNA_9d,genomeblockssatDNA_10d,genomeblockssatDNA_11d,genomeblockssatDNA_12d,genomeblockssatDNA_13d,genomeblockssatDNA_14d,genomeblockssatDNA_15d,genomeblockssatDNA_16d,genomeblockssatDNA_17d,genomeblockssatDNA_18d,genomeblockssatDNA_19d,genomeblockssatDNA_20d,genomeblockssatDNA_21d,genomeblockssatDNA_22d,genomeblockssatDNA_23d,genomeblockssatDNA_24d,genomeblockssatDNA_25d,genomeblockssatDNA_26d,genomeblockssatDNA_27d,genomeblockssatDNA_28d,genomeblockssatDNA_29d)

length(genomeblockssatDNA_alld[,4][genomeblockssatDNA_alld[,4] == "conserved"])
length(genomeblockssatDNA_alld[,4][genomeblockssatDNA_alld[,4] == "fisfus"])
length(genomeblockssatDNA_alld[,4][genomeblockssatDNA_alld[,4] == "singlemarker"])

mean(as.numeric(genomeblockssatDNA_alld[,5][genomeblockssatDNA_alld[,4] == "conserved"]))
mean(as.numeric(genomeblockssatDNA_alld[,5][genomeblockssatDNA_alld[,4] == "fisfus"]))
mean(as.numeric(genomeblockssatDNA_alld[,5][genomeblockssatDNA_alld[,4] == "singlemarker"]))

sd(as.numeric(genomeblockssatDNA_alld[,5][genomeblockssatDNA_alld[,4] == "conserved"]))
sd(as.numeric(genomeblockssatDNA_alld[,5][genomeblockssatDNA_alld[,4] == "fisfus"]))
sd(as.numeric(genomeblockssatDNA_alld[,5][genomeblockssatDNA_alld[,4] == "singlemarker"]))

dim(genomeblockssatDNA_alld)
dim(genomeblockssatDNA_alld[genomeblockssatDNA_alld[,4] != "singlemarker", ])

dim(genomeblocksGenesalld)
dim(genomeblocksGenesalld[genomeblocksGenesalld[,5] != "singlemarker", ])

glm_genomeblocks_alld <- genomeblocksGenesalld[genomeblocksGenesalld[,5] != "singlemarker", -c(2:4)]

glm_genomeblocks_alld <- cbind(glm_genomeblocks_alld,genomeblocksTE_alld[genomeblocksTE_alld[,4] != "singlemarker", 5],genomeblockssatDNA_alld[genomeblockssatDNA_alld[,4] != "singlemarker", 5])

glm_genomeblocks_alld[,2][glm_genomeblocks_alld[,2]=="conserved"] <- 0
glm_genomeblocks_alld[,2][glm_genomeblocks_alld[,2]=="fisfus"] <- 1

library(lme4)
glm_genomeblocks_alld<- data.frame (glm_genomeblocks_alld)
colnames(glm_genomeblocks_alld) <- c("chromosome", "type", "BCR",  "GC", "Genes", "TE", "satDNA")
fittedModel <- glmer (as.numeric(type) ~ as.numeric(BCR) + as.numeric(GC) + as.numeric(Genes) + as.numeric(TE) + as.numeric(satDNA) + (1|chromosome), data = glm_genomeblocks_alld, family = binomial(link = "logit"))
summary(fittedModel)

fittedModel2 <- glm (as.numeric(type) ~ as.numeric(BCR) + as.numeric(GC) + as.numeric(Genes) + as.numeric(TE) + as.numeric(satDNA), family = binomial(link = "logit"), data = glm_genomeblocks_alld)
summary(fittedModel2)

fittedModel3 <- glm (as.numeric(type) ~ as.numeric(BCR) + as.numeric(GC) + as.numeric(Genes) + as.numeric(TE)+ as.numeric(satDNA), family = gaussian(link = "identity"), data = glm_genomeblocks_alld)
summary(fittedModel3)

fittedModel4 <- glm (as.numeric(type) ~ as.numeric(BCR) + as.numeric(GC) + as.numeric(Genes)+ as.numeric(TE) + as.numeric(satDNA), family = poisson(link = "log"), data = glm_genomeblocks_alld)
summary(fittedModel4)

library(DHARMa)

simulationOutput <- simulateResiduals(fittedModel = fittedModel, plot = F)
plot(simulationOutput)

simulationOutput2 <- simulateResiduals(fittedModel = fittedModel2, plot = F)
plot(simulationOutput2)

simulationOutput3 <- simulateResiduals(fittedModel = fittedModel3, plot = F)
plot(simulationOutput3)

simulationOutput4 <- simulateResiduals(fittedModel = fittedModel4, plot = F)
plot(simulationOutput4)
```

Now we are goint to test different types of TEs. First class_II.

```{r}

gypsy <- read.table("Kobresia_littledalei_gypsy.bed", sep = "\t")
gypsy <- gypsy[,1:3]
gypsy[gypsy[,1]=="CM022079",1] <- "chr1"
gypsy[gypsy[,1]=="CM022080",1] <- "chr2"
gypsy[gypsy[,1]=="CM022081",1] <- "chr3"
gypsy[gypsy[,1]=="CM022082",1] <- "chr4"
gypsy[gypsy[,1]=="CM022083",1] <- "chr5"
gypsy[gypsy[,1]=="CM022084",1] <- "chr6"
gypsy[gypsy[,1]=="CM022085",1] <- "chr7"
gypsy[gypsy[,1]=="CM022086",1] <- "chr8"
gypsy[gypsy[,1]=="CM022087",1] <- "chr9"
gypsy[gypsy[,1]=="CM022088",1] <- "chr10"
gypsy[gypsy[,1]=="CM022089",1] <- "chr11"
gypsy[gypsy[,1]=="CM022090",1] <- "chr12"
gypsy[gypsy[,1]=="CM022091",1] <- "chr13"
gypsy[gypsy[,1]=="CM022092",1] <- "chr14"
gypsy[gypsy[,1]=="CM022093",1] <- "chr15"
gypsy[gypsy[,1]=="CM022094",1] <- "chr16"
gypsy[gypsy[,1]=="CM022095",1] <- "chr17"
gypsy[gypsy[,1]=="CM022096",1] <- "chr18"
gypsy[gypsy[,1]=="CM022097",1] <- "chr19"
gypsy[gypsy[,1]=="CM022098",1] <- "chr20"
gypsy[gypsy[,1]=="CM022099",1] <- "chr21"
gypsy[gypsy[,1]=="CM022100",1] <- "chr22"
gypsy[gypsy[,1]=="CM022101",1] <- "chr23"
gypsy[gypsy[,1]=="CM022102",1] <- "chr24"
gypsy[gypsy[,1]=="CM022103",1] <- "chr25"
gypsy[gypsy[,1]=="CM022104",1] <- "chr26"
gypsy[gypsy[,1]=="CM022105",1] <- "chr27"
gypsy[gypsy[,1]=="CM022106",1] <- "chr28"
gypsy[gypsy[,1]=="CM022107",1] <- "chr29"
colnames(gypsy) <- c("Chromosome", "Start", "End")

gypsy_1 <- gypsy[gypsy$Chromosome=="chr1",]
gypsy_1
gypsy_2 <- gypsy[gypsy$Chromosome=="chr2",]
gypsy_2
gypsy_3 <- gypsy[gypsy$Chromosome=="chr3",]
gypsy_3
gypsy_4 <- gypsy[gypsy$Chromosome=="chr4",]
gypsy_4
gypsy_5 <- gypsy[gypsy$Chromosome=="chr5",]
gypsy_5
gypsy_6 <- gypsy[gypsy$Chromosome=="chr6",]
gypsy_6
gypsy_7 <- gypsy[gypsy$Chromosome=="chr7",]
gypsy_7
gypsy_8 <- gypsy[gypsy$Chromosome=="chr8",]
gypsy_8
gypsy_9 <- gypsy[gypsy$Chromosome=="chr9",]
gypsy_9
gypsy_10 <- gypsy[gypsy$Chromosome=="chr10",]
gypsy_10
gypsy_11 <- gypsy[gypsy$Chromosome=="chr11",]
gypsy_11
gypsy_12 <- gypsy[gypsy$Chromosome=="chr12",]
gypsy_12
gypsy_13 <- gypsy[gypsy$Chromosome=="chr13",]
gypsy_13
gypsy_14 <- gypsy[gypsy$Chromosome=="chr14",]
gypsy_14
gypsy_15 <- gypsy[gypsy$Chromosome=="chr15",]
gypsy_15
gypsy_16 <- gypsy[gypsy$Chromosome=="chr16",]
gypsy_16
gypsy_17 <- gypsy[gypsy$Chromosome=="chr17",]
gypsy_17
gypsy_18 <- gypsy[gypsy$Chromosome=="chr18",]
gypsy_18
gypsy_19 <- gypsy[gypsy$Chromosome=="chr19",]
gypsy_19
gypsy_20 <- gypsy[gypsy$Chromosome=="chr20",]
gypsy_20
gypsy_21 <- gypsy[gypsy$Chromosome=="chr21",]
gypsy_21
gypsy_22 <- gypsy[gypsy$Chromosome=="chr22",]
gypsy_22
gypsy_23 <- gypsy[gypsy$Chromosome=="chr23",]
gypsy_23
gypsy_24 <- gypsy[gypsy$Chromosome=="chr24",]
gypsy_24
gypsy_25 <- gypsy[gypsy$Chromosome=="chr25",]
gypsy_25
gypsy_26 <- gypsy[gypsy$Chromosome=="chr26",]
gypsy_26
gypsy_27 <- gypsy[gypsy$Chromosome=="chr27",]
gypsy_27
gypsy_28 <- gypsy[gypsy$Chromosome=="chr28",]
gypsy_28
gypsy_29 <- gypsy[gypsy$Chromosome=="chr29",]
gypsy_29

genomeblocksgypsy_1d <- genomeblocksBCR(outblock1d, gypsy_1)
genomeblocksgypsy_1d
genomeblocksgypsy_2d <- genomeblocksBCR(outblock2d, gypsy_2)
genomeblocksgypsy_2d
genomeblocksgypsy_3d <- genomeblocksBCR(outblock3d, gypsy_3)
genomeblocksgypsy_3d
genomeblocksgypsy_4d <- genomeblocksBCR(outblock4d, gypsy_4)
genomeblocksgypsy_4d
genomeblocksgypsy_5d <- genomeblocksBCR(outblock5d, gypsy_5)
genomeblocksgypsy_5d
genomeblocksgypsy_6d <- genomeblocksBCR(outblock6d, gypsy_6)
genomeblocksgypsy_6d
genomeblocksgypsy_7d <- genomeblocksBCR(outblock7d, gypsy_7)
genomeblocksgypsy_7d
genomeblocksgypsy_8d <- genomeblocksBCR(outblock8d, gypsy_8)
genomeblocksgypsy_8d
genomeblocksgypsy_9d <- genomeblocksBCR(outblock9d, gypsy_9)
genomeblocksgypsy_9d
genomeblocksgypsy_10d <- genomeblocksBCR(outblock10d, gypsy_10)
genomeblocksgypsy_10d
genomeblocksgypsy_11d <- genomeblocksBCR(outblock11d, gypsy_11)
genomeblocksgypsy_11d
genomeblocksgypsy_12d <- genomeblocksBCR(outblock12d, gypsy_12)
genomeblocksgypsy_12d
genomeblocksgypsy_13d <- genomeblocksBCR(outblock13d, gypsy_13)
genomeblocksgypsy_13d
genomeblocksgypsy_14d <- genomeblocksBCR(outblock14d, gypsy_14)
genomeblocksgypsy_14d
genomeblocksgypsy_15d <- genomeblocksBCR(outblock15d, gypsy_15)
genomeblocksgypsy_15d
genomeblocksgypsy_16d <- genomeblocksBCR(outblock16d, gypsy_16)
genomeblocksgypsy_16d
genomeblocksgypsy_17d <- genomeblocksBCR(outblock17d, gypsy_17)
genomeblocksgypsy_17d
genomeblocksgypsy_18d <- genomeblocksBCR(outblock18d, gypsy_18)
genomeblocksgypsy_18d
genomeblocksgypsy_19d <- genomeblocksBCR(outblock19d, gypsy_19)
genomeblocksgypsy_19d
genomeblocksgypsy_20d <- genomeblocksBCR(outblock20d, gypsy_20)
genomeblocksgypsy_20d
genomeblocksgypsy_21d <- genomeblocksBCR(outblock21d, gypsy_21)
genomeblocksgypsy_21d
genomeblocksgypsy_22d <- genomeblocksBCR(outblock22d, gypsy_22)
genomeblocksgypsy_22d
genomeblocksgypsy_23d <- genomeblocksBCR(outblock23d, gypsy_23)
genomeblocksgypsy_23d
genomeblocksgypsy_24d <- genomeblocksBCR(outblock24d, gypsy_24)
genomeblocksgypsy_24d
genomeblocksgypsy_25d <- genomeblocksBCR(outblock25d, gypsy_25)
genomeblocksgypsy_25d
genomeblocksgypsy_26d <- genomeblocksBCR(outblock26d, gypsy_26)
genomeblocksgypsy_26d
genomeblocksgypsy_27d <- genomeblocksBCR(outblock27d, gypsy_27)
genomeblocksgypsy_27d
genomeblocksgypsy_28d <- genomeblocksBCR(outblock28d, gypsy_28)
genomeblocksgypsy_28d
genomeblocksgypsy_29d <- genomeblocksBCR(outblock29d, gypsy_29)
genomeblocksgypsy_29d

genomeblocksgypsy_alld <- rbind(genomeblocksgypsy_1d,genomeblocksgypsy_2d,genomeblocksgypsy_3d,genomeblocksgypsy_4d,genomeblocksgypsy_5d,genomeblocksgypsy_6d,genomeblocksgypsy_7d,genomeblocksgypsy_8d,genomeblocksgypsy_9d,genomeblocksgypsy_10d,genomeblocksgypsy_11d,genomeblocksgypsy_12d,genomeblocksgypsy_13d,genomeblocksgypsy_14d,genomeblocksgypsy_15d,genomeblocksgypsy_16d,genomeblocksgypsy_17d,genomeblocksgypsy_18d,genomeblocksgypsy_19d,genomeblocksgypsy_20d,genomeblocksgypsy_21d,genomeblocksgypsy_22d,genomeblocksgypsy_23d,genomeblocksgypsy_24d,genomeblocksgypsy_25d,genomeblocksgypsy_26d,genomeblocksgypsy_27d,genomeblocksgypsy_28d,genomeblocksgypsy_29d)

length(genomeblocksgypsy_alld[,4][genomeblocksgypsy_alld[,4] == "conserved"])
length(genomeblocksgypsy_alld[,4][genomeblocksgypsy_alld[,4] == "fisfus"])
length(genomeblocksgypsy_alld[,4][genomeblocksgypsy_alld[,4] == "singlemarker"])

mean(as.numeric(genomeblocksgypsy_alld[,5][genomeblocksgypsy_alld[,4] == "conserved"]))
mean(as.numeric(genomeblocksgypsy_alld[,5][genomeblocksgypsy_alld[,4] == "fisfus"]))
mean(as.numeric(genomeblocksgypsy_alld[,5][genomeblocksgypsy_alld[,4] == "singlemarker"]))

sd(as.numeric(genomeblocksgypsy_alld[,5][genomeblocksgypsy_alld[,4] == "conserved"]))
sd(as.numeric(genomeblocksgypsy_alld[,5][genomeblocksgypsy_alld[,4] == "fisfus"]))
sd(as.numeric(genomeblocksgypsy_alld[,5][genomeblocksgypsy_alld[,4] == "singlemarker"]))

dim(genomeblocksgypsy_alld)
dim(genomeblocksgypsy_alld[genomeblocksgypsy_alld[,4] != "singlemarker", ])

dim(genomeblocksGenesalld)
dim(genomeblocksGenesalld[genomeblocksGenesalld[,5] != "singlemarker", ])


```


Now we are goint to test different types of TEs. classII.

```{r}

class_II <- read.table("Kobresia_littledalei_class_II.bed", sep = "\t")
class_II <- class_II[,1:3]
class_II[class_II[,1]=="CM022079",1] <- "chr1"
class_II[class_II[,1]=="CM022080",1] <- "chr2"
class_II[class_II[,1]=="CM022081",1] <- "chr3"
class_II[class_II[,1]=="CM022082",1] <- "chr4"
class_II[class_II[,1]=="CM022083",1] <- "chr5"
class_II[class_II[,1]=="CM022084",1] <- "chr6"
class_II[class_II[,1]=="CM022085",1] <- "chr7"
class_II[class_II[,1]=="CM022086",1] <- "chr8"
class_II[class_II[,1]=="CM022087",1] <- "chr9"
class_II[class_II[,1]=="CM022088",1] <- "chr10"
class_II[class_II[,1]=="CM022089",1] <- "chr11"
class_II[class_II[,1]=="CM022090",1] <- "chr12"
class_II[class_II[,1]=="CM022091",1] <- "chr13"
class_II[class_II[,1]=="CM022092",1] <- "chr14"
class_II[class_II[,1]=="CM022093",1] <- "chr15"
class_II[class_II[,1]=="CM022094",1] <- "chr16"
class_II[class_II[,1]=="CM022095",1] <- "chr17"
class_II[class_II[,1]=="CM022096",1] <- "chr18"
class_II[class_II[,1]=="CM022097",1] <- "chr19"
class_II[class_II[,1]=="CM022098",1] <- "chr20"
class_II[class_II[,1]=="CM022099",1] <- "chr21"
class_II[class_II[,1]=="CM022100",1] <- "chr22"
class_II[class_II[,1]=="CM022101",1] <- "chr23"
class_II[class_II[,1]=="CM022102",1] <- "chr24"
class_II[class_II[,1]=="CM022103",1] <- "chr25"
class_II[class_II[,1]=="CM022104",1] <- "chr26"
class_II[class_II[,1]=="CM022105",1] <- "chr27"
class_II[class_II[,1]=="CM022106",1] <- "chr28"
class_II[class_II[,1]=="CM022107",1] <- "chr29"
colnames(class_II) <- c("Chromosome", "Start", "End")

class_II_1 <- class_II[class_II$Chromosome=="chr1",]
class_II_1
class_II_2 <- class_II[class_II$Chromosome=="chr2",]
class_II_2
class_II_3 <- class_II[class_II$Chromosome=="chr3",]
class_II_3
class_II_4 <- class_II[class_II$Chromosome=="chr4",]
class_II_4
class_II_5 <- class_II[class_II$Chromosome=="chr5",]
class_II_5
class_II_6 <- class_II[class_II$Chromosome=="chr6",]
class_II_6
class_II_7 <- class_II[class_II$Chromosome=="chr7",]
class_II_7
class_II_8 <- class_II[class_II$Chromosome=="chr8",]
class_II_8
class_II_9 <- class_II[class_II$Chromosome=="chr9",]
class_II_9
class_II_10 <- class_II[class_II$Chromosome=="chr10",]
class_II_10
class_II_11 <- class_II[class_II$Chromosome=="chr11",]
class_II_11
class_II_12 <- class_II[class_II$Chromosome=="chr12",]
class_II_12
class_II_13 <- class_II[class_II$Chromosome=="chr13",]
class_II_13
class_II_14 <- class_II[class_II$Chromosome=="chr14",]
class_II_14
class_II_15 <- class_II[class_II$Chromosome=="chr15",]
class_II_15
class_II_16 <- class_II[class_II$Chromosome=="chr16",]
class_II_16
class_II_17 <- class_II[class_II$Chromosome=="chr17",]
class_II_17
class_II_18 <- class_II[class_II$Chromosome=="chr18",]
class_II_18
class_II_19 <- class_II[class_II$Chromosome=="chr19",]
class_II_19
class_II_20 <- class_II[class_II$Chromosome=="chr20",]
class_II_20
class_II_21 <- class_II[class_II$Chromosome=="chr21",]
class_II_21
class_II_22 <- class_II[class_II$Chromosome=="chr22",]
class_II_22
class_II_23 <- class_II[class_II$Chromosome=="chr23",]
class_II_23
class_II_24 <- class_II[class_II$Chromosome=="chr24",]
class_II_24
class_II_25 <- class_II[class_II$Chromosome=="chr25",]
class_II_25
class_II_26 <- class_II[class_II$Chromosome=="chr26",]
class_II_26
class_II_27 <- class_II[class_II$Chromosome=="chr27",]
class_II_27
class_II_28 <- class_II[class_II$Chromosome=="chr28",]
class_II_28
class_II_29 <- class_II[class_II$Chromosome=="chr29",]
class_II_29

genomeblocksclass_II_1d <- genomeblocksBCR(outblock1d, class_II_1)
genomeblocksclass_II_1d
genomeblocksclass_II_2d <- genomeblocksBCR(outblock2d, class_II_2)
genomeblocksclass_II_2d
genomeblocksclass_II_3d <- genomeblocksBCR(outblock3d, class_II_3)
genomeblocksclass_II_3d
genomeblocksclass_II_4d <- genomeblocksBCR(outblock4d, class_II_4)
genomeblocksclass_II_4d
genomeblocksclass_II_5d <- genomeblocksBCR(outblock5d, class_II_5)
genomeblocksclass_II_5d
genomeblocksclass_II_6d <- genomeblocksBCR(outblock6d, class_II_6)
genomeblocksclass_II_6d
genomeblocksclass_II_7d <- genomeblocksBCR(outblock7d, class_II_7)
genomeblocksclass_II_7d
genomeblocksclass_II_8d <- genomeblocksBCR(outblock8d, class_II_8)
genomeblocksclass_II_8d
genomeblocksclass_II_9d <- genomeblocksBCR(outblock9d, class_II_9)
genomeblocksclass_II_9d
genomeblocksclass_II_10d <- genomeblocksBCR(outblock10d, class_II_10)
genomeblocksclass_II_10d
genomeblocksclass_II_11d <- genomeblocksBCR(outblock11d, class_II_11)
genomeblocksclass_II_11d
genomeblocksclass_II_12d <- genomeblocksBCR(outblock12d, class_II_12)
genomeblocksclass_II_12d
genomeblocksclass_II_13d <- genomeblocksBCR(outblock13d, class_II_13)
genomeblocksclass_II_13d
genomeblocksclass_II_14d <- genomeblocksBCR(outblock14d, class_II_14)
genomeblocksclass_II_14d
genomeblocksclass_II_15d <- genomeblocksBCR(outblock15d, class_II_15)
genomeblocksclass_II_15d
genomeblocksclass_II_16d <- genomeblocksBCR(outblock16d, class_II_16)
genomeblocksclass_II_16d
genomeblocksclass_II_17d <- genomeblocksBCR(outblock17d, class_II_17)
genomeblocksclass_II_17d
genomeblocksclass_II_18d <- genomeblocksBCR(outblock18d, class_II_18)
genomeblocksclass_II_18d
genomeblocksclass_II_19d <- genomeblocksBCR(outblock19d, class_II_19)
genomeblocksclass_II_19d
genomeblocksclass_II_20d <- genomeblocksBCR(outblock20d, class_II_20)
genomeblocksclass_II_20d
genomeblocksclass_II_21d <- genomeblocksBCR(outblock21d, class_II_21)
genomeblocksclass_II_21d
genomeblocksclass_II_22d <- genomeblocksBCR(outblock22d, class_II_22)
genomeblocksclass_II_22d
genomeblocksclass_II_23d <- genomeblocksBCR(outblock23d, class_II_23)
genomeblocksclass_II_23d
genomeblocksclass_II_24d <- genomeblocksBCR(outblock24d, class_II_24)
genomeblocksclass_II_24d
genomeblocksclass_II_25d <- genomeblocksBCR(outblock25d, class_II_25)
genomeblocksclass_II_25d
genomeblocksclass_II_26d <- genomeblocksBCR(outblock26d, class_II_26)
genomeblocksclass_II_26d
genomeblocksclass_II_27d <- genomeblocksBCR(outblock27d, class_II_27)
genomeblocksclass_II_27d
genomeblocksclass_II_28d <- genomeblocksBCR(outblock28d, class_II_28)
genomeblocksclass_II_28d
genomeblocksclass_II_29d <- genomeblocksBCR(outblock29d, class_II_29)
genomeblocksclass_II_29d

genomeblocksclass_II_alld <- rbind(genomeblocksclass_II_1d,genomeblocksclass_II_2d,genomeblocksclass_II_3d,genomeblocksclass_II_4d,genomeblocksclass_II_5d,genomeblocksclass_II_6d,genomeblocksclass_II_7d,genomeblocksclass_II_8d,genomeblocksclass_II_9d,genomeblocksclass_II_10d,genomeblocksclass_II_11d,genomeblocksclass_II_12d,genomeblocksclass_II_13d,genomeblocksclass_II_14d,genomeblocksclass_II_15d,genomeblocksclass_II_16d,genomeblocksclass_II_17d,genomeblocksclass_II_18d,genomeblocksclass_II_19d,genomeblocksclass_II_20d,genomeblocksclass_II_21d,genomeblocksclass_II_22d,genomeblocksclass_II_23d,genomeblocksclass_II_24d,genomeblocksclass_II_25d,genomeblocksclass_II_26d,genomeblocksclass_II_27d,genomeblocksclass_II_28d,genomeblocksclass_II_29d)

length(genomeblocksclass_II_alld[,4][genomeblocksclass_II_alld[,4] == "conserved"])
length(genomeblocksclass_II_alld[,4][genomeblocksclass_II_alld[,4] == "fisfus"])
length(genomeblocksclass_II_alld[,4][genomeblocksclass_II_alld[,4] == "singlemarker"])

mean(as.numeric(genomeblocksclass_II_alld[,5][genomeblocksclass_II_alld[,4] == "conserved"]))
mean(as.numeric(genomeblocksclass_II_alld[,5][genomeblocksclass_II_alld[,4] == "fisfus"]))
mean(as.numeric(genomeblocksclass_II_alld[,5][genomeblocksclass_II_alld[,4] == "singlemarker"]))

sd(as.numeric(genomeblocksclass_II_alld[,5][genomeblocksclass_II_alld[,4] == "conserved"]))
sd(as.numeric(genomeblocksclass_II_alld[,5][genomeblocksclass_II_alld[,4] == "fisfus"]))
sd(as.numeric(genomeblocksclass_II_alld[,5][genomeblocksclass_II_alld[,4] == "singlemarker"]))

dim(genomeblocksclass_II_alld)
dim(genomeblocksclass_II_alld[genomeblocksclass_II_alld[,4] != "singlemarker", ])

dim(genomeblocksGenesalld)
dim(genomeblocksGenesalld[genomeblocksGenesalld[,5] != "singlemarker", ])


```


Now we are goint to test different types of TEs. copia.

```{r}

copia <- read.table("Kobresia_littledalei_copia.bed", sep = "\t")
copia <- copia[,1:3]
copia[copia[,1]=="CM022079",1] <- "chr1"
copia[copia[,1]=="CM022080",1] <- "chr2"
copia[copia[,1]=="CM022081",1] <- "chr3"
copia[copia[,1]=="CM022082",1] <- "chr4"
copia[copia[,1]=="CM022083",1] <- "chr5"
copia[copia[,1]=="CM022084",1] <- "chr6"
copia[copia[,1]=="CM022085",1] <- "chr7"
copia[copia[,1]=="CM022086",1] <- "chr8"
copia[copia[,1]=="CM022087",1] <- "chr9"
copia[copia[,1]=="CM022088",1] <- "chr10"
copia[copia[,1]=="CM022089",1] <- "chr11"
copia[copia[,1]=="CM022090",1] <- "chr12"
copia[copia[,1]=="CM022091",1] <- "chr13"
copia[copia[,1]=="CM022092",1] <- "chr14"
copia[copia[,1]=="CM022093",1] <- "chr15"
copia[copia[,1]=="CM022094",1] <- "chr16"
copia[copia[,1]=="CM022095",1] <- "chr17"
copia[copia[,1]=="CM022096",1] <- "chr18"
copia[copia[,1]=="CM022097",1] <- "chr19"
copia[copia[,1]=="CM022098",1] <- "chr20"
copia[copia[,1]=="CM022099",1] <- "chr21"
copia[copia[,1]=="CM022100",1] <- "chr22"
copia[copia[,1]=="CM022101",1] <- "chr23"
copia[copia[,1]=="CM022102",1] <- "chr24"
copia[copia[,1]=="CM022103",1] <- "chr25"
copia[copia[,1]=="CM022104",1] <- "chr26"
copia[copia[,1]=="CM022105",1] <- "chr27"
copia[copia[,1]=="CM022106",1] <- "chr28"
copia[copia[,1]=="CM022107",1] <- "chr29"
colnames(copia) <- c("Chromosome", "Start", "End")

copia_1 <- copia[copia$Chromosome=="chr1",]
copia_1
copia_2 <- copia[copia$Chromosome=="chr2",]
copia_2
copia_3 <- copia[copia$Chromosome=="chr3",]
copia_3
copia_4 <- copia[copia$Chromosome=="chr4",]
copia_4
copia_5 <- copia[copia$Chromosome=="chr5",]
copia_5
copia_6 <- copia[copia$Chromosome=="chr6",]
copia_6
copia_7 <- copia[copia$Chromosome=="chr7",]
copia_7
copia_8 <- copia[copia$Chromosome=="chr8",]
copia_8
copia_9 <- copia[copia$Chromosome=="chr9",]
copia_9
copia_10 <- copia[copia$Chromosome=="chr10",]
copia_10
copia_11 <- copia[copia$Chromosome=="chr11",]
copia_11
copia_12 <- copia[copia$Chromosome=="chr12",]
copia_12
copia_13 <- copia[copia$Chromosome=="chr13",]
copia_13
copia_14 <- copia[copia$Chromosome=="chr14",]
copia_14
copia_15 <- copia[copia$Chromosome=="chr15",]
copia_15
copia_16 <- copia[copia$Chromosome=="chr16",]
copia_16
copia_17 <- copia[copia$Chromosome=="chr17",]
copia_17
copia_18 <- copia[copia$Chromosome=="chr18",]
copia_18
copia_19 <- copia[copia$Chromosome=="chr19",]
copia_19
copia_20 <- copia[copia$Chromosome=="chr20",]
copia_20
copia_21 <- copia[copia$Chromosome=="chr21",]
copia_21
copia_22 <- copia[copia$Chromosome=="chr22",]
copia_22
copia_23 <- copia[copia$Chromosome=="chr23",]
copia_23
copia_24 <- copia[copia$Chromosome=="chr24",]
copia_24
copia_25 <- copia[copia$Chromosome=="chr25",]
copia_25
copia_26 <- copia[copia$Chromosome=="chr26",]
copia_26
copia_27 <- copia[copia$Chromosome=="chr27",]
copia_27
copia_28 <- copia[copia$Chromosome=="chr28",]
copia_28
copia_29 <- copia[copia$Chromosome=="chr29",]
copia_29

genomeblockscopia_1d <- genomeblocksBCR(outblock1d, copia_1)
genomeblockscopia_1d
genomeblockscopia_2d <- genomeblocksBCR(outblock2d, copia_2)
genomeblockscopia_2d
genomeblockscopia_3d <- genomeblocksBCR(outblock3d, copia_3)
genomeblockscopia_3d
genomeblockscopia_4d <- genomeblocksBCR(outblock4d, copia_4)
genomeblockscopia_4d
genomeblockscopia_5d <- genomeblocksBCR(outblock5d, copia_5)
genomeblockscopia_5d
genomeblockscopia_6d <- genomeblocksBCR(outblock6d, copia_6)
genomeblockscopia_6d
genomeblockscopia_7d <- genomeblocksBCR(outblock7d, copia_7)
genomeblockscopia_7d
genomeblockscopia_8d <- genomeblocksBCR(outblock8d, copia_8)
genomeblockscopia_8d
genomeblockscopia_9d <- genomeblocksBCR(outblock9d, copia_9)
genomeblockscopia_9d
genomeblockscopia_10d <- genomeblocksBCR(outblock10d, copia_10)
genomeblockscopia_10d
genomeblockscopia_11d <- genomeblocksBCR(outblock11d, copia_11)
genomeblockscopia_11d
genomeblockscopia_12d <- genomeblocksBCR(outblock12d, copia_12)
genomeblockscopia_12d
genomeblockscopia_13d <- genomeblocksBCR(outblock13d, copia_13)
genomeblockscopia_13d
genomeblockscopia_14d <- genomeblocksBCR(outblock14d, copia_14)
genomeblockscopia_14d
genomeblockscopia_15d <- genomeblocksBCR(outblock15d, copia_15)
genomeblockscopia_15d
genomeblockscopia_16d <- genomeblocksBCR(outblock16d, copia_16)
genomeblockscopia_16d
genomeblockscopia_17d <- genomeblocksBCR(outblock17d, copia_17)
genomeblockscopia_17d
genomeblockscopia_18d <- genomeblocksBCR(outblock18d, copia_18)
genomeblockscopia_18d
genomeblockscopia_19d <- genomeblocksBCR(outblock19d, copia_19)
genomeblockscopia_19d
genomeblockscopia_20d <- genomeblocksBCR(outblock20d, copia_20)
genomeblockscopia_20d
genomeblockscopia_21d <- genomeblocksBCR(outblock21d, copia_21)
genomeblockscopia_21d
genomeblockscopia_22d <- genomeblocksBCR(outblock22d, copia_22)
genomeblockscopia_22d
genomeblockscopia_23d <- genomeblocksBCR(outblock23d, copia_23)
genomeblockscopia_23d
genomeblockscopia_24d <- genomeblocksBCR(outblock24d, copia_24)
genomeblockscopia_24d
genomeblockscopia_25d <- genomeblocksBCR(outblock25d, copia_25)
genomeblockscopia_25d
genomeblockscopia_26d <- genomeblocksBCR(outblock26d, copia_26)
genomeblockscopia_26d
genomeblockscopia_27d <- genomeblocksBCR(outblock27d, copia_27)
genomeblockscopia_27d
genomeblockscopia_28d <- genomeblocksBCR(outblock28d, copia_28)
genomeblockscopia_28d
genomeblockscopia_29d <- genomeblocksBCR(outblock29d, copia_29)
genomeblockscopia_29d

genomeblockscopia_alld <- rbind(genomeblockscopia_1d,genomeblockscopia_2d,genomeblockscopia_3d,genomeblockscopia_4d,genomeblockscopia_5d,genomeblockscopia_6d,genomeblockscopia_7d,genomeblockscopia_8d,genomeblockscopia_9d,genomeblockscopia_10d,genomeblockscopia_11d,genomeblockscopia_12d,genomeblockscopia_13d,genomeblockscopia_14d,genomeblockscopia_15d,genomeblockscopia_16d,genomeblockscopia_17d,genomeblockscopia_18d,genomeblockscopia_19d,genomeblockscopia_20d,genomeblockscopia_21d,genomeblockscopia_22d,genomeblockscopia_23d,genomeblockscopia_24d,genomeblockscopia_25d,genomeblockscopia_26d,genomeblockscopia_27d,genomeblockscopia_28d,genomeblockscopia_29d)

length(genomeblockscopia_alld[,4][genomeblockscopia_alld[,4] == "conserved"])
length(genomeblockscopia_alld[,4][genomeblockscopia_alld[,4] == "fisfus"])
length(genomeblockscopia_alld[,4][genomeblockscopia_alld[,4] == "singlemarker"])

mean(as.numeric(genomeblockscopia_alld[,5][genomeblockscopia_alld[,4] == "conserved"]))
mean(as.numeric(genomeblockscopia_alld[,5][genomeblockscopia_alld[,4] == "fisfus"]))
mean(as.numeric(genomeblockscopia_alld[,5][genomeblockscopia_alld[,4] == "singlemarker"]))

sd(as.numeric(genomeblockscopia_alld[,5][genomeblockscopia_alld[,4] == "conserved"]))
sd(as.numeric(genomeblockscopia_alld[,5][genomeblockscopia_alld[,4] == "fisfus"]))
sd(as.numeric(genomeblockscopia_alld[,5][genomeblockscopia_alld[,4] == "singlemarker"]))

dim(genomeblockscopia_alld)
dim(genomeblockscopia_alld[genomeblockscopia_alld[,4] != "singlemarker", ])

dim(genomeblocksGenesalld)
dim(genomeblocksGenesalld[genomeblocksGenesalld[,5] != "singlemarker", ])


```


Now we are goint to test different types of TEs. non_LTR.

```{r}

non_LTR <- read.table("Kobresia_littledalei_non_LTR.bed", sep = "\t")
non_LTR <- non_LTR[,1:3]
non_LTR[non_LTR[,1]=="CM022079",1] <- "chr1"
non_LTR[non_LTR[,1]=="CM022080",1] <- "chr2"
non_LTR[non_LTR[,1]=="CM022081",1] <- "chr3"
non_LTR[non_LTR[,1]=="CM022082",1] <- "chr4"
non_LTR[non_LTR[,1]=="CM022083",1] <- "chr5"
non_LTR[non_LTR[,1]=="CM022084",1] <- "chr6"
non_LTR[non_LTR[,1]=="CM022085",1] <- "chr7"
non_LTR[non_LTR[,1]=="CM022086",1] <- "chr8"
non_LTR[non_LTR[,1]=="CM022087",1] <- "chr9"
non_LTR[non_LTR[,1]=="CM022088",1] <- "chr10"
non_LTR[non_LTR[,1]=="CM022089",1] <- "chr11"
non_LTR[non_LTR[,1]=="CM022090",1] <- "chr12"
non_LTR[non_LTR[,1]=="CM022091",1] <- "chr13"
non_LTR[non_LTR[,1]=="CM022092",1] <- "chr14"
non_LTR[non_LTR[,1]=="CM022093",1] <- "chr15"
non_LTR[non_LTR[,1]=="CM022094",1] <- "chr16"
non_LTR[non_LTR[,1]=="CM022095",1] <- "chr17"
non_LTR[non_LTR[,1]=="CM022096",1] <- "chr18"
non_LTR[non_LTR[,1]=="CM022097",1] <- "chr19"
non_LTR[non_LTR[,1]=="CM022098",1] <- "chr20"
non_LTR[non_LTR[,1]=="CM022099",1] <- "chr21"
non_LTR[non_LTR[,1]=="CM022100",1] <- "chr22"
non_LTR[non_LTR[,1]=="CM022101",1] <- "chr23"
non_LTR[non_LTR[,1]=="CM022102",1] <- "chr24"
non_LTR[non_LTR[,1]=="CM022103",1] <- "chr25"
non_LTR[non_LTR[,1]=="CM022104",1] <- "chr26"
non_LTR[non_LTR[,1]=="CM022105",1] <- "chr27"
non_LTR[non_LTR[,1]=="CM022106",1] <- "chr28"
non_LTR[non_LTR[,1]=="CM022107",1] <- "chr29"
colnames(non_LTR) <- c("Chromosome", "Start", "End")

non_LTR_1 <- non_LTR[non_LTR$Chromosome=="chr1",]
non_LTR_1
non_LTR_2 <- non_LTR[non_LTR$Chromosome=="chr2",]
non_LTR_2
non_LTR_3 <- non_LTR[non_LTR$Chromosome=="chr3",]
non_LTR_3
non_LTR_4 <- non_LTR[non_LTR$Chromosome=="chr4",]
non_LTR_4
non_LTR_5 <- non_LTR[non_LTR$Chromosome=="chr5",]
non_LTR_5
non_LTR_6 <- non_LTR[non_LTR$Chromosome=="chr6",]
non_LTR_6
non_LTR_7 <- non_LTR[non_LTR$Chromosome=="chr7",]
non_LTR_7
non_LTR_8 <- non_LTR[non_LTR$Chromosome=="chr8",]
non_LTR_8
non_LTR_9 <- non_LTR[non_LTR$Chromosome=="chr9",]
non_LTR_9
non_LTR_10 <- non_LTR[non_LTR$Chromosome=="chr10",]
non_LTR_10
non_LTR_11 <- non_LTR[non_LTR$Chromosome=="chr11",]
non_LTR_11
non_LTR_12 <- non_LTR[non_LTR$Chromosome=="chr12",]
non_LTR_12
non_LTR_13 <- non_LTR[non_LTR$Chromosome=="chr13",]
non_LTR_13
non_LTR_14 <- non_LTR[non_LTR$Chromosome=="chr14",]
non_LTR_14
non_LTR_15 <- non_LTR[non_LTR$Chromosome=="chr15",]
non_LTR_15
non_LTR_16 <- non_LTR[non_LTR$Chromosome=="chr16",]
non_LTR_16
non_LTR_17 <- non_LTR[non_LTR$Chromosome=="chr17",]
non_LTR_17
non_LTR_18 <- non_LTR[non_LTR$Chromosome=="chr18",]
non_LTR_18
non_LTR_19 <- non_LTR[non_LTR$Chromosome=="chr19",]
non_LTR_19
non_LTR_20 <- non_LTR[non_LTR$Chromosome=="chr20",]
non_LTR_20
non_LTR_21 <- non_LTR[non_LTR$Chromosome=="chr21",]
non_LTR_21
non_LTR_22 <- non_LTR[non_LTR$Chromosome=="chr22",]
non_LTR_22
non_LTR_23 <- non_LTR[non_LTR$Chromosome=="chr23",]
non_LTR_23
non_LTR_24 <- non_LTR[non_LTR$Chromosome=="chr24",]
non_LTR_24
non_LTR_25 <- non_LTR[non_LTR$Chromosome=="chr25",]
non_LTR_25
non_LTR_26 <- non_LTR[non_LTR$Chromosome=="chr26",]
non_LTR_26
non_LTR_27 <- non_LTR[non_LTR$Chromosome=="chr27",]
non_LTR_27
non_LTR_28 <- non_LTR[non_LTR$Chromosome=="chr28",]
non_LTR_28
non_LTR_29 <- non_LTR[non_LTR$Chromosome=="chr29",]
non_LTR_29

genomeblocksnon_LTR_1d <- genomeblocksBCR(outblock1d, non_LTR_1)
genomeblocksnon_LTR_1d
genomeblocksnon_LTR_2d <- genomeblocksBCR(outblock2d, non_LTR_2)
genomeblocksnon_LTR_2d
genomeblocksnon_LTR_3d <- genomeblocksBCR(outblock3d, non_LTR_3)
genomeblocksnon_LTR_3d
genomeblocksnon_LTR_4d <- genomeblocksBCR(outblock4d, non_LTR_4)
genomeblocksnon_LTR_4d
genomeblocksnon_LTR_5d <- genomeblocksBCR(outblock5d, non_LTR_5)
genomeblocksnon_LTR_5d
genomeblocksnon_LTR_6d <- genomeblocksBCR(outblock6d, non_LTR_6)
genomeblocksnon_LTR_6d
genomeblocksnon_LTR_7d <- genomeblocksBCR(outblock7d, non_LTR_7)
genomeblocksnon_LTR_7d
genomeblocksnon_LTR_8d <- genomeblocksBCR(outblock8d, non_LTR_8)
genomeblocksnon_LTR_8d
genomeblocksnon_LTR_9d <- genomeblocksBCR(outblock9d, non_LTR_9)
genomeblocksnon_LTR_9d
genomeblocksnon_LTR_10d <- genomeblocksBCR(outblock10d, non_LTR_10)
genomeblocksnon_LTR_10d
genomeblocksnon_LTR_11d <- genomeblocksBCR(outblock11d, non_LTR_11)
genomeblocksnon_LTR_11d
genomeblocksnon_LTR_12d <- genomeblocksBCR(outblock12d, non_LTR_12)
genomeblocksnon_LTR_12d
genomeblocksnon_LTR_13d <- genomeblocksBCR(outblock13d, non_LTR_13)
genomeblocksnon_LTR_13d
genomeblocksnon_LTR_14d <- genomeblocksBCR(outblock14d, non_LTR_14)
genomeblocksnon_LTR_14d
genomeblocksnon_LTR_15d <- genomeblocksBCR(outblock15d, non_LTR_15)
genomeblocksnon_LTR_15d
genomeblocksnon_LTR_16d <- genomeblocksBCR(outblock16d, non_LTR_16)
genomeblocksnon_LTR_16d
genomeblocksnon_LTR_17d <- genomeblocksBCR(outblock17d, non_LTR_17)
genomeblocksnon_LTR_17d
genomeblocksnon_LTR_18d <- genomeblocksBCR(outblock18d, non_LTR_18)
genomeblocksnon_LTR_18d
genomeblocksnon_LTR_19d <- genomeblocksBCR(outblock19d, non_LTR_19)
genomeblocksnon_LTR_19d
genomeblocksnon_LTR_20d <- genomeblocksBCR(outblock20d, non_LTR_20)
genomeblocksnon_LTR_20d
genomeblocksnon_LTR_21d <- genomeblocksBCR(outblock21d, non_LTR_21)
genomeblocksnon_LTR_21d
genomeblocksnon_LTR_22d <- genomeblocksBCR(outblock22d, non_LTR_22)
genomeblocksnon_LTR_22d
genomeblocksnon_LTR_23d <- genomeblocksBCR(outblock23d, non_LTR_23)
genomeblocksnon_LTR_23d
genomeblocksnon_LTR_24d <- genomeblocksBCR(outblock24d, non_LTR_24)
genomeblocksnon_LTR_24d
genomeblocksnon_LTR_25d <- genomeblocksBCR(outblock25d, non_LTR_25)
genomeblocksnon_LTR_25d
genomeblocksnon_LTR_26d <- genomeblocksBCR(outblock26d, non_LTR_26)
genomeblocksnon_LTR_26d
genomeblocksnon_LTR_27d <- genomeblocksBCR(outblock27d, non_LTR_27)
genomeblocksnon_LTR_27d
genomeblocksnon_LTR_28d <- genomeblocksBCR(outblock28d, non_LTR_28)
genomeblocksnon_LTR_28d
genomeblocksnon_LTR_29d <- genomeblocksBCR(outblock29d, non_LTR_29)
genomeblocksnon_LTR_29d

genomeblocksnon_LTR_alld <- rbind(genomeblocksnon_LTR_1d,genomeblocksnon_LTR_2d,genomeblocksnon_LTR_3d,genomeblocksnon_LTR_4d,genomeblocksnon_LTR_5d,genomeblocksnon_LTR_6d,genomeblocksnon_LTR_7d,genomeblocksnon_LTR_8d,genomeblocksnon_LTR_9d,genomeblocksnon_LTR_10d,genomeblocksnon_LTR_11d,genomeblocksnon_LTR_12d,genomeblocksnon_LTR_13d,genomeblocksnon_LTR_14d,genomeblocksnon_LTR_15d,genomeblocksnon_LTR_16d,genomeblocksnon_LTR_17d,genomeblocksnon_LTR_18d,genomeblocksnon_LTR_19d,genomeblocksnon_LTR_20d,genomeblocksnon_LTR_21d,genomeblocksnon_LTR_22d,genomeblocksnon_LTR_23d,genomeblocksnon_LTR_24d,genomeblocksnon_LTR_25d,genomeblocksnon_LTR_26d,genomeblocksnon_LTR_27d,genomeblocksnon_LTR_28d,genomeblocksnon_LTR_29d)

length(genomeblocksnon_LTR_alld[,4][genomeblocksnon_LTR_alld[,4] == "conserved"])
length(genomeblocksnon_LTR_alld[,4][genomeblocksnon_LTR_alld[,4] == "fisfus"])
length(genomeblocksnon_LTR_alld[,4][genomeblocksnon_LTR_alld[,4] == "singlemarker"])

mean(as.numeric(genomeblocksnon_LTR_alld[,5][genomeblocksnon_LTR_alld[,4] == "conserved"]))
mean(as.numeric(genomeblocksnon_LTR_alld[,5][genomeblocksnon_LTR_alld[,4] == "fisfus"]))
mean(as.numeric(genomeblocksnon_LTR_alld[,5][genomeblocksnon_LTR_alld[,4] == "singlemarker"]))

sd(as.numeric(genomeblocksnon_LTR_alld[,5][genomeblocksnon_LTR_alld[,4] == "conserved"]))
sd(as.numeric(genomeblocksnon_LTR_alld[,5][genomeblocksnon_LTR_alld[,4] == "fisfus"]))
sd(as.numeric(genomeblocksnon_LTR_alld[,5][genomeblocksnon_LTR_alld[,4] == "singlemarker"]))

dim(genomeblocksnon_LTR_alld)
dim(genomeblocksnon_LTR_alld[genomeblocksnon_LTR_alld[,4] != "singlemarker", ])

dim(genomeblocksGenesalld)
dim(genomeblocksGenesalld[genomeblocksGenesalld[,5] != "singlemarker", ])

glm_genomeblocks_alld <- genomeblocksGenesalld[genomeblocksGenesalld[,5] != "singlemarker", -c(2:4)]


```


TEs separately

```{r}

glm_genomeblocks_alld <- genomeblocksGenesalld[genomeblocksGenesalld[,5] != "singlemarker", -c(2:4)]

glm_genomeblocks_alld <- cbind(glm_genomeblocks_alld,genomeblockssatDNA_alld[genomeblockssatDNA_alld[,4] != "singlemarker", 5],genomeblocksclass_II_alld[genomeblocksclass_II_alld[,4] != "singlemarker", 5], genomeblockscopia_alld[genomeblockscopia_alld[,4] != "singlemarker", 5], genomeblocksgypsy_alld[genomeblocksgypsy_alld[,4] != "singlemarker", 5], genomeblocksnon_LTR_alld[genomeblocksnon_LTR_alld[,4] != "singlemarker", 5])

glm_genomeblocks_alld[,2][glm_genomeblocks_alld[,2]=="conserved"] <- 0
glm_genomeblocks_alld[,2][glm_genomeblocks_alld[,2]=="fisfus"] <- 1

library(lme4)
glm_genomeblocks_alld<- data.frame (glm_genomeblocks_alld)
colnames(glm_genomeblocks_alld) <- c("chromosome", "type", "BCR",  "GC", "Genes", "satDNA", "class_II", "copia", "gypsy", "non_LTR")
fittedModel <- glmer (as.numeric(type) ~ as.numeric(BCR) + as.numeric(GC) + as.numeric(Genes) + as.numeric(satDNA)+ as.numeric(class_II)+ as.numeric(copia) + as.numeric(gypsy) + as.numeric(non_LTR) + (1|chromosome), data = glm_genomeblocks_alld, family = binomial(link = "logit"))
summary(fittedModel)

fittedModel2 <- glm (as.numeric(type) ~ as.numeric(Genes) + as.numeric(satDNA)+ as.numeric(class_II)+ as.numeric(copia) + as.numeric(gypsy) + as.numeric(non_LTR), family = binomial(link = "logit"), data = glm_genomeblocks_alld)
summary(fittedModel2)

fittedModel3 <- glm (as.numeric(type) ~ as.numeric(BCR) + as.numeric(GC) + as.numeric(Genes) + as.numeric(satDNA)+ as.numeric(class_II)+ as.numeric(copia) + as.numeric(gypsy) + as.numeric(non_LTR), family = gaussian(link = "identity"), data = glm_genomeblocks_alld)
summary(fittedModel3)

fittedModel4 <- glm (as.numeric(type) ~ as.numeric(BCR) + as.numeric(GC) + as.numeric(Genes) + as.numeric(satDNA)+ as.numeric(class_II)+ as.numeric(copia) + as.numeric(gypsy) + as.numeric(non_LTR), family = poisson(link = "log"), data = glm_genomeblocks_alld)
summary(fittedModel4)

library(DHARMa)

simulationOutput <- simulateResiduals(fittedModel = fittedModel, plot = F)
plot(simulationOutput)

simulationOutput2 <- simulateResiduals(fittedModel = fittedModel2, plot = F)
plot(simulationOutput2)

simulationOutput3 <- simulateResiduals(fittedModel = fittedModel3, plot = F)
plot(simulationOutput3)

simulationOutput4 <- simulateResiduals(fittedModel = fittedModel4, plot = F)
plot(simulationOutput4)
```

