---
title: "CarexComparativeGenomics"
author: "Marcial Escudero"
date: "7/13/2021"
output: html_document
editor_options: 
  chunk_output_type: console
---

Lerning with the tutorial

```{r}
library(RCircos)

data(RCircos.Histogram.Data)
head(RCircos.Histogram.Data)

data(RCircos.Heatmap.Data)
head(RCircos.Heatmap.Data)

data(RCircos.Link.Data)
head(RCircos.Link.Data)

data(UCSC.HG19.Human.CytoBandIdeogram)
head(UCSC.HG19.Human.CytoBandIdeogram)

data(UCSC.Mouse.GRCm38.CytoBandIdeogram)
head(UCSC.Mouse.GRCm38.CytoBandIdeogram)

data(UCSC.Baylor.3.4.Rat.cytoBandIdeogram)
head(UCSC.Baylor.3.4.Rat.cytoBandIdeogram)


chr.exclude <- NULL
cyto.info <- UCSC.HG19.Human.CytoBandIdeogram
#cyto.info <- RCircos.Gene.Label.Data
tracks.inside <- 10
tracks.outside <- 0
RCircos.Set.Core.Components(cyto.info, chr.exclude,tracks.inside, tracks.outside)

rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.cyto <- RCircos.Get.Plot.Ideogram()
rcircos.position <- RCircos.Get.Plot.Positions()
RCircos.List.Plot.Parameters()

rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.params$base.per.unit <- 3000
RCircos.Reset.Plot.Parameters(rcircos.params)
RCircos.List.Plot.Parameters()

par(mai=c(0.25, 0.25, 0.25, 0.25))
plot.new()
plot.window(c(-2.5,2.5), c(-2.5, 2.5))

out.file <- "RCircosDemoHumanGenome.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)

RCircos.Set.Plot.Area()

RCircos.Chromosome.Ideogram.Plot()


data(RCircos.Gene.Label.Data)
head(RCircos.Gene.Label.Data)
name.col <- 4
side <- "in"
track.num <- 1
RCircos.Gene.Connector.Plot(RCircos.Gene.Label.Data,track.num, side)
track.num <- 2
RCircos.Gene.Name.Plot(RCircos.Gene.Label.Data,name.col,track.num, side)

data(RCircos.Heatmap.Data)
data.col <- 6
track.num <- 5
side <- "in"
RCircos.Heatmap.Plot(RCircos.Heatmap.Data, data.col, track.num, side)
data(RCircos.Scatter.Data)
data.col <- 5
track.num <- 6
side <- "in"
by.fold <- 1
RCircos.Scatter.Plot(RCircos.Scatter.Data, data.col,  track.num, side, by.fold)
data(RCircos.Line.Data)
data.col <- 5
track.num <- 7
side <- "in"
RCircos.Line.Plot(RCircos.Line.Data, data.col,track.num, side)
data(RCircos.Histogram.Data)
data.col <- 4
track.num <- 8
side <- "in"
RCircos.Histogram.Plot(RCircos.Histogram.Data,data.col, track.num, side)
data(RCircos.Tile.Data)
track.num <- 9
side <- "in"
RCircos.Tile.Plot(RCircos.Tile.Data, track.num, side)


data(RCircos.Link.Data)
track.num <- 11
RCircos.Link.Plot(RCircos.Link.Data, track.num, TRUE)
data(RCircos.Ribbon.Data)
RCircos.Ribbon.Plot(ribbon.data=RCircos.Ribbon.Data,track.num=11, by.chromosome=FALSE, twist=FALSE)


dev.off()

```


Opening some Carex data

```{r}

both <- read.csv("both.csv")

both

chr.exclude <- NULL
cyto.info <- both
tracks.inside <- 0
tracks.outside <- 0
RCircos.Set.Core.Components(cyto.info, chr.exclude,tracks.inside, tracks.outside)


rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.cyto <- RCircos.Get.Plot.Ideogram()
rcircos.position <- RCircos.Get.Plot.Positions()
RCircos.List.Plot.Parameters()

 

rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.params$base.per.unit <- 3000
RCircos.Reset.Plot.Parameters(rcircos.params)
RCircos.List.Plot.Parameters()

RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()



```

Setting up the bowtie2 analysis

```{r}

readsdata <- read.csv("batch_3.catalog.tags.tsv", sep = "")

readsdata[,3] <- paste0("M", readsdata[,3])

readsdata <- readsdata[,c(3,9)]

readsdata2 <- readsdata[readsdata[,1]%in%both[,4],]

write.csv(readsdata2, file = "reads.csv")



```


Making the comparative genomics analyses
```{r}

bowtie2 <- read.csv("bowtie2.csv", sep = "")
bowtie2

both

links <- merge(bowtie2, both, by.x = "Marker", by.y = "Band", sort = F)
links <- links[,c(2,3,4,5,6,7)]
links

track.num <- 1
RCircos.Link.Plot(links, track.num, TRUE)

out.file <- "ComparativeGenomicsLG1.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG1",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG2.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG2",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG3.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG3",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG4.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG4",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG5.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG5",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG6.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG6",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG7.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG7",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG8.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG8",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG9.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG9",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG10.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG10",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG11.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG11",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG12.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG12",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG13.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG13",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG14.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG14",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG15.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG15",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG16.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG16",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG17.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG17",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG18.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG18",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG19.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG19",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG20.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG20",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG21.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG21",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG22.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG22",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG23.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG23",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG24.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG24",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG25.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG25",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG26.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG26",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG27.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG27",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG28.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG28",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG29.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG29",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG30.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG30",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG31.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG31",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG32.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,4]=="LG32",], track.num, TRUE)
dev.off()




out.file <- "ComparativeGenomicschr1.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr1",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr2.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr2",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr3.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr3",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr4.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr4",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr5.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr5",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr6.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr6",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr7.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr7",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr8.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr8",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr9.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr9",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr10.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr10",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr11.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr11",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr12.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr12",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr13.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr13",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr14.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr14",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr15.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr15",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr16.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr16",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr17.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr17",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr18.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr18",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr19.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr19",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr20.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr20",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr21.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr21",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr22.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr22",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr23.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr23",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr24.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr24",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr25.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr25",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr26.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr26",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr27.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr27",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr28.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr28",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr29.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(links[links[,1]=="chr29",], track.num, TRUE)
dev.off()

```

Let's try to plot something that could be publish

```{r}

library(RColorBrewer)
n <- 30
qual_col_pals = brewer.pal.info[brewer.pal.info$category == 'qual',]
col_vector = unlist(mapply(brewer.pal, qual_col_pals$maxcolors, rownames(qual_col_pals)))
pie(rep(1,n), col=sample(col_vector, n))

col_vector[1:30]
col_vector <- c(col_vector[1:30],rep(col_vector[30],1381))

length(col_vector)
dim (both)

library(RCircos)
both <- read.csv("both.csv")

RCircos.Get.Plot.Colors(both, col_vector)

both$PlotColor <- col_vector

chr.exclude <- NULL
cyto.info <- both
tracks.inside <- 0
tracks.outside <- 0
RCircos.Set.Core.Components(cyto.info, chr.exclude,tracks.inside, tracks.outside)


rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.params
rcircos.cyto <- RCircos.Get.Plot.Ideogram()
rcircos.cyto
rcircos.position <- RCircos.Get.Plot.Positions()
rcircos.position
RCircos.List.Plot.Parameters()

#rcircos.params <- RCircos.Get.Plot.Parameters()
#rcircos.params$base.per.unit <- 3000
#RCircos.Reset.Plot.Parameters(rcircos.params)
#RCircos.List.Plot.Parameters()


rcircos.cyto
rcircos.cyto$BandColor <- rcircos.cyto$PlotColor
rcircos.cyto$ChrColor <-  rcircos.cyto$PlotColor
RCircos.Reset.Plot.Ideogram(rcircos.cyto)
RCircos.List.Plot.Parameters()


RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()


links$PlotColor <- 1
links[links[,1]=="chr1", "PlotColor"] <- col_vector[1]
links[links[,1]=="chr2", "PlotColor"] <- col_vector[2]
links[links[,1]=="chr3", "PlotColor"] <- col_vector[3]
links[links[,1]=="chr4", "PlotColor"] <- col_vector[4]
links[links[,1]=="chr5", "PlotColor"] <- col_vector[5]
links[links[,1]=="chr6", "PlotColor"] <- col_vector[6]
links[links[,1]=="chr7", "PlotColor"] <- col_vector[7]
links[links[,1]=="chr8", "PlotColor"] <- col_vector[8]
links[links[,1]=="chr9", "PlotColor"] <- col_vector[9]
links[links[,1]=="chr10", "PlotColor"] <- col_vector[10]
links[links[,1]=="chr11", "PlotColor"] <- col_vector[11]
links[links[,1]=="chr12", "PlotColor"] <- col_vector[12]
links[links[,1]=="chr13", "PlotColor"] <- col_vector[13]
links[links[,1]=="chr14", "PlotColor"] <- col_vector[14]
links[links[,1]=="chr15", "PlotColor"] <- col_vector[15]
links[links[,1]=="chr16", "PlotColor"] <- col_vector[16]
links[links[,1]=="chr17", "PlotColor"] <- col_vector[17]
links[links[,1]=="chr18", "PlotColor"] <- col_vector[18]
links[links[,1]=="chr19", "PlotColor"] <- col_vector[19]
links[links[,1]=="chr20", "PlotColor"] <- col_vector[20]
links[links[,1]=="chr21", "PlotColor"] <- col_vector[21]
links[links[,1]=="chr22", "PlotColor"] <- col_vector[22]
links[links[,1]=="chr23", "PlotColor"] <- col_vector[23]
links[links[,1]=="chr24", "PlotColor"] <- col_vector[24]
links[links[,1]=="chr25", "PlotColor"] <- col_vector[25]
links[links[,1]=="chr26", "PlotColor"] <- col_vector[26]
links[links[,1]=="chr27", "PlotColor"] <- col_vector[27]
links[links[,1]=="chr28", "PlotColor"] <- col_vector[28]
links[links[,1]=="chr29", "PlotColor"] <- col_vector[29]

RCircos.Link.Plot(links, track.num = 1)



```


Now, let's check the result with the reads that align only once

```{r}

library(RCircos)
both <- read.csv("both.csv")

RCircos.Get.Plot.Colors(both, col_vector)

both$PlotColor <- col_vector

chr.exclude <- NULL
cyto.info <- both
tracks.inside <- 0
tracks.outside <- 0
RCircos.Set.Core.Components(cyto.info, chr.exclude,tracks.inside, tracks.outside)


rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.params
rcircos.cyto <- RCircos.Get.Plot.Ideogram()
rcircos.cyto
rcircos.position <- RCircos.Get.Plot.Positions()
rcircos.position
RCircos.List.Plot.Parameters()

#rcircos.params <- RCircos.Get.Plot.Parameters()
#rcircos.params$base.per.unit <- 3000
#RCircos.Reset.Plot.Parameters(rcircos.params)
#RCircos.List.Plot.Parameters()


rcircos.cyto
rcircos.cyto$BandColor <- rcircos.cyto$PlotColor
rcircos.cyto$ChrColor <-  rcircos.cyto$PlotColor
RCircos.Reset.Plot.Ideogram(rcircos.cyto)
RCircos.List.Plot.Parameters()


RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()


bowtie2dup <- read.csv("bowtie2dup.csv", sep = "")
bowtie2dup
dim(bowtie2dup)
duplicatedreads <- bowtie2dup$Marker[duplicated(bowtie2dup$Marker)]
noduplicatedreads <- setdiff(bowtie2dup$Marker,duplicatedreads)
noduplicatedreads

bowtie2nodup <-  bowtie2dup[bowtie2dup$Marker %in% noduplicatedreads,]
bowtie2nodup


both

linksnodup <- merge(bowtie2nodup, both, by.x = "Marker", by.y = "Band", sort = F)
linksnodup <- linksnodup[,c(2,3,4,5,6,7)]
linksnodup

RCircos.Link.Plot(linksnodup, track.num, TRUE)






linksnodup$PlotColor <- 1
linksnodup[linksnodup[,1]=="chr1", "PlotColor"] <- col_vector[1]
linksnodup[linksnodup[,1]=="chr2", "PlotColor"] <- col_vector[2]
linksnodup[linksnodup[,1]=="chr3", "PlotColor"] <- col_vector[3]
linksnodup[linksnodup[,1]=="chr4", "PlotColor"] <- col_vector[4]
linksnodup[linksnodup[,1]=="chr5", "PlotColor"] <- col_vector[5]
linksnodup[linksnodup[,1]=="chr6", "PlotColor"] <- col_vector[6]
linksnodup[linksnodup[,1]=="chr7", "PlotColor"] <- col_vector[7]
linksnodup[linksnodup[,1]=="chr8", "PlotColor"] <- col_vector[8]
linksnodup[linksnodup[,1]=="chr9", "PlotColor"] <- col_vector[9]
linksnodup[linksnodup[,1]=="chr10", "PlotColor"] <- col_vector[10]
linksnodup[linksnodup[,1]=="chr11", "PlotColor"] <- col_vector[11]
linksnodup[linksnodup[,1]=="chr12", "PlotColor"] <- col_vector[12]
linksnodup[linksnodup[,1]=="chr13", "PlotColor"] <- col_vector[13]
linksnodup[linksnodup[,1]=="chr14", "PlotColor"] <- col_vector[14]
linksnodup[linksnodup[,1]=="chr15", "PlotColor"] <- col_vector[15]
linksnodup[linksnodup[,1]=="chr16", "PlotColor"] <- col_vector[16]
linksnodup[linksnodup[,1]=="chr17", "PlotColor"] <- col_vector[17]
linksnodup[linksnodup[,1]=="chr18", "PlotColor"] <- col_vector[18]
linksnodup[linksnodup[,1]=="chr19", "PlotColor"] <- col_vector[19]
linksnodup[linksnodup[,1]=="chr20", "PlotColor"] <- col_vector[20]
linksnodup[linksnodup[,1]=="chr21", "PlotColor"] <- col_vector[21]
linksnodup[linksnodup[,1]=="chr22", "PlotColor"] <- col_vector[22]
linksnodup[linksnodup[,1]=="chr23", "PlotColor"] <- col_vector[23]
linksnodup[linksnodup[,1]=="chr24", "PlotColor"] <- col_vector[24]
linksnodup[linksnodup[,1]=="chr25", "PlotColor"] <- col_vector[25]
linksnodup[linksnodup[,1]=="chr26", "PlotColor"] <- col_vector[26]
linksnodup[linksnodup[,1]=="chr27", "PlotColor"] <- col_vector[27]
linksnodup[linksnodup[,1]=="chr28", "PlotColor"] <- col_vector[28]
linksnodup[linksnodup[,1]=="chr29", "PlotColor"] <- col_vector[29]

RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup, track.num = 1)

out.file <- "ComparativeGenomicsLG1nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG1",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG2nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG2",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG3nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG3",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG4nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG4",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG5nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG5",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG6nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG6",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG7nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG7",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG8nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG8",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG9nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG9",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG10nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG10",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG11nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG11",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG12nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG12",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG13nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG13",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG14nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG14",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG15nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG15",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG16nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG16",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG17nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG17",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG18nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG18",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG19nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG19",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG20nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG20",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG21nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG21",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG22nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG22",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG23nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG23",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG24nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG24",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG25nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG25",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG26nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG26",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG27nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG27",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG28nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG28",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG29nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG29",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG30nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG30",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG31nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG31",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicsLG32nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,4]=="LG32",], track.num, TRUE)
dev.off()




out.file <- "ComparativeGenomicschr1nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr1",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr2nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr2",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr3nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr3",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr4nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr4",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr5nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr5",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr6nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr6",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr7nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr7",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr8nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr8",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr9nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr9",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr10nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr10",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr11nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr11",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr12nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr12",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr13nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr13",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr14nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr14",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr15nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr15",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr16nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr16",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr17nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr17",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr18nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr18",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr19nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr19",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr20nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr20",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr21nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr21",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr22nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr22",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr23nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr23",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr24nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr24",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr25nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr25",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr26nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr26",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr27nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr27",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr28nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr28",], track.num, TRUE)
dev.off()

out.file <- "ComparativeGenomicschr29nodup.pdf"
pdf(file=out.file, height=8, width=8, compress=TRUE)
RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()
RCircos.Link.Plot(linksnodup[linksnodup[,1]=="chr29",], track.num, TRUE)
dev.off()


```


Defining blocks!
And compare with BRC

```{r}

BCR <- read.csv("BCR.csv")
#Somewhat similar sequences (blastn), default
BCR2 <- read.csv("BCR2.csv")
#Somewhat similar sequences (blastn), Word size 7, Match/Mitmatch Score 4,-5, 


chr1 <- linksnodup[linksnodup$Chromosome.x=="chr1",]
chr1 <- chr1[order(chr1$chromStart),]
chr1
chr1blocks <- chr1[c(1,2,5,8),c(1,2,3,4)] 
# the markers in 6th, and 7th position are slightly unsorted but we are not going to consider it.
chr1blocks[1,3] <- chr1[2,2]-1
chr1blocks[2,3] <- chr1[4,2]
chr1blocks[3,3] <- chr1[7,2]
chr1blocks[4,3] <- chr1[10,2]
newrow <- c("chr1", 5952569, 8518522, "fisfus")
chr1blocks <- insertRow(chr1blocks, newrow, 3)
newrow <- c("chr1", 13077361, 16671836, "fisfus")
chr1blocks <- insertRow(chr1blocks, newrow, 5)
chr1blocks[1,4]<- "maybefisfus"
chr1blocks[2,4]<- "conserved"
chr1blocks[4,4]<- "conserved"
chr1blocks[6,4]<- "conserved"
chr1blocks


BCR[BCR$Chromosome=="chr1",]
colnames(chr1blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr1blocks$PatternCode <- c(0.5,0,1,0,1,0) 
chr1blocks$CBR <- c(1,0,1,0,2,1)
chr1blocks

chr2 <- linksnodup[linksnodup$Chromosome.x=="chr2",]
chr2 <- chr2[order(chr2$chromStart),]
chr2
chr2blocks <- chr2[c(1,9,10,14),c(1,2,3,4)]
chr2blocks[1,3] <- chr2[8,2]
chr2blocks[2,2] <- chr2[8,2]+1
chr2blocks[2,3] <- chr2[10,2]-1
chr2blocks[3,3] <- chr2[13,2]
chr2blocks[4,2] <- chr2[13,2]+1
chr2blocks[4,3] <- chr2[16,2]
chr2blocks[1,4] <- "conserved"
chr2blocks[2,4] <- "maybefisfus"
chr2blocks[3,4] <- "conserved"
chr2blocks[4,4] <- "unsortedx1&fisfus"
chr2blocks

BCR[BCR$Chromosome=="chr2",]
colnames(chr2blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr2blocks$PatternCode <- c(0,0.5,0,1.5)
chr2blocks$CBR <- c(0,2,1,0)
#there is one BCR just before the first conserved area
chr2blocks

chr3 <- linksnodup[linksnodup$Chromosome.x=="chr3",]
chr3 <- chr3[order(chr3$chromStart),]
chr3
chr3blocks <- chr3[c(1,3,9,10,13),c(1,2,3,4)]
chr3blocks[1,3] <- chr3[3,2]-1
chr3blocks[2,3] <- chr3[8,2]
chr3blocks[3,2] <- chr3[8,2]+1
chr3blocks[3,3] <- chr3[10,2]-1
chr3blocks[4,3] <- chr3[12,2]
chr3blocks[5,3] <- chr3[14,2]
chr3blocks
newrow <- c("chr3", 13772270, 15865085, "fisfus")
chr3blocks <- insertRow(chr3blocks, newrow, 5)
chr3blocks[1,4] <- "fisfus"
chr3blocks[2,4] <- "conserved"
chr3blocks[3,4] <- "fisfus"
chr3blocks[4,4] <- "conserved"
chr3blocks[6,4] <- "maybeconserved"
chr3blocks

BCR[BCR$Chromosome=="chr3",]
colnames(chr3blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr3blocks$PatternCode <- c(1,0,1,0,1,0) 
chr3blocks$CBR <- c(0,0,1,0,1,0)
#there is one BCR just before the first fisfus, single repeat using BCR2 setting but we are not considering it
chr3blocks

chr4 <- linksnodup[linksnodup$Chromosome.x=="chr4",]
chr4 <- chr4[order(chr4$chromStart),]
chr4
chr4blocks <- chr4[c(1,4,7,8,13,17),c(1,2,3,4)]
chr4blocks[1,3] <- chr4[4,2]-1
chr4blocks[2,3] <- chr4[6,2]
chr4blocks[3,2] <- chr4[6,2]+1
chr4blocks[3,3] <- chr4[8,2]-1
chr4blocks[4,3] <- chr4[12,2]
chr4blocks[5,3] <- chr4[16,2]
chr4blocks[6,2] <- chr4[16,2]+1
chr4blocks[6,3] <- chr4[18,2]-1
newrow <- c("chr4", 12453692, 13155061, "fisfus")
chr4blocks <- insertRow(chr4blocks, newrow, 5)
chr4blocks[1,4] <- "maybefisfusx2"
chr4blocks[2,4] <- "conserved"
chr4blocks[3,4] <- "maybefisfus"
chr4blocks[4,4] <- "conserved"
chr4blocks[6,4] <- "conserved"
chr4blocks[7,4] <- "unsorted&maybefisfus"
chr4blocks

BCR[BCR$Chromosome=="chr4",]
colnames(chr4blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr4blocks$PatternCode <- c(1,0,0.5,0,1,0,1) 
chr4blocks$CBR <- c(1,0,0,1,0,0,1)
chr4blocks


chr5 <- linksnodup[linksnodup$Chromosome.x=="chr5",]
chr5 <- chr5[order(chr5$chromStart),]
chr5
chr5blocks <- chr5[c(1,5,7,10),c(1,2,3,4)]
chr5blocks[1,3] <- chr5[4,2]
chr5blocks[2,2] <- chr5[4,2]+1
chr5blocks[2,3] <- chr5[7,2]-1
chr5blocks[3,3] <- chr5[9,2]
chr5blocks[4,2] <- chr5[9,2]+1
chr5blocks[4,3] <- chr5[10,2]
chr5blocks[1,4] <- "conserved"
chr5blocks[2,4] <- "unsorted&fisfus"
chr5blocks[3,4] <- "conserved"
chr5blocks[4,4] <- "unsorted"
chr5blocks

BCR[BCR$Chromosome=="chr5",]
colnames(chr5blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr5blocks$PatternCode <- c(0,1.5,0,0.5) 
chr5blocks$CBR <- c(0,3,0,0)
#there are three CBR areas but both are in the fisfus of the patterns and not in the change order
chr5blocks


chr6 <- linksnodup[linksnodup$Chromosome.x=="chr6",]
chr6 <- chr6[order(chr6$chromStart),]
chr6
chr6blocks <- chr6[c(1,4,7),c(1,2,3,4)]
chr6blocks[1,3] <- chr6[3,2]
chr6blocks[2,2] <- chr6[3,2]+1
chr6blocks[2,3] <- chr6[7,2]-1
chr6blocks[3,3] <- chr6[11,2]
chr6blocks[1,4] <- "conserved"
chr6blocks[2,4] <- "fisfusx3"
chr6blocks[3,4] <- "conserved"
chr6blocks

BCR2[BCR2$Chromosome=="chr6",]
#There are not BCR this chr, but there is one when relaxing parameters
colnames(chr6blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr6blocks$PatternCode <- c(0,3,0) 
chr6blocks$CBR <- c(0,1,0)
chr6blocks


chr7 <- linksnodup[linksnodup$Chromosome.x=="chr7",]
chr7 <- chr7[order(chr7$chromStart),]
chr7
chr7blocks <- chr7[c(1,8,11),c(1,2,3,4)]
chr7blocks[1,3] <- chr7[8,2]-1
chr7blocks[2,3] <- chr7[10,2]
chr7blocks[3,3] <- chr7[13,2]
newrow <- c("chr7", 9609375, 10386170, "fisfus")
chr7blocks <- insertRow(chr7blocks, newrow, 3)
chr7blocks[1,4] <- "unsortedx2&fisfus"
chr7blocks[2,4] <- "conserved"
chr7blocks[4,4] <- "conserved"
chr7blocks

BCR[BCR$Chromosome=="chr7",]
colnames(chr7blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr7blocks$PatternCode <- c(2,0,1,0) 
chr7blocks$CBR <- c(2,0,0,0)
#There are two BCR, but the second one in after and out of the known conserved
chr7blocks


chr8 <- linksnodup[linksnodup$Chromosome.x=="chr8",]
chr8 <- chr8[order(chr8$chromStart),]
chr8
chr8blocks <- chr8[c(1,6,8,10,11),c(1,2,3,4)]
# the markers in 14th, 15th and 16th position are slightly unsorted but we are not going to consider it.
chr8blocks[1,3] <- chr8[5,2]
chr8blocks[2,2] <- chr8[5,2]+1
chr8blocks[2,3] <- chr8[8,2]-1
chr8blocks[3,3] <- chr8[9,2]
chr8blocks[4,2] <- chr8[9,2]+1
chr8blocks[4,3] <- chr8[11,2]-1
chr8blocks[5,3] <- chr8[17,2]
chr8blocks[1,4] <- "conserved"
chr8blocks[2,4] <- "fisfusx2"
chr8blocks[3,4] <- "maybeconserved"
chr8blocks[4,4] <- "maybefisfus"
chr8blocks[5,4] <- "conserved"
chr8blocks

BCR[BCR$Chromosome=="chr8",]
colnames(chr8blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr8blocks$PatternCode <- c(0,2,0,0.5,0) 
chr8blocks$CBR <- c(0,1,0,1,0)
#there are 1 BCR before the 1st conserved area, and one after the last conserved area
#the 1 BCR in the maybefisfus is just one single
chr8blocks

chr9 <- linksnodup[linksnodup$Chromosome.x=="chr9",]
chr9 <- chr9[order(chr9$chromStart),]
chr9
chr9blocks <- chr9[c(1,2,6),c(1,2,3,4)]
# the markers in 7th and 9th position are slightly unsorted but we are not going to consider it. The same for 10th and 11th.
chr9blocks[1,3] <- chr9[2,2]-1
chr9blocks[2,3] <- chr9[5,2]
chr9blocks[3,3] <- chr9[11,2]
newrow <- c("chr9", 5660034, 12014988, "fisfus")
chr9blocks <- insertRow(chr9blocks, newrow, 3)
chr9blocks[1,4] <- "maybefisfus"
chr9blocks[2,4] <- "conserved"
chr9blocks[4,4] <- "conserved"
chr9blocks

BCR[BCR$Chromosome=="chr9",]
BCR2[BCR2$Chromosome=="chr9",]
#There are not BCR this chr
colnames(chr9blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr9blocks$PatternCode <- c(0.5,0,1,0) 
chr9blocks$CBR <- c(0,0,0,0)
chr9blocks
#We are missing something

chr10 <- linksnodup[linksnodup$Chromosome.x=="chr10",]
chr10 <- chr10[order(chr10$chromStart),]
chr10
chr10blocks <- chr10[c(1,13,18,19,23,24),c(1,2,3,4)]
# the markers in 19th and 20th position are slightly unsorted but we are not going to consider it. The same for 21st and 22nd.
chr10blocks[1,3] <- chr10[12,2]
chr10blocks[2,3] <- chr10[17,2]
chr10blocks[3,2] <- chr10[17,2]+1
chr10blocks[3,3] <- chr10[19,2]-1
chr10blocks[4,3] <- chr10[22,2]
chr10blocks[5,2] <- chr10[22,2]+1
chr10blocks[5,3] <- chr10[24,2]-1
chr10blocks[6,3] <- chr10[26,2]
newrow <- c("chr10", 4244125, 5269404, "fisfus")
chr10blocks <- insertRow(chr10blocks, newrow, 2)
chr10blocks[1,4] <- "conserved"
chr10blocks[3,4] <- "conserved"
chr10blocks[4,4] <- "maybefisfus"
chr10blocks[5,4] <- "conserved"
chr10blocks[6,4] <- "maybefisfus"
chr10blocks[7,4] <- "conserved"
chr10blocks

BCR[BCR$Chromosome=="chr10",]
BCR2[BCR2$Chromosome=="chr10",]
#There are not BCR this chr
colnames(chr10blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr10blocks$PatternCode <- c(0,1,0,0.5,0,0.5,0) 
chr10blocks$CBR <- c(0,0,0,0,0,0,0)
chr10blocks


chr11 <- linksnodup[linksnodup$Chromosome.x=="chr11",]
chr11 <- chr11[order(chr11$chromStart),]
chr11
chr11blocks <- chr11[c(1,2,6,9,11),c(1,2,3,4)]
# the markers in 18th and 19th position are slightly unsorted but we are not going to consider it.
chr11blocks[1,3] <- chr11[2,2]-1
chr11blocks[2,3] <- chr11[5,2]
chr11blocks[3,3] <- chr11[8,2]
chr11blocks[4,3] <- chr11[10,2]
chr11blocks[5,3] <- chr11[23,2]
newrow <- c("chr11", 2606506, 3836219, "fisfus")
chr11blocks <- insertRow(chr11blocks, newrow, 3)
newrow <- c("chr11", 5751244, 6162157, "maybefisfus")
chr11blocks <- insertRow(chr11blocks, newrow, 5)
newrow <- c("chr11", 7481618, 10687339, "maybefisfus")
chr11blocks <- insertRow(chr11blocks, newrow, 7)
chr11blocks[1,4] <- "maybefisfus"
chr11blocks[2,4] <- "conserved"
chr11blocks[4,4] <- "conserved"
chr11blocks[6,4] <- "maybeconserved"
chr11blocks[8,4] <- "conserved"
chr11blocks

BCR[BCR$Chromosome=="chr11",]
colnames(chr11blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr11blocks$PatternCode <- c(0.5,0,1,0,0.5,0,0.5,0) 
chr11blocks$CBR <- c(1,1,0,0,0,0,0,0)
#the first BCR is in the area of some small unsorting that we have not considered
#There are two BCR, but the second one in after and out of the known conserved
chr11blocks


chr12 <- linksnodup[linksnodup$Chromosome.x=="chr12",]
chr12 <- chr12[order(chr12$chromStart),]
chr12
chr12blocks <- chr12[c(1,3,6,12,19,21),c(1,2,3,4)] 
# the markers in 16th and 17th position are slightly unsorted but we are not going to consider it.
chr12blocks[1,3] <- chr12[2,2]
chr12blocks[2,2] <- chr12[2,2]+1
chr12blocks[2,3] <- chr12[6,2]-1
chr12blocks[3,3] <- chr12[11,2]
chr12blocks[4,3] <- chr12[17,2]
chr12blocks[5,3] <- chr12[20,2]
chr12blocks[6,3] <- chr12[23,2]
newrow <- c("chr12", 6078481, 7374679, "fisfus")
chr12blocks <- insertRow(chr12blocks, newrow, 4)
newrow <- c("chr12", 9012769, 10409253, "maybefisfus")
chr12blocks <- insertRow(chr12blocks, newrow, 6)
newrow <- c("chr12", 10917073, 11405499, "maybefisfus")
chr12blocks <- insertRow(chr12blocks, newrow, 8)
chr12blocks[1,4] <- "maybeconserved"
chr12blocks[2,4] <- "maybefisfusx2"
chr12blocks[3,4] <- "conserved"
chr12blocks[5,4] <- "conserved"
chr12blocks[7,4] <- "maybeconserved"
chr12blocks[9,4] <- "conserved"
chr12blocks

BCR[BCR$Chromosome=="chr12",]
BCR2[BCR2$Chromosome=="chr12",]
colnames(chr12blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr12blocks$PatternCode <- c(0,1,0,1,0,0.5,0,0.5,0) 
chr12blocks$CBR <- c(0,0,0,0,0,0,0,0,0)
#the first BCR in after and out of the known conserved, the same with BCR2 settings
chr12blocks
#We are missing something

chr13 <- linksnodup[linksnodup$Chromosome.x=="chr13",]
chr13 <- chr13[order(chr13$chromStart),]
chr13
chr13blocks <- chr13[c(1,8,13,19),c(1,2,3,4)] 
chr13blocks[1,3] <- chr13[8,2]-1
chr13blocks[2,3] <- chr13[12,2]
chr13blocks[3,3] <- chr13[18,2]
chr13blocks[4,2] <- chr13[18,2]+1
chr13blocks[4,3] <- chr13[21,2]
newrow <- c("chr13", 9428904, 10457675, "fisfus")
chr13blocks <- insertRow(chr13blocks, newrow, 3)
chr13blocks[1,4] <- "unsortedx3&mayfisfus"
chr13blocks[2,4] <- "conserved"
chr13blocks[4,4] <- "conserved"
chr13blocks[5,4] <- "fisfus&unsorted"
chr13blocks

BCR[BCR$Chromosome=="chr13",]
colnames(chr13blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr13blocks$PatternCode <- c(2.5,0,1,0,1.5)
chr13blocks$CBR <- c(1,0,0,0,0)
chr13blocks


chr14 <- linksnodup[linksnodup$Chromosome.x=="chr14",]
chr14 <- chr14[order(chr14$chromStart),]
chr14
# the markers in 12th and 13th position are slightly unsorted but we are not going to consider it. The same for the markers in the positions 15th and 16th.
chr14blocks <- chr14[c(1,3,8,15,22),c(1,2,3,4)] 
chr14blocks[1,3]<-chr14[3,2]-1
chr14blocks[2,3]<-chr14[7,2]
chr14blocks[3,3]<-chr14[14,2]
chr14blocks[4,3]<-chr14[21,2]
chr14blocks[5,2]<-chr14[21,2]+1
chr14blocks[5,3]<-chr14[25,2]
newrow <- c("chr14", 5018142, 5279697, "fisfus")
chr14blocks <- insertRow(chr14blocks, newrow, 3)
newrow <- c("chr14", 8444297, 10345369, "fisfus")
chr14blocks  <- insertRow(chr14blocks, newrow, 5)
chr14blocks[1,4] <- "maybefisfusx2"
chr14blocks[2,4] <- "conserved"
chr14blocks[4,4] <- "conserved"
chr14blocks[6,4] <- "conserved"
chr14blocks[7,4] <- "maybefisfusx2&unsorted"
chr14blocks

BCR[BCR$Chromosome=="chr14",]
colnames(chr14blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr14blocks$PatternCode <- c(1,0,1,0,1,0,1.5)
chr14blocks$CBR <- c(1,0,0,1,1,0,0)
#the BCR is before the first maybefisfusx2
chr14blocks

chr15 <- linksnodup[linksnodup$Chromosome.x=="chr15",]
chr15 <- chr15[order(chr15$chromStart),]
chr15
# the markers in third and fourth position are slightly unsorted but we are not going to consider it. The same for the markers in the positions 9th and 10th.
chr15blocks <- chr15[c(1,6,12,13,16),c(1,2,3,4)] 
chr15blocks[1,3] <- chr15[5,2]
chr15blocks[2,3] <- chr15[11,2]
chr15blocks[3,2] <- chr15[11,2]+1
chr15blocks[3,3] <- chr15[13,2]-1
chr15blocks[4,3] <- chr15[15,2]
chr15blocks[5,2] <- chr15[15,2]+1
chr15blocks[5,3] <- chr15[16,2]
newrow <- c("chr15", 4376634, 8917572, "fisfus")
chr15blocks <- insertRow(chr15blocks, newrow, 2)
chr15blocks[1,4] <- "conserved"
chr15blocks[3,4] <- "conserved"
chr15blocks[4,4] <- "maybefisfus"
chr15blocks[5,4] <- "conserved"
chr15blocks[6,4] <- "maybefisfus"
chr15blocks

#There are not BCR this chr
BCR[BCR$Chromosome=="chr15",]
BCR2[BCR2$Chromosome=="chr15",]
colnames(chr15blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr15blocks$PatternCode <- c(0,1,0,0.5,0,0.5)
chr15blocks$CBR <- c(0,0,0,0,0,0)
chr15blocks


chr16 <- linksnodup[linksnodup$Chromosome.x=="chr16",]
chr16 <- chr16[order(chr16$chromStart),]
chr16
# the markers in sixth and seventh position are slightly unsorted but we are not going to consider it
chr16blocks <- chr16[c(1,9,10,13),c(1,2,3,4)] 
chr16blocks[1,3] <- chr16[8,2]
chr16blocks[2,2] <- chr16[8,2]+1
chr16blocks[2,3] <- chr16[10,2]-1
chr16blocks[3,3] <- chr16[12,2]
chr16blocks[4,2] <- chr16[12,2]+1
chr16blocks[4,3] <- chr16[14,2]
chr16blocks[1,4] <- "conserved"
chr16blocks[2,4] <- "maybefisfus"
chr16blocks[3,4] <- "conserved"
chr16blocks[4,4] <- "maybefisfusx2"
chr16blocks

BCR[BCR$Chromosome=="chr16",]
colnames(chr16blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr16blocks$PatternCode <- c(0,0.5,0,1)
chr16blocks$CBR <- c(0,1,1,1)
chr16blocks


chr17 <- linksnodup[linksnodup$Chromosome.x=="chr17",]
chr17 <- chr17[order(chr17$chromStart),]
chr17
chr17blocks <- chr17[c(1,7,10,11,13),c(1,2,3,4)] 
chr17blocks[1,3]<-chr17[7,2]-1
chr17blocks[2,3]<-chr17[9,2]
chr17blocks[3,2]<-chr17[9,2]+1
chr17blocks[3,3]<-chr17[11,2]-1
chr17blocks[4,3]<-chr17[12,2]
chr17blocks[5,2]<-chr17[12,2]+1
chr17blocks[5,3]<-chr17[14,2]
chr17blocks[1,4]<-"maybefisfus&unsortedx2"
chr17blocks[2,4]<-"conserved"
chr17blocks[3,4]<-"unsorted&fisfus&unsorted"
chr17blocks[4,4]<-"maybeconserved"
chr17blocks[5,4]<-"fisfusx1.5"
chr17blocks
#the maybeconserved could be problably gone

BCR[BCR$Chromosome=="chr17",]
colnames(chr17blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr17blocks$PatternCode <- c(1.5,0,2,0,1.5)
chr17blocks$CBR <- c(0,0,0,0,2)
#The second BCR is after the whole thing
chr17blocks


chr18 <- linksnodup[linksnodup$Chromosome.x=="chr18",]
chr18 <- chr18[order(chr18$chromStart),]
chr18
chr18blocks <- chr18[c(1,3,5,8,9,13),c(1,2,3,4)] 
chr18blocks[1,3] <- chr18[2,2]
chr18blocks[2,2] <- chr18[2,2]+1
chr18blocks[2,3] <- chr18[5,2]-1
chr18blocks[3,3] <- chr18[7,2]
chr18blocks[4,2] <- chr18[7,2]+1
chr18blocks[4,3] <- chr18[9,2]-1
chr18blocks[5,3] <- chr18[12,2]
chr18blocks[6,3] <- chr18[20,2]
newrow <- c("chr18", 6038837, 6531250, "fisfus")
chr18blocks <- insertRow(chr18blocks, newrow, 6)
chr18blocks[1,4] <- "maybeconserved"
chr18blocks[2,4] <- "maybefisfus&unsorted"
chr18blocks[3,4] <- "conserved"
chr18blocks[4,4] <- "unsorted&maybefisfus"
chr18blocks[5,4] <- "conserved"
chr18blocks[7,4] <- "conserved"
chr18blocks

BCR[BCR$Chromosome=="chr18",]
colnames(chr18blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr18blocks$PatternCode <- c(0,1,0,1,0,1,0)
chr18blocks$CBR <- c(0,0,0,0,0,0,1)
#There is one BCR before everything
chr18blocks

chr19 <- linksnodup[linksnodup$Chromosome.x=="chr19",]
chr19 <- chr19[order(chr19$chromStart),]
chr19
chr19blocks <- chr19[c(1,4,9,12),c(1,2,3,4)] 
chr19blocks[1,3] <- chr19[3,2]
chr19blocks[2,2] <- chr19[3,2] + 1
chr19blocks[2,3] <- chr19[9,2] -1
chr19blocks[3,3] <- chr19[11,2]
chr19blocks[4,3] <- chr19[15,2]
newrow <- c("chr19", 7006970, 8323676, "fisfus")
chr19blocks <- insertRow(chr19blocks, newrow, 4)
chr19blocks[1,4] <- "conserved"
chr19blocks[2,4] <- "maybefisfus&unsortedx2"
chr19blocks[3,4] <- "conserved"
chr19blocks[5,4] <- "conserved"
chr19blocks

#There are not BCR this chr
BCR[BCR$Chromosome=="chr19",]
BCR2[BCR2$Chromosome=="chr19",]
colnames(chr19blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr19blocks$PatternCode <- c(0,1.5,0,1,0)
chr19blocks$CBR <- c(0,0,0,0,0)
chr19blocks

chr20 <- linksnodup[linksnodup$Chromosome.x=="chr20",]
chr20 <- chr20[order(chr20$chromStart),]
chr20
chr20blocks <- chr20[c(1),c(1,2,3,4)] 
chr20blocks[1,3] <- chr20[5,2]
chr20blocks[1,4] <- "unsorted&maybefisfus"
chr20blocks

#There are not BCR this chr
BCR[BCR$Chromosome=="chr20",]
BCR2[BCR2$Chromosome=="chr20",]
colnames(chr20blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr20blocks$PatternCode <- c(1)
chr20blocks$CBR <- c(0)
chr20blocks

chr21 <- linksnodup[linksnodup$Chromosome.x=="chr21",]
chr21 <- chr21[order(chr21$chromStart),]
chr21
chr21blocks <- chr21[c(1,3,5),c(1,2,3,4)] 
chr21blocks[1,3] <- chr21[3,2]-1 
chr21blocks[2,3] <- chr21[4,2]
chr21blocks[3,3] <- chr21[6,2]
newrow <- c("chr21", 5440874, 6958720, "maybefisfus")
chr21blocks <- insertRow(chr21blocks, newrow, 3)
chr21blocks[1,4] <- "maybefisfusx2"
chr21blocks[2,4] <- "maybeconserved"
chr21blocks[4,4] <- "maybeconserved"
chr21blocks

#There are not BCR this chr
BCR[BCR$Chromosome=="chr21",]
BCR2[BCR2$Chromosome=="chr21",]
colnames(chr21blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr21blocks$PatternCode <- c(1, 0, 0.5, 0)
chr21blocks$CBR <- c(0,0,0,0)
chr21blocks


chr22 <- linksnodup[linksnodup$Chromosome.x=="chr22",]
chr22 <- chr22[order(chr22$chromStart),]
chr22
chr22blocks <- chr22[c(1,4,9,10,13),c(1,2,3,4)] 
chr22blocks[1,3] <- chr22[4,2] - 1
chr22blocks[2,3] <- chr22[8,2]
chr22blocks[3,2] <- chr22[8,2] + 1
chr22blocks[3,3] <- chr22[10,2] - 1
chr22blocks[4,3] <- chr22[12,2]
chr22blocks[5,2] <- chr22[12,2] + 1
chr22blocks[5,3] <- chr22[17,2]
chr22blocks[1,4] <- "unsorted&maybefisfus"
chr22blocks[2,4] <- "conserved"
chr22blocks[3,4] <- "unsorted&fisfus"
chr22blocks[4,4] <- "conserved"
chr22blocks[5,4] <- "fisfusx1.5&unsorted"
chr22blocks

BCR[BCR$Chromosome=="chr22",]
colnames(chr22blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr22blocks$PatternCode <- c(1, 0, 1.5, 0,2)
chr22blocks$CBR <- c(0,0,0,0,1)
chr22blocks


chr23 <- linksnodup[linksnodup$Chromosome.x=="chr23",]
chr23 <- chr23[order(chr23$chromStart),]
chr23
chr23blocks <- chr23[c(1,5,10,16),c(1,2,3,4)] 
chr23blocks[1,3] <- chr23[5,2] - 1 
chr23blocks[2,3] <- chr23[9,2]
chr23blocks[3,3] <- chr23[15,2]
chr23blocks[4,2] <- chr23[15,2]+1
chr23blocks[4,3] <- chr23[19,2]
newrow <- c("chr23", 5567451, 5942385, "fisfus")
chr23blocks <- insertRow(chr23blocks, newrow, 3)
chr23blocks[1,4] <- "unsortedx2&fisfus"
chr23blocks[2,4] <- "conserved"
chr23blocks[4,4] <- "conserved"
chr23blocks[5,4] <- "unsorted&maybefisfus"
chr23blocks

#There are not BCR this chr
BCR[BCR$Chromosome=="chr23",]
BCR2[BCR2$Chromosome=="chr23",]
colnames(chr23blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr23blocks$PatternCode <- c(2, 0, 1, 0,1)
chr23blocks$CBR <- c(0,0,0,1,0)
chr23blocks


chr24 <- linksnodup[linksnodup$Chromosome.x=="chr24",]
chr24 <- chr24[order(chr24$chromStart),]
chr24
chr24blocks <- chr24[c(1,8),c(1,2,3,4)] 
chr24blocks
chr24blocks[1,3] <- chr24[8,2]-1
chr24blocks[2,3] <- chr24[18,2]
chr24blocks[1,4] <- "unsortedx2&fisfus"
chr24blocks[2,4] <- "conserved"
chr24blocks

BCR[BCR$Chromosome=="chr24",]
colnames(chr24blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr24blocks$PatternCode <- c(2, 0)
chr24blocks$CBR <- c(1,0)
#There are only 2 single repeat at the beginning, before everything
chr24blocks

chr25 <- linksnodup[linksnodup$Chromosome.x=="chr25",]
chr25 <- chr25[order(chr25$chromStart),]
chr25
chr25blocks <- chr25[c(1,3,7,9),c(1,2,3,4)] 
chr25blocks[1,3] <- chr25[2,3]
chr25blocks[2,3] <- chr25[6,3]
chr25blocks[3,2] <- chr25[6,3] + 1
chr25blocks[3,3] <- chr25[9,2] - 1
chr25blocks[4,3] <- chr25[15,3]
newrow <- c("chr25", 325961, 1321736, "fisfus")
#insertRow <- function(existingDF, newrow, r) {
#  existingDF[seq(r+1,nrow(existingDF)+1),] <- existingDF[seq(r,nrow(existingDF)),]
#  existingDF[r,] <- newrow
#  existingDF
#}
chr25blocks <- insertRow(chr25blocks, newrow, 2)
chr25blocks[1,4] <- "maybeconserved"
chr25blocks[3,4] <- "conserved"
chr25blocks[4,4] <- "maybefisfus&fisfus"
chr25blocks[5,4] <- "conserved"
chr25blocks
#2 conserved block and 3rd possible, and two rearrangementes between then

#There are not BCR this chr
BCR[BCR$Chromosome=="chr25",]
colnames(chr25blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr25blocks$PatternCode <- c(0, 1, 0, 1.5,0)
#we are going to remove the maybeconserved
chr25blocks$CBR <- c(0,0,0,0,0)
chr25blocks


chr26 <- linksnodup[linksnodup$Chromosome.x=="chr26",]
chr26 <- chr26[order(chr26$chromStart),]
chr26
chr26blocks <- chr26[c(1,5),c(1,2,3,4)] 
chr26blocks[1,3] <- chr26[4,2]
chr26blocks[2,2] <- chr26[4,2] + 1
chr26blocks[2,3] <- chr26[8,2]
chr26blocks[2,4] <- "fisfus&maybefisfusx2"
chr26blocks[1,4] <- "conserved"
chr26blocks
#single conserved block and area with possible multiple rearrangements

#There are not BCR this chr
BCR[BCR$Chromosome=="chr26",]
colnames(chr26blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr26blocks$PatternCode <- c(0, 2)
chr26blocks$CBR <- c(0,0)
chr26blocks



chr27 <- linksnodup[linksnodup$Chromosome.x=="chr27",]
chr27 <- chr27[order(chr27$chromStart),]
chr27
chr27blocks <- chr27[c(1,4),c(1,2,3,4)] 
chr27blocks[1,3] <- chr27[3,2]
chr27blocks[1,4] <- "conserved"
chr27blocks[2,3] <- chr27[6,2]
chr27blocks[2,4] <- "conserved"
chr27blocks <- rbind(chr27blocks[1,],c("chr27", 3696901,4879022,"fisfus"),chr27blocks[2,])
chr27blocks
#two conserved blocks detected, 1 rearrangement between both

#There are not BCR this chr
BCR[BCR$Chromosome=="chr27",]
colnames(chr27blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr27blocks$PatternCode <- c(0, 1, 0)
chr27blocks$CBR <- c(0,0,0)
chr27blocks

chr28 <- linksnodup[linksnodup$Chromosome.x=="chr28",]
chr28 <- chr28[order(chr28$chromStart),]
chr28
chr28blocks <- chr28[c(1,5),c(1,2,3,4)] 
chr28blocks[1,3] <- chr28[4,2]
chr28blocks[2,2] <- chr28[4,2] + 1
chr28blocks[2,3] <- chr28[6,2]
chr28blocks[1,4] <- "conserved"
chr28blocks[2,4] <- "unsorted"
chr28blocks
#single conserved block, a little unsorting

#There are not BCR this chr
BCR[BCR$Chromosome=="chr28",]
colnames(chr28blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr28blocks$PatternCode <- c(0, 0.5)
chr28blocks$CBR <- c(0,0)
chr28blocks

chr29 <- linksnodup[linksnodup$Chromosome.x=="chr29",]
chr29 <- chr29[order(chr29$chromStart),]
chr29
chr29blocks <- chr29[c(1),c(1,2,3,4)] 
chr29blocks[1,3] <- chr29[3,2]
chr29blocks[1,4] <- "conserved"
chr29blocks
#single conserved block

#There are not BCR this chr
BCR[BCR$Chromosome=="chr29",]
colnames(chr29blocks)<- c("Chromosome", "Start", "End", "Pattern")
chr29blocks$PatternCode <- c(0)
chr29blocks$CBR <- c(0)
chr29blocks



```

Let's make this more authomatic

```{r}

source("genomeblocks.R")

outblock1 <- genomeblocks(chr1)
outblock1
outblock2 <- genomeblocks(chr2)
outblock2
outblock3 <- genomeblocks(chr3)
outblock3
outblock4 <- genomeblocks(chr4)
outblock4
outblock5 <- genomeblocks(chr5)
outblock5
outblock6 <- genomeblocks(chr6)
outblock6
outblock7 <- genomeblocks(chr7)
outblock7
outblock8 <- genomeblocks(chr8)
outblock8
outblock9 <- genomeblocks(chr9)
outblock9
outblock10 <- genomeblocks(chr10)
outblock10
outblock11 <- genomeblocks(chr11)
outblock11
outblock12 <- genomeblocks(chr12)
outblock12
outblock13 <- genomeblocks(chr13)
outblock13
outblock14 <- genomeblocks(chr14)
outblock14
outblock15 <- genomeblocks(chr15)
outblock15
outblock16 <- genomeblocks(chr16)
outblock16
outblock17 <- genomeblocks(chr17)
outblock17
outblock18 <- genomeblocks(chr18)
outblock18
outblock19 <- genomeblocks(chr19)
outblock19
outblock20 <- genomeblocks(chr20)
outblock20
outblock21 <- genomeblocks(chr21)
outblock21
outblock22 <- genomeblocks(chr22)
outblock22
outblock23 <- genomeblocks(chr23)
outblock23
outblock24 <- genomeblocks(chr24)
outblock24
outblock25 <- genomeblocks(chr25)
outblock25
outblock26 <- genomeblocks(chr26)
outblock26
outblock27 <- genomeblocks(chr27)
outblock27
outblock28 <- genomeblocks(chr28)
outblock28
outblock29 <- genomeblocks(chr29)
outblock29






```

Let's search for BCR in the blocks and interblocks
```{r}
BCR <- read.csv("BCR.csv")
#Somewhat similar sequences (blastn), default

BCR1 <- BCR[BCR$Chromosome=="chr1",]
BCR1
BCR2 <- BCR[BCR$Chromosome=="chr2",]
BCR2
BCR3 <- BCR[BCR$Chromosome=="chr3",]
BCR3
BCR4 <- BCR[BCR$Chromosome=="chr4",]
BCR4
BCR5 <- BCR[BCR$Chromosome=="chr5",]
BCR5
BCR6 <- BCR[BCR$Chromosome=="chr6",]
BCR6
BCR7 <- BCR[BCR$Chromosome=="chr7",]
BCR7
BCR8 <- BCR[BCR$Chromosome=="chr8",]
BCR8
BCR9 <- BCR[BCR$Chromosome=="chr9",]
BCR9
BCR10 <- BCR[BCR$Chromosome=="chr10",]
BCR10
BCR11 <- BCR[BCR$Chromosome=="chr11",]
BCR11
BCR12 <- BCR[BCR$Chromosome=="chr12",]
BCR12
BCR13 <- BCR[BCR$Chromosome=="chr13",]
BCR13
BCR14 <- BCR[BCR$Chromosome=="chr14",]
BCR14
BCR15 <- BCR[BCR$Chromosome=="chr15",]
BCR15
BCR16 <- BCR[BCR$Chromosome=="chr16",]
BCR16
BCR17 <- BCR[BCR$Chromosome=="chr17",]
BCR17
BCR18 <- BCR[BCR$Chromosome=="chr18",]
BCR18
BCR19 <- BCR[BCR$Chromosome=="chr19",]
BCR19
BCR20 <- BCR[BCR$Chromosome=="chr20",]
BCR20
BCR21 <- BCR[BCR$Chromosome=="chr21",]
BCR21
BCR22 <- BCR[BCR$Chromosome=="chr22",]
BCR22
BCR23 <- BCR[BCR$Chromosome=="chr23",]
BCR23
BCR24 <- BCR[BCR$Chromosome=="chr24",]
BCR24
BCR25 <- BCR[BCR$Chromosome=="chr25",]
BCR25
BCR26 <- BCR[BCR$Chromosome=="chr26",]
BCR26
BCR27 <- BCR[BCR$Chromosome=="chr27",]
BCR27
BCR28 <- BCR[BCR$Chromosome=="chr28",]
BCR28
BCR29 <- BCR[BCR$Chromosome=="chr29",]
BCR29

outblock1

source("genomeblocksBCR.R") 

genomeblocksBCR1 <- genomeblocksBCR(outblock1, BCR1)
genomeblocksBCR1
genomeblocksBCR2 <- genomeblocksBCR(outblock2, BCR2)
genomeblocksBCR2
genomeblocksBCR3 <- genomeblocksBCR(outblock3, BCR3)
genomeblocksBCR3
genomeblocksBCR4 <- genomeblocksBCR(outblock4, BCR4)
genomeblocksBCR4
genomeblocksBCR5 <- genomeblocksBCR(outblock5, BCR5)
genomeblocksBCR5
genomeblocksBCR6 <- genomeblocksBCR(outblock6, BCR6)
genomeblocksBCR6
genomeblocksBCR7 <- genomeblocksBCR(outblock7, BCR7)
genomeblocksBCR7
genomeblocksBCR8 <- genomeblocksBCR(outblock8, BCR8)
genomeblocksBCR8
genomeblocksBCR9 <- genomeblocksBCR(outblock9, BCR9)
genomeblocksBCR9
genomeblocksBCR10 <- genomeblocksBCR(outblock10, BCR10)
genomeblocksBCR10
genomeblocksBCR11 <- genomeblocksBCR(outblock11, BCR11)
genomeblocksBCR11
genomeblocksBCR12 <- genomeblocksBCR(outblock12, BCR12)
genomeblocksBCR12
genomeblocksBCR13 <- genomeblocksBCR(outblock13, BCR13)
genomeblocksBCR13
genomeblocksBCR14 <- genomeblocksBCR(outblock14, BCR14)
genomeblocksBCR14
genomeblocksBCR15 <- genomeblocksBCR(outblock15, BCR15)
genomeblocksBCR15
genomeblocksBCR16 <- genomeblocksBCR(outblock16, BCR16)
genomeblocksBCR16
genomeblocksBCR17 <- genomeblocksBCR(outblock17, BCR17)
genomeblocksBCR17
genomeblocksBCR18 <- genomeblocksBCR(outblock18, BCR18)
genomeblocksBCR18
genomeblocksBCR19 <- genomeblocksBCR(outblock19, BCR19)
genomeblocksBCR19
genomeblocksBCR20 <- genomeblocksBCR(outblock20, BCR20)
genomeblocksBCR20
genomeblocksBCR21 <- genomeblocksBCR(outblock21, BCR21)
genomeblocksBCR21
genomeblocksBCR22 <- genomeblocksBCR(outblock22, BCR22)
genomeblocksBCR22
genomeblocksBCR23 <- genomeblocksBCR(outblock23, BCR23)
genomeblocksBCR23
genomeblocksBCR24 <- genomeblocksBCR(outblock24, BCR24)
genomeblocksBCR24
genomeblocksBCR25 <- genomeblocksBCR(outblock25, BCR25)
genomeblocksBCR25
genomeblocksBCR26 <- genomeblocksBCR(outblock26, BCR26)
genomeblocksBCR26
genomeblocksBCR27 <- genomeblocksBCR(outblock27, BCR27)
genomeblocksBCR27
genomeblocksBCR28 <- genomeblocksBCR(outblock28, BCR28)
genomeblocksBCR28
genomeblocksBCR29 <- genomeblocksBCR(outblock29, BCR29)
genomeblocksBCR29

genomeblocksBCRall <- rbind(genomeblocksBCR1,genomeblocksBCR2,genomeblocksBCR3,genomeblocksBCR4,genomeblocksBCR5,genomeblocksBCR6,genomeblocksBCR7,genomeblocksBCR8,genomeblocksBCR9,genomeblocksBCR10,genomeblocksBCR11,genomeblocksBCR12,genomeblocksBCR13,genomeblocksBCR14,genomeblocksBCR15,genomeblocksBCR16,genomeblocksBCR17,genomeblocksBCR18,genomeblocksBCR19,genomeblocksBCR20,genomeblocksBCR21,genomeblocksBCR22,genomeblocksBCR23,genomeblocksBCR24,genomeblocksBCR25,genomeblocksBCR26,genomeblocksBCR27,genomeblocksBCR28,genomeblocksBCR29)

genomeblocksBCRall
boxplot(as.numeric(genomeblocksBCRall[,"BCR"]) ~ genomeblocksBCRall[,"type"])

anova(lm(as.numeric(genomeblocksBCRall[,"BCR"]) ~ as.character(genomeblocksBCRall[,"type"])))
summary(aov(as.numeric(genomeblocksBCRall[,"BCR"]) ~ as.character(genomeblocksBCRall[,"type"])))


#MORE RELAXED SEARCH FOR BCR
BCR2 <- read.csv("BCR2.csv")
#Somewhat similar sequences (blastn), Word size 7, Match/Mitmatch Score 4,-5, 

BCR2_1 <- BCR2[BCR2$Chromosome=="chr1",]
BCR2_1
BCR2_2 <- BCR2[BCR2$Chromosome=="chr2",]
BCR2_2
BCR2_3 <- BCR2[BCR2$Chromosome=="chr3",]
BCR2_3
BCR2_4 <- BCR2[BCR2$Chromosome=="chr4",]
BCR2_4
BCR2_5 <- BCR2[BCR2$Chromosome=="chr5",]
BCR2_5
BCR2_6 <- BCR2[BCR2$Chromosome=="chr6",]
BCR2_6
BCR2_7 <- BCR2[BCR2$Chromosome=="chr7",]
BCR2_7
BCR2_8 <- BCR2[BCR2$Chromosome=="chr8",]
BCR2_8
BCR2_9 <- BCR2[BCR2$Chromosome=="chr9",]
BCR2_9
BCR2_10 <- BCR2[BCR2$Chromosome=="chr10",]
BCR2_10
BCR2_11 <- BCR2[BCR2$Chromosome=="chr11",]
BCR2_11
BCR2_12 <- BCR2[BCR2$Chromosome=="chr12",]
BCR2_12
BCR2_13 <- BCR2[BCR2$Chromosome=="chr13",]
BCR2_13
BCR2_14 <- BCR2[BCR2$Chromosome=="chr14",]
BCR2_14
BCR2_15 <- BCR2[BCR2$Chromosome=="chr15",]
BCR2_15
BCR2_16 <- BCR2[BCR2$Chromosome=="chr16",]
BCR2_16
BCR2_17 <- BCR2[BCR2$Chromosome=="chr17",]
BCR2_17
BCR2_18 <- BCR2[BCR2$Chromosome=="chr18",]
BCR2_18
BCR2_19 <- BCR2[BCR2$Chromosome=="chr19",]
BCR2_19
BCR2_20 <- BCR2[BCR2$Chromosome=="chr20",]
BCR2_20
BCR2_21 <- BCR2[BCR2$Chromosome=="chr21",]
BCR2_21
BCR2_22 <- BCR2[BCR2$Chromosome=="chr22",]
BCR2_22
BCR2_23 <- BCR2[BCR2$Chromosome=="chr23",]
BCR2_23
BCR2_24 <- BCR2[BCR2$Chromosome=="chr24",]
BCR2_24
BCR2_25 <- BCR2[BCR2$Chromosome=="chr25",]
BCR2_25
BCR2_26 <- BCR2[BCR2$Chromosome=="chr26",]
BCR2_26
BCR2_27 <- BCR2[BCR2$Chromosome=="chr27",]
BCR2_27
BCR2_28 <- BCR2[BCR2$Chromosome=="chr28",]
BCR2_28
BCR2_29 <- BCR2[BCR2$Chromosome=="chr29",]
BCR2_29


genomeblocksBCR2_1 <- genomeblocksBCR(outblock1, BCR2_1)
genomeblocksBCR2_1
genomeblocksBCR2_2 <- genomeblocksBCR(outblock2, BCR2_2)
genomeblocksBCR2_2
genomeblocksBCR2_3 <- genomeblocksBCR(outblock3, BCR2_3)
genomeblocksBCR2_3
genomeblocksBCR2_4 <- genomeblocksBCR(outblock4, BCR2_4)
genomeblocksBCR2_4
genomeblocksBCR2_5 <- genomeblocksBCR(outblock5, BCR2_5)
genomeblocksBCR2_5
genomeblocksBCR2_6 <- genomeblocksBCR(outblock6, BCR2_6)
genomeblocksBCR2_6
genomeblocksBCR2_7 <- genomeblocksBCR(outblock7, BCR2_7)
genomeblocksBCR2_7
genomeblocksBCR2_8 <- genomeblocksBCR(outblock8, BCR2_8)
genomeblocksBCR2_8
genomeblocksBCR2_9 <- genomeblocksBCR(outblock9, BCR2_9)
genomeblocksBCR2_9
genomeblocksBCR2_10 <- genomeblocksBCR(outblock10, BCR2_10)
genomeblocksBCR2_10
genomeblocksBCR2_11 <- genomeblocksBCR(outblock11, BCR2_11)
genomeblocksBCR2_11
genomeblocksBCR2_12 <- genomeblocksBCR(outblock12, BCR2_12)
genomeblocksBCR2_12
genomeblocksBCR2_13 <- genomeblocksBCR(outblock13, BCR2_13)
genomeblocksBCR2_13
genomeblocksBCR2_14 <- genomeblocksBCR(outblock14, BCR2_14)
genomeblocksBCR2_14
genomeblocksBCR2_15 <- genomeblocksBCR(outblock15, BCR2_15)
genomeblocksBCR2_15
genomeblocksBCR2_16 <- genomeblocksBCR(outblock16, BCR2_16)
genomeblocksBCR2_16
genomeblocksBCR2_17 <- genomeblocksBCR(outblock17, BCR2_17)
genomeblocksBCR2_17
genomeblocksBCR2_18 <- genomeblocksBCR(outblock18, BCR2_18)
genomeblocksBCR2_18
genomeblocksBCR2_19 <- genomeblocksBCR(outblock19, BCR2_19)
genomeblocksBCR2_19
genomeblocksBCR2_20 <- genomeblocksBCR(outblock20, BCR2_20)
genomeblocksBCR2_20
genomeblocksBCR2_21 <- genomeblocksBCR(outblock21, BCR2_21)
genomeblocksBCR2_21
genomeblocksBCR2_22 <- genomeblocksBCR(outblock22, BCR2_22)
genomeblocksBCR2_22
genomeblocksBCR2_23 <- genomeblocksBCR(outblock23, BCR2_23)
genomeblocksBCR2_23
genomeblocksBCR2_24 <- genomeblocksBCR(outblock24, BCR2_24)
genomeblocksBCR2_24
genomeblocksBCR2_25 <- genomeblocksBCR(outblock25, BCR2_25)
genomeblocksBCR2_25
genomeblocksBCR2_26 <- genomeblocksBCR(outblock26, BCR2_26)
genomeblocksBCR2_26
genomeblocksBCR2_27 <- genomeblocksBCR(outblock27, BCR2_27)
genomeblocksBCR2_27
genomeblocksBCR2_28 <- genomeblocksBCR(outblock28, BCR2_28)
genomeblocksBCR2_28
genomeblocksBCR2_29 <- genomeblocksBCR(outblock29, BCR2_29)
genomeblocksBCR2_29

genomeblocksBCR2_all <- rbind(genomeblocksBCR2_1,genomeblocksBCR2_2,genomeblocksBCR2_3,genomeblocksBCR2_4,genomeblocksBCR2_5,genomeblocksBCR2_6,genomeblocksBCR2_7,genomeblocksBCR2_8,genomeblocksBCR2_9,genomeblocksBCR2_10,genomeblocksBCR2_11,genomeblocksBCR2_12,genomeblocksBCR2_13,genomeblocksBCR2_14,genomeblocksBCR2_15,genomeblocksBCR2_16,genomeblocksBCR2_17,genomeblocksBCR2_18,genomeblocksBCR2_19,genomeblocksBCR2_20,genomeblocksBCR2_21,genomeblocksBCR2_22,genomeblocksBCR2_23,genomeblocksBCR2_24,genomeblocksBCR2_25,genomeblocksBCR2_26,genomeblocksBCR2_27,genomeblocksBCR2_28,genomeblocksBCR2_29)

genomeblocksBCR2_all
boxplot(as.numeric(genomeblocksBCR2_all[,"BCR"]) ~ genomeblocksBCR2_all[,"type"])

anova(lm(as.numeric(genomeblocksBCR2_all[,"BCR"]) ~ as.character(genomeblocksBCR2_all[,"type"])))
summary(aov(as.numeric(genomeblocksBCR2_all[,"BCR"]) ~ as.character(genomeblocksBCR2_all[,"type"])))
```


LET'S USE ANDR CENTROMERIC REPEAT

```{r}
BCR3 <- read.csv("BCR3.csv")
#Somewhat similar sequences (blastn), default

BCR3_1 <- BCR3[BCR3$Chromosome=="chr1",]
BCR3_1
BCR3_2 <- BCR3[BCR3$Chromosome=="chr2",]
BCR3_2
BCR3_3 <- BCR3[BCR3$Chromosome=="chr3",]
BCR3_3
BCR3_4 <- BCR3[BCR3$Chromosome=="chr4",]
BCR3_4
BCR3_5 <- BCR3[BCR3$Chromosome=="chr5",]
BCR3_5
BCR3_6 <- BCR3[BCR3$Chromosome=="chr6",]
BCR3_6
BCR3_7 <- BCR3[BCR3$Chromosome=="chr7",]
BCR3_7
BCR3_8 <- BCR3[BCR3$Chromosome=="chr8",]
BCR3_8
BCR3_9 <- BCR3[BCR3$Chromosome=="chr9",]
BCR3_9
BCR3_10 <- BCR3[BCR3$Chromosome=="chr10",]
BCR3_10
BCR3_11 <- BCR3[BCR3$Chromosome=="chr11",]
BCR3_11
BCR3_12 <- BCR3[BCR3$Chromosome=="chr12",]
BCR3_12
BCR3_13 <- BCR3[BCR3$Chromosome=="chr13",]
BCR3_13
BCR3_14 <- BCR3[BCR3$Chromosome=="chr14",]
BCR3_14
BCR3_15 <- BCR3[BCR3$Chromosome=="chr15",]
BCR3_15
BCR3_16 <- BCR3[BCR3$Chromosome=="chr16",]
BCR3_16
BCR3_17 <- BCR3[BCR3$Chromosome=="chr17",]
BCR3_17
BCR3_18 <- BCR3[BCR3$Chromosome=="chr18",]
BCR3_18
BCR3_19 <- BCR3[BCR3$Chromosome=="chr19",]
BCR3_19
BCR3_20 <- BCR3[BCR3$Chromosome=="chr20",]
BCR3_20
BCR3_21 <- BCR3[BCR3$Chromosome=="chr21",]
BCR3_21
BCR3_22 <- BCR3[BCR3$Chromosome=="chr22",]
BCR3_22
BCR3_23 <- BCR3[BCR3$Chromosome=="chr23",]
BCR3_23
BCR3_24 <- BCR3[BCR3$Chromosome=="chr24",]
BCR3_24
BCR3_25 <- BCR3[BCR3$Chromosome=="chr25",]
BCR3_25
BCR3_26 <- BCR3[BCR3$Chromosome=="chr26",]
BCR3_26
BCR3_27 <- BCR3[BCR3$Chromosome=="chr27",]
BCR3_27
BCR3_28 <- BCR3[BCR3$Chromosome=="chr28",]
BCR3_28
BCR3_29 <- BCR3[BCR3$Chromosome=="chr29",]
BCR3_29


genomeblocksBCR3_1 <- genomeblocksBCR(outblock1, BCR3_1)
genomeblocksBCR3_1
genomeblocksBCR3_2 <- genomeblocksBCR(outblock2, BCR3_2)
genomeblocksBCR3_2
genomeblocksBCR3_3 <- genomeblocksBCR(outblock3, BCR3_3)
genomeblocksBCR3_3
genomeblocksBCR3_4 <- genomeblocksBCR(outblock4, BCR3_4)
genomeblocksBCR3_4
genomeblocksBCR3_5 <- genomeblocksBCR(outblock5, BCR3_5)
genomeblocksBCR3_5
genomeblocksBCR3_6 <- genomeblocksBCR(outblock6, BCR3_6)
genomeblocksBCR3_6
genomeblocksBCR3_7 <- genomeblocksBCR(outblock7, BCR3_7)
genomeblocksBCR3_7
genomeblocksBCR3_8 <- genomeblocksBCR(outblock8, BCR3_8)
genomeblocksBCR3_8
genomeblocksBCR3_9 <- genomeblocksBCR(outblock9, BCR3_9)
genomeblocksBCR3_9
genomeblocksBCR3_10 <- genomeblocksBCR(outblock10, BCR3_10)
genomeblocksBCR3_10
genomeblocksBCR3_11 <- genomeblocksBCR(outblock11, BCR3_11)
genomeblocksBCR3_11
genomeblocksBCR3_12 <- genomeblocksBCR(outblock12, BCR3_12)
genomeblocksBCR3_12
genomeblocksBCR3_13 <- genomeblocksBCR(outblock13, BCR3_13)
genomeblocksBCR3_13
genomeblocksBCR3_14 <- genomeblocksBCR(outblock14, BCR3_14)
genomeblocksBCR3_14
genomeblocksBCR3_15 <- genomeblocksBCR(outblock15, BCR3_15)
genomeblocksBCR3_15
genomeblocksBCR3_16 <- genomeblocksBCR(outblock16, BCR3_16)
genomeblocksBCR3_16
genomeblocksBCR3_17 <- genomeblocksBCR(outblock17, BCR3_17)
genomeblocksBCR3_17
genomeblocksBCR3_18 <- genomeblocksBCR(outblock18, BCR3_18)
genomeblocksBCR3_18
genomeblocksBCR3_19 <- genomeblocksBCR(outblock19, BCR3_19)
genomeblocksBCR3_19
genomeblocksBCR3_20 <- genomeblocksBCR(outblock20, BCR3_20)
genomeblocksBCR3_20
genomeblocksBCR3_21 <- genomeblocksBCR(outblock21, BCR3_21)
genomeblocksBCR3_21
genomeblocksBCR3_22 <- genomeblocksBCR(outblock22, BCR3_22)
genomeblocksBCR3_22
genomeblocksBCR3_23 <- genomeblocksBCR(outblock23, BCR3_23)
genomeblocksBCR3_23
genomeblocksBCR3_24 <- genomeblocksBCR(outblock24, BCR3_24)
genomeblocksBCR3_24
genomeblocksBCR3_25 <- genomeblocksBCR(outblock25, BCR3_25)
genomeblocksBCR3_25
genomeblocksBCR3_26 <- genomeblocksBCR(outblock26, BCR3_26)
genomeblocksBCR3_26
genomeblocksBCR3_27 <- genomeblocksBCR(outblock27, BCR3_27)
genomeblocksBCR3_27
genomeblocksBCR3_28 <- genomeblocksBCR(outblock28, BCR3_28)
genomeblocksBCR3_28
genomeblocksBCR3_29 <- genomeblocksBCR(outblock29, BCR3_29)
genomeblocksBCR3_29

genomeblocksBCR3_all <- rbind(genomeblocksBCR3_1,genomeblocksBCR3_2,genomeblocksBCR3_3,genomeblocksBCR3_4,genomeblocksBCR3_5,genomeblocksBCR3_6,genomeblocksBCR3_7,genomeblocksBCR3_8,genomeblocksBCR3_9,genomeblocksBCR3_10,genomeblocksBCR3_11,genomeblocksBCR3_12,genomeblocksBCR3_13,genomeblocksBCR3_14,genomeblocksBCR3_15,genomeblocksBCR3_16,genomeblocksBCR3_17,genomeblocksBCR3_18,genomeblocksBCR3_19,genomeblocksBCR3_20,genomeblocksBCR3_21,genomeblocksBCR3_22,genomeblocksBCR3_23,genomeblocksBCR3_24,genomeblocksBCR3_25,genomeblocksBCR3_26,genomeblocksBCR3_27,genomeblocksBCR3_28,genomeblocksBCR3_29)

genomeblocksBCR3_all
boxplot(as.numeric(genomeblocksBCR3_all[,"BCR"]) ~ genomeblocksBCR3_all[,"type"])

anova(lm(as.numeric(genomeblocksBCR3_all[,"BCR"]) ~ as.character(genomeblocksBCR3_all[,"type"])))
summary(aov(as.numeric(genomeblocksBCR3_all[,"BCR"]) ~ as.character(genomeblocksBCR3_all[,"type"])))



BCR4 <- read.csv("BCR4.csv")
#megablast, default

BCR4_1 <- BCR4[BCR4$Chromosome=="chr1",]
BCR4_1
BCR4_2 <- BCR4[BCR4$Chromosome=="chr2",]
BCR4_2
BCR4_3 <- BCR4[BCR4$Chromosome=="chr3",]
BCR4_3
BCR4_4 <- BCR4[BCR4$Chromosome=="chr4",]
BCR4_4
BCR4_5 <- BCR4[BCR4$Chromosome=="chr5",]
BCR4_5
BCR4_6 <- BCR4[BCR4$Chromosome=="chr6",]
BCR4_6
BCR4_7 <- BCR4[BCR4$Chromosome=="chr7",]
BCR4_7
BCR4_8 <- BCR4[BCR4$Chromosome=="chr8",]
BCR4_8
BCR4_9 <- BCR4[BCR4$Chromosome=="chr9",]
BCR4_9
BCR4_10 <- BCR4[BCR4$Chromosome=="chr10",]
BCR4_10
BCR4_11 <- BCR4[BCR4$Chromosome=="chr11",]
BCR4_11
BCR4_12 <- BCR4[BCR4$Chromosome=="chr12",]
BCR4_12
BCR4_13 <- BCR4[BCR4$Chromosome=="chr13",]
BCR4_13
BCR4_14 <- BCR4[BCR4$Chromosome=="chr14",]
BCR4_14
BCR4_15 <- BCR4[BCR4$Chromosome=="chr15",]
BCR4_15
BCR4_16 <- BCR4[BCR4$Chromosome=="chr16",]
BCR4_16
BCR4_17 <- BCR4[BCR4$Chromosome=="chr17",]
BCR4_17
BCR4_18 <- BCR4[BCR4$Chromosome=="chr18",]
BCR4_18
BCR4_19 <- BCR4[BCR4$Chromosome=="chr19",]
BCR4_19
BCR4_20 <- BCR4[BCR4$Chromosome=="chr20",]
BCR4_20
BCR4_21 <- BCR4[BCR4$Chromosome=="chr21",]
BCR4_21
BCR4_22 <- BCR4[BCR4$Chromosome=="chr22",]
BCR4_22
BCR4_23 <- BCR4[BCR4$Chromosome=="chr23",]
BCR4_23
BCR4_24 <- BCR4[BCR4$Chromosome=="chr24",]
BCR4_24
BCR4_25 <- BCR4[BCR4$Chromosome=="chr25",]
BCR4_25
BCR4_26 <- BCR4[BCR4$Chromosome=="chr26",]
BCR4_26
BCR4_27 <- BCR4[BCR4$Chromosome=="chr27",]
BCR4_27
BCR4_28 <- BCR4[BCR4$Chromosome=="chr28",]
BCR4_28
BCR4_29 <- BCR4[BCR4$Chromosome=="chr29",]
BCR4_29


genomeblocksBCR4_1 <- genomeblocksBCR(outblock1, BCR4_1)
genomeblocksBCR4_1
genomeblocksBCR4_2 <- genomeblocksBCR(outblock2, BCR4_2)
genomeblocksBCR4_2
genomeblocksBCR4_3 <- genomeblocksBCR(outblock3, BCR4_3)
genomeblocksBCR4_3
genomeblocksBCR4_4 <- genomeblocksBCR(outblock4, BCR4_4)
genomeblocksBCR4_4
genomeblocksBCR4_5 <- genomeblocksBCR(outblock5, BCR4_5)
genomeblocksBCR4_5
genomeblocksBCR4_6 <- genomeblocksBCR(outblock6, BCR4_6)
genomeblocksBCR4_6
genomeblocksBCR4_7 <- genomeblocksBCR(outblock7, BCR4_7)
genomeblocksBCR4_7
genomeblocksBCR4_8 <- genomeblocksBCR(outblock8, BCR4_8)
genomeblocksBCR4_8
genomeblocksBCR4_9 <- genomeblocksBCR(outblock9, BCR4_9)
genomeblocksBCR4_9
genomeblocksBCR4_10 <- genomeblocksBCR(outblock10, BCR4_10)
genomeblocksBCR4_10
genomeblocksBCR4_11 <- genomeblocksBCR(outblock11, BCR4_11)
genomeblocksBCR4_11
genomeblocksBCR4_12 <- genomeblocksBCR(outblock12, BCR4_12)
genomeblocksBCR4_12
genomeblocksBCR4_13 <- genomeblocksBCR(outblock13, BCR4_13)
genomeblocksBCR4_13
genomeblocksBCR4_14 <- genomeblocksBCR(outblock14, BCR4_14)
genomeblocksBCR4_14
genomeblocksBCR4_15 <- genomeblocksBCR(outblock15, BCR4_15)
genomeblocksBCR4_15
genomeblocksBCR4_16 <- genomeblocksBCR(outblock16, BCR4_16)
genomeblocksBCR4_16
genomeblocksBCR4_17 <- genomeblocksBCR(outblock17, BCR4_17)
genomeblocksBCR4_17
genomeblocksBCR4_18 <- genomeblocksBCR(outblock18, BCR4_18)
genomeblocksBCR4_18
genomeblocksBCR4_19 <- genomeblocksBCR(outblock19, BCR4_19)
genomeblocksBCR4_19
genomeblocksBCR4_20 <- genomeblocksBCR(outblock20, BCR4_20)
genomeblocksBCR4_20
genomeblocksBCR4_21 <- genomeblocksBCR(outblock21, BCR4_21)
genomeblocksBCR4_21
genomeblocksBCR4_22 <- genomeblocksBCR(outblock22, BCR4_22)
genomeblocksBCR4_22
genomeblocksBCR4_23 <- genomeblocksBCR(outblock23, BCR4_23)
genomeblocksBCR4_23
genomeblocksBCR4_24 <- genomeblocksBCR(outblock24, BCR4_24)
genomeblocksBCR4_24
genomeblocksBCR4_25 <- genomeblocksBCR(outblock25, BCR4_25)
genomeblocksBCR4_25
genomeblocksBCR4_26 <- genomeblocksBCR(outblock26, BCR4_26)
genomeblocksBCR4_26
genomeblocksBCR4_27 <- genomeblocksBCR(outblock27, BCR4_27)
genomeblocksBCR4_27
genomeblocksBCR4_28 <- genomeblocksBCR(outblock28, BCR4_28)
genomeblocksBCR4_28
genomeblocksBCR4_29 <- genomeblocksBCR(outblock29, BCR4_29)
genomeblocksBCR4_29

genomeblocksBCR4_all <- rbind(genomeblocksBCR4_1,genomeblocksBCR4_2,genomeblocksBCR4_3,genomeblocksBCR4_4,genomeblocksBCR4_5,genomeblocksBCR4_6,genomeblocksBCR4_7,genomeblocksBCR4_8,genomeblocksBCR4_9,genomeblocksBCR4_10,genomeblocksBCR4_11,genomeblocksBCR4_12,genomeblocksBCR4_13,genomeblocksBCR4_14,genomeblocksBCR4_15,genomeblocksBCR4_16,genomeblocksBCR4_17,genomeblocksBCR4_18,genomeblocksBCR4_19,genomeblocksBCR4_20,genomeblocksBCR4_21,genomeblocksBCR4_22,genomeblocksBCR4_23,genomeblocksBCR4_24,genomeblocksBCR4_25,genomeblocksBCR4_26,genomeblocksBCR4_27,genomeblocksBCR4_28,genomeblocksBCR4_29)

genomeblocksBCR4_all
boxplot(as.numeric(genomeblocksBCR4_all[,"BCR"]) ~ genomeblocksBCR4_all[,"type"])

anova(lm(as.numeric(genomeblocksBCR4_all[,"BCR"]) ~ as.character(genomeblocksBCR4_all[,"type"])))
summary(aov(as.numeric(genomeblocksBCR4_all[,"BCR"]) ~ as.character(genomeblocksBCR4_all[,"type"])))
```


We are going to use the hits that have duplicates

```{r}

chr1d <- links[links$Chromosome.x=="chr1",]
chr1d <- chr1[order(chr1$chromStart),]
chr1d

chr2d <- links[links$Chromosome.x=="chr2",]
chr2d <- chr2[order(chr2$chromStart),]
chr2d

chr3d <- links[links$Chromosome.x=="chr3",]
chr3d <- chr3[order(chr3$chromStart),]
chr3d

chr4d <- links[links$Chromosome.x=="chr4",]
chr4d <- chr4[order(chr4$chromStart),]
chr4d

chr5d <- links[links$Chromosome.x=="chr5",]
chr5d <- chr5[order(chr5$chromStart),]
chr5d

chr6d <- links[links$Chromosome.x=="chr6",]
chr6d <- chr6[order(chr6$chromStart),]
chr6d

chr7d <- links[links$Chromosome.x=="chr7",]
chr7d <- chr7[order(chr7$chromStart),]
chr7d

chr8d <- links[links$Chromosome.x=="chr8",]
chr8d <- chr8[order(chr8$chromStart),]
chr8d

chr9d <- links[links$Chromosome.x=="chr9",]
chr9d <- chr9[order(chr9$chromStart),]
chr9d

chr10d <- links[links$Chromosome.x=="chr10",]
chr10d <- chr10[order(chr10$chromStart),]
chr10d

chr11d <- links[links$Chromosome.x=="chr11",]
chr11d <- chr11[order(chr11$chromStart),]
chr11d

chr12d <- links[links$Chromosome.x=="chr12",]
chr12d <- chr12[order(chr12$chromStart),]
chr12d

chr13d <- links[links$Chromosome.x=="chr13",]
chr13d <- chr13[order(chr13$chromStart),]
chr13d

chr14d <- links[links$Chromosome.x=="chr14",]
chr14d <- chr14[order(chr14$chromStart),]
chr14d

chr15d <- links[links$Chromosome.x=="chr15",]
chr15d <- chr15[order(chr15$chromStart),]
chr15d

chr16d <- links[links$Chromosome.x=="chr16",]
chr16d <- chr16[order(chr16$chromStart),]
chr16d

chr17d <- links[links$Chromosome.x=="chr17",]
chr17d <- chr17[order(chr17$chromStart),]
chr17d

chr18d <- links[links$Chromosome.x=="chr18",]
chr18d <- chr18[order(chr18$chromStart),]
chr18d

chr19d <- links[links$Chromosome.x=="chr19",]
chr19d <- chr19[order(chr19$chromStart),]
chr19d

chr20d <- links[links$Chromosome.x=="chr20",]
chr20d <- chr20[order(chr20$chromStart),]
chr20d

chr21d <- links[links$Chromosome.x=="chr21",]
chr21d <- chr21[order(chr21$chromStart),]
chr21d

chr22d <- links[links$Chromosome.x=="chr22",]
chr22d <- chr22[order(chr22$chromStart),]
chr22d

chr23d <- links[links$Chromosome.x=="chr23",]
chr23d <- chr23[order(chr23$chromStart),]
chr23d

chr24d <- links[links$Chromosome.x=="chr24",]
chr24d <- chr24[order(chr24$chromStart),]
chr24d

chr25d <- links[links$Chromosome.x=="chr25",]
chr25d <- chr25[order(chr25$chromStart),]
chr25d

chr26d <- links[links$Chromosome.x=="chr26",]
chr26d <- chr26[order(chr26$chromStart),]
chr26d

chr27d <- links[links$Chromosome.x=="chr27",]
chr27d <- chr27[order(chr27$chromStart),]
chr27d

chr28d <- links[links$Chromosome.x=="chr28",]
chr28d <- chr28[order(chr28$chromStart),]
chr28d

chr29d <- links[links$Chromosome.x=="chr29",]
chr29d <- chr29[order(chr29$chromStart),]
chr29d

outblock1d <- genomeblocks(chr1d)
outblock1d
outblock2d <- genomeblocks(chr2d)
outblock2d
outblock3d <- genomeblocks(chr3d)
outblock3d
outblock4d <- genomeblocks(chr4d)
outblock4d
outblock5d <- genomeblocks(chr5d)
outblock5d
outblock6d <- genomeblocks(chr6d)
outblock6d
outblock7d <- genomeblocks(chr7d)
outblock7d
outblock8d <- genomeblocks(chr8d)
outblock8d
outblock9d <- genomeblocks(chr9d)
outblock9d
outblock10d <- genomeblocks(chr10d)
outblock10d
outblock11d <- genomeblocks(chr11d)
outblock11d
outblock12d <- genomeblocks(chr12d)
outblock12d
outblock13d <- genomeblocks(chr13d)
outblock13d
outblock14d <- genomeblocks(chr14d)
outblock14d
outblock15d <- genomeblocks(chr15d)
outblock15d
outblock16d <- genomeblocks(chr16d)
outblock16d
outblock17d <- genomeblocks(chr17d)
outblock17d
outblock18d <- genomeblocks(chr18d)
outblock18d
outblock19d <- genomeblocks(chr19d)
outblock19d
outblock20d <- genomeblocks(chr20d)
outblock20d
outblock21d <- genomeblocks(chr21d)
outblock21d
outblock22d <- genomeblocks(chr22d)
outblock22d
outblock23d <- genomeblocks(chr23d)
outblock23d
outblock24d <- genomeblocks(chr24d)
outblock24d
outblock25d <- genomeblocks(chr25d)
outblock25d
outblock26d <- genomeblocks(chr26d)
outblock26d
outblock27d <- genomeblocks(chr27d)
outblock27d
outblock28d <- genomeblocks(chr28d)
outblock28d
outblock29d <- genomeblocks(chr29d)
outblock29d


#Using Andre BCR and somewhat similar search
genomeblocksBCR3_1d <- genomeblocksBCR(outblock1d, BCR3_1)
genomeblocksBCR3_1d
genomeblocksBCR3_2d <- genomeblocksBCR(outblock2d, BCR3_2)
genomeblocksBCR3_2d
genomeblocksBCR3_3d <- genomeblocksBCR(outblock3d, BCR3_3)
genomeblocksBCR3_3d
genomeblocksBCR3_4d <- genomeblocksBCR(outblock4d, BCR3_4)
genomeblocksBCR3_4d
genomeblocksBCR3_5d <- genomeblocksBCR(outblock5d, BCR3_5)
genomeblocksBCR3_5d
genomeblocksBCR3_6d <- genomeblocksBCR(outblock6d, BCR3_6)
genomeblocksBCR3_6d
genomeblocksBCR3_7d <- genomeblocksBCR(outblock7d, BCR3_7)
genomeblocksBCR3_7d
genomeblocksBCR3_8d <- genomeblocksBCR(outblock8d, BCR3_8)
genomeblocksBCR3_8d
genomeblocksBCR3_9d <- genomeblocksBCR(outblock9d, BCR3_9)
genomeblocksBCR3_9d
genomeblocksBCR3_10d <- genomeblocksBCR(outblock10d, BCR3_10)
genomeblocksBCR3_10d
genomeblocksBCR3_11d <- genomeblocksBCR(outblock11d, BCR3_11)
genomeblocksBCR3_11d
genomeblocksBCR3_12d <- genomeblocksBCR(outblock12d, BCR3_12)
genomeblocksBCR3_12d
genomeblocksBCR3_13d <- genomeblocksBCR(outblock13d, BCR3_13)
genomeblocksBCR3_13d
genomeblocksBCR3_14d <- genomeblocksBCR(outblock14d, BCR3_14)
genomeblocksBCR3_14d
genomeblocksBCR3_15d <- genomeblocksBCR(outblock15d, BCR3_15)
genomeblocksBCR3_15d
genomeblocksBCR3_16d <- genomeblocksBCR(outblock16d, BCR3_16)
genomeblocksBCR3_16d
genomeblocksBCR3_17d <- genomeblocksBCR(outblock17d, BCR3_17)
genomeblocksBCR3_17d
genomeblocksBCR3_18d <- genomeblocksBCR(outblock18d, BCR3_18)
genomeblocksBCR3_18d
genomeblocksBCR3_19d <- genomeblocksBCR(outblock19d, BCR3_19)
genomeblocksBCR3_19d
genomeblocksBCR3_20d <- genomeblocksBCR(outblock20d, BCR3_20)
genomeblocksBCR3_20d
genomeblocksBCR3_21d <- genomeblocksBCR(outblock21d, BCR3_21)
genomeblocksBCR3_21d
genomeblocksBCR3_22d <- genomeblocksBCR(outblock22d, BCR3_22)
genomeblocksBCR3_22d
genomeblocksBCR3_23d <- genomeblocksBCR(outblock23d, BCR3_23)
genomeblocksBCR3_23d
genomeblocksBCR3_24d <- genomeblocksBCR(outblock24d, BCR3_24)
genomeblocksBCR3_24d
genomeblocksBCR3_25d <- genomeblocksBCR(outblock25d, BCR3_25)
genomeblocksBCR3_25d
genomeblocksBCR3_26d <- genomeblocksBCR(outblock26d, BCR3_26)
genomeblocksBCR3_26d
genomeblocksBCR3_27d <- genomeblocksBCR(outblock27d, BCR3_27)
genomeblocksBCR3_27d
genomeblocksBCR3_28d <- genomeblocksBCR(outblock28d, BCR3_28)
genomeblocksBCR3_28d
genomeblocksBCR3_29d <- genomeblocksBCR(outblock29d, BCR3_29)
genomeblocksBCR3_29d

genomeblocksBCR3_alld <- rbind(genomeblocksBCR3_1d,genomeblocksBCR3_2d,genomeblocksBCR3_3d,genomeblocksBCR3_4d,genomeblocksBCR3_5d,genomeblocksBCR3_6d,genomeblocksBCR3_7d,genomeblocksBCR3_8d,genomeblocksBCR3_9d,genomeblocksBCR3_10d,genomeblocksBCR3_11d,genomeblocksBCR3_12d,genomeblocksBCR3_13d,genomeblocksBCR3_14d,genomeblocksBCR3_15d,genomeblocksBCR3_16d,genomeblocksBCR3_17d,genomeblocksBCR3_18d,genomeblocksBCR3_19d,genomeblocksBCR3_20d,genomeblocksBCR3_21d,genomeblocksBCR3_22d,genomeblocksBCR3_23d,genomeblocksBCR3_24d,genomeblocksBCR3_25d,genomeblocksBCR3_26d,genomeblocksBCR3_27d,genomeblocksBCR3_28d,genomeblocksBCR3_29d)

genomeblocksBCR3_alld
boxplot(as.numeric(genomeblocksBCR3_alld[,"BCR"]) ~ genomeblocksBCR3_alld[,"type"])

anova(lm(as.numeric(genomeblocksBCR3_alld[,"BCR"]) ~ as.character(genomeblocksBCR3_alld[,"type"])))
summary(aov(as.numeric(genomeblocksBCR3_alld[,"BCR"]) ~ as.character(genomeblocksBCR3_alld[,"type"])))


```

Now we will see the patterns with CG content

```{r}

source("genomeblocksGC.R")

#y<-chr2GC
#x<-genomeblocksBCR3_2d
#i <- 3
#genomeblocksGC(x,y)

chr1GC <- read.csv("./GCcontent/chr1GCcontent.csv")
genomeblocksBCR3_1dGC <- genomeblocksGC(genomeblocksBCR3_1d,chr1GC)
genomeblocksBCR3_1dGC

chr2GC <- read.csv("./GCcontent/chr2GCcontent.csv")
genomeblocksBCR3_2dGC <- genomeblocksGC(genomeblocksBCR3_2d,chr2GC)
genomeblocksBCR3_2dGC

chr3GC <- read.csv("./GCcontent/chr3GCcontent.csv")
genomeblocksBCR3_3dGC <- genomeblocksGC(genomeblocksBCR3_3d,chr3GC)
genomeblocksBCR3_3dGC

chr4GC <- read.csv("./GCcontent/chr4GCcontent.csv")
genomeblocksBCR3_4dGC <- genomeblocksGC(genomeblocksBCR3_4d,chr4GC)
genomeblocksBCR3_4dGC

chr5GC <- read.csv("./GCcontent/chr5GCcontent.csv")
genomeblocksBCR3_5dGC <- genomeblocksGC(genomeblocksBCR3_5d,chr5GC)
genomeblocksBCR3_5dGC

chr6GC <- read.csv("./GCcontent/chr6GCcontent.csv")
genomeblocksBCR3_6dGC <- genomeblocksGC(genomeblocksBCR3_6d,chr6GC)
genomeblocksBCR3_6dGC

chr7GC <- read.csv("./GCcontent/chr7GCcontent.csv")
genomeblocksBCR3_7dGC <- genomeblocksGC(genomeblocksBCR3_7d,chr7GC)
genomeblocksBCR3_7dGC

chr8GC <- read.csv("./GCcontent/chr8GCcontent.csv")
genomeblocksBCR3_8dGC <- genomeblocksGC(genomeblocksBCR3_8d,chr8GC)
genomeblocksBCR3_8dGC

chr9GC <- read.csv("./GCcontent/chr9GCcontent.csv")
genomeblocksBCR3_9dGC <- genomeblocksGC(genomeblocksBCR3_9d,chr9GC)
genomeblocksBCR3_9dGC

chr10GC <- read.csv("./GCcontent/chr10GCcontent.csv")
genomeblocksBCR3_10dGC <- genomeblocksGC(genomeblocksBCR3_10d,chr10GC)
genomeblocksBCR3_10dGC

chr11GC <- read.csv("./GCcontent/chr11GCcontent.csv")
genomeblocksBCR3_11dGC <- genomeblocksGC(genomeblocksBCR3_11d,chr11GC)
genomeblocksBCR3_11dGC

chr12GC <- read.csv("./GCcontent/chr12GCcontent.csv")
genomeblocksBCR3_12dGC <- genomeblocksGC(genomeblocksBCR3_12d,chr12GC)
genomeblocksBCR3_12dGC

chr13GC <- read.csv("./GCcontent/chr13GCcontent.csv")
genomeblocksBCR3_13dGC <- genomeblocksGC(genomeblocksBCR3_13d,chr13GC)
genomeblocksBCR3_13dGC

chr14GC <- read.csv("./GCcontent/chr14GCcontent.csv")
genomeblocksBCR3_14dGC <- genomeblocksGC(genomeblocksBCR3_14d,chr14GC)
genomeblocksBCR3_14dGC

chr15GC <- read.csv("./GCcontent/chr15GCcontent.csv")
genomeblocksBCR3_15dGC <- genomeblocksGC(genomeblocksBCR3_15d,chr15GC)
genomeblocksBCR3_15dGC

chr16GC <- read.csv("./GCcontent/chr16GCcontent.csv")
genomeblocksBCR3_16dGC <- genomeblocksGC(genomeblocksBCR3_16d,chr16GC)
genomeblocksBCR3_16dGC

chr17GC <- read.csv("./GCcontent/chr17GCcontent.csv")
genomeblocksBCR3_17dGC <- genomeblocksGC(genomeblocksBCR3_17d,chr17GC)
genomeblocksBCR3_17dGC

chr18GC <- read.csv("./GCcontent/chr18GCcontent.csv")
genomeblocksBCR3_18dGC <- genomeblocksGC(genomeblocksBCR3_18d,chr18GC)
genomeblocksBCR3_18dGC

chr19GC <- read.csv("./GCcontent/chr19GCcontent.csv")
genomeblocksBCR3_19dGC <- genomeblocksGC(genomeblocksBCR3_19d,chr19GC)
genomeblocksBCR3_19dGC

chr20GC <- read.csv("./GCcontent/chr20GCcontent.csv")
genomeblocksBCR3_20dGC <- genomeblocksGC(genomeblocksBCR3_20d,chr20GC)
genomeblocksBCR3_20dGC

chr21GC <- read.csv("./GCcontent/chr21GCcontent.csv")
genomeblocksBCR3_21dGC <- genomeblocksGC(genomeblocksBCR3_21d,chr21GC)
genomeblocksBCR3_21dGC

chr22GC <- read.csv("./GCcontent/chr22GCcontent.csv")
genomeblocksBCR3_22dGC <- genomeblocksGC(genomeblocksBCR3_22d,chr22GC)
genomeblocksBCR3_22dGC

chr23GC <- read.csv("./GCcontent/chr23GCcontent.csv")
genomeblocksBCR3_23dGC <- genomeblocksGC(genomeblocksBCR3_23d,chr23GC)
genomeblocksBCR3_23dGC

chr24GC <- read.csv("./GCcontent/chr24GCcontent.csv")
genomeblocksBCR3_24dGC <- genomeblocksGC(genomeblocksBCR3_24d,chr24GC)
genomeblocksBCR3_24dGC

chr25GC <- read.csv("./GCcontent/chr25GCcontent.csv")
genomeblocksBCR3_25dGC <- genomeblocksGC(genomeblocksBCR3_25d,chr25GC)
genomeblocksBCR3_25dGC

chr26GC <- read.csv("./GCcontent/chr26GCcontent.csv")
genomeblocksBCR3_26dGC <- genomeblocksGC(genomeblocksBCR3_26d,chr26GC)
genomeblocksBCR3_26dGC

chr27GC <- read.csv("./GCcontent/chr27GCcontent.csv")
genomeblocksBCR3_27dGC <- genomeblocksGC(genomeblocksBCR3_27d,chr27GC)
genomeblocksBCR3_27dGC

chr28GC <- read.csv("./GCcontent/chr28GCcontent.csv")
genomeblocksBCR3_28dGC <- genomeblocksGC(genomeblocksBCR3_28d,chr28GC)
genomeblocksBCR3_28dGC

chr29GC <- read.csv("./GCcontent/chr29GCcontent.csv")
genomeblocksBCR3_29dGC <- genomeblocksGC(genomeblocksBCR3_29d,chr29GC)
genomeblocksBCR3_29dGC


genomeblocksBCR3_alldGC <- rbind(genomeblocksBCR3_1dGC,genomeblocksBCR3_2dGC,genomeblocksBCR3_3dGC,genomeblocksBCR3_4dGC,genomeblocksBCR3_5dGC,genomeblocksBCR3_6dGC,genomeblocksBCR3_7dGC,genomeblocksBCR3_8dGC,genomeblocksBCR3_9dGC,genomeblocksBCR3_10dGC,genomeblocksBCR3_11dGC,genomeblocksBCR3_12dGC,genomeblocksBCR3_13dGC,genomeblocksBCR3_14dGC,genomeblocksBCR3_15dGC,genomeblocksBCR3_16dGC,genomeblocksBCR3_17dGC,genomeblocksBCR3_18dGC,genomeblocksBCR3_19dGC,genomeblocksBCR3_20dGC,genomeblocksBCR3_21dGC,genomeblocksBCR3_22dGC,genomeblocksBCR3_23dGC,genomeblocksBCR3_24dGC,genomeblocksBCR3_25dGC,genomeblocksBCR3_26dGC,genomeblocksBCR3_27dGC,genomeblocksBCR3_28dGC,genomeblocksBCR3_29dGC)

genomeblocksBCR3_alld
boxplot(as.numeric(genomeblocksBCR3_alldGC[,"GC"]) ~ genomeblocksBCR3_alldGC[,"type"])

anova(lm(as.numeric(genomeblocksBCR3_alldGC[,"GC"]) ~ as.character(genomeblocksBCR3_alldGC[,"type"])))
summary(aov(as.numeric(genomeblocksBCR3_alldGC[,"GC"]) ~ as.character(genomeblocksBCR3_alldGC[,"type"])))

summary(lm(as.numeric(genomeblocksBCR3_alldGC[,"GC"]) ~ as.numeric(genomeblocksBCR3_alldGC[,"BCR"])))

```

Now we are going to check whats going on with genes

```{r}

#Using the genome with duplicates

Genes <- read.csv("genes.csv")
#Somewhat similar sequences (blastn), default

Genes1 <- Genes[Genes$Chromosome=="chr1",]
Genes1
Genes2 <- Genes[Genes$Chromosome=="chr2",]
Genes2
Genes3 <- Genes[Genes$Chromosome=="chr3",]
Genes3
Genes4 <- Genes[Genes$Chromosome=="chr4",]
Genes4
Genes5 <- Genes[Genes$Chromosome=="chr5",]
Genes5
Genes6 <- Genes[Genes$Chromosome=="chr6",]
Genes6
Genes7 <- Genes[Genes$Chromosome=="chr7",]
Genes7
Genes8 <- Genes[Genes$Chromosome=="chr8",]
Genes8
Genes9 <- Genes[Genes$Chromosome=="chr9",]
Genes9
Genes10 <- Genes[Genes$Chromosome=="chr10",]
Genes10
Genes11 <- Genes[Genes$Chromosome=="chr11",]
Genes11
Genes12 <- Genes[Genes$Chromosome=="chr12",]
Genes12
Genes13 <- Genes[Genes$Chromosome=="chr13",]
Genes13
Genes14 <- Genes[Genes$Chromosome=="chr14",]
Genes14
Genes15 <- Genes[Genes$Chromosome=="chr15",]
Genes15
Genes16 <- Genes[Genes$Chromosome=="chr16",]
Genes16
Genes17 <- Genes[Genes$Chromosome=="chr17",]
Genes17
Genes18 <- Genes[Genes$Chromosome=="chr18",]
Genes18
Genes19 <- Genes[Genes$Chromosome=="chr19",]
Genes19
Genes20 <- Genes[Genes$Chromosome=="chr20",]
Genes20
Genes21 <- Genes[Genes$Chromosome=="chr21",]
Genes21
Genes22 <- Genes[Genes$Chromosome=="chr22",]
Genes22
Genes23 <- Genes[Genes$Chromosome=="chr23",]
Genes23
Genes24 <- Genes[Genes$Chromosome=="chr24",]
Genes24
Genes25 <- Genes[Genes$Chromosome=="chr25",]
Genes25
Genes26 <- Genes[Genes$Chromosome=="chr26",]
Genes26
Genes27 <- Genes[Genes$Chromosome=="chr27",]
Genes27
Genes28 <- Genes[Genes$Chromosome=="chr28",]
Genes28
Genes29 <- Genes[Genes$Chromosome=="chr29",]
Genes29


#We can use the same function than with the BCR, something very similar
source("genomeblocksGenes.R")
genomeblocksGenes1d <- genomeblocksGenes(genomeblocksBCR3_1dGC, Genes1)
genomeblocksGenes1d
genomeblocksGenes2d <- genomeblocksGenes(genomeblocksBCR3_2dGC, Genes2)
genomeblocksGenes2d
genomeblocksGenes3d <- genomeblocksGenes(genomeblocksBCR3_3dGC, Genes3)
genomeblocksGenes3d
genomeblocksGenes4d <- genomeblocksGenes(genomeblocksBCR3_4dGC, Genes4)
genomeblocksGenes4d
genomeblocksGenes5d <- genomeblocksGenes(genomeblocksBCR3_5dGC, Genes5)
genomeblocksGenes5d
genomeblocksGenes6d <- genomeblocksGenes(genomeblocksBCR3_6dGC, Genes6)
genomeblocksGenes6d
genomeblocksGenes7d <- genomeblocksGenes(genomeblocksBCR3_7dGC, Genes7)
genomeblocksGenes7d
genomeblocksGenes8d <- genomeblocksGenes(genomeblocksBCR3_8dGC, Genes8)
genomeblocksGenes8d
genomeblocksGenes9d <- genomeblocksGenes(genomeblocksBCR3_9dGC, Genes9)
genomeblocksGenes9d
genomeblocksGenes10d <- genomeblocksGenes(genomeblocksBCR3_10dGC, Genes10)
genomeblocksGenes10d
genomeblocksGenes11d <- genomeblocksGenes(genomeblocksBCR3_11dGC, Genes11)
genomeblocksGenes11d
genomeblocksGenes12d <- genomeblocksGenes(genomeblocksBCR3_12dGC, Genes12)
genomeblocksGenes12d
genomeblocksGenes13d <- genomeblocksGenes(genomeblocksBCR3_13dGC, Genes13)
genomeblocksGenes13d
genomeblocksGenes14d <- genomeblocksGenes(genomeblocksBCR3_14dGC, Genes14)
genomeblocksGenes14d
genomeblocksGenes15d <- genomeblocksGenes(genomeblocksBCR3_15dGC, Genes15)
genomeblocksGenes15d
genomeblocksGenes16d <- genomeblocksGenes(genomeblocksBCR3_16dGC, Genes16)
genomeblocksGenes16d
genomeblocksGenes17d <- genomeblocksGenes(genomeblocksBCR3_17dGC, Genes17)
genomeblocksGenes17d
genomeblocksGenes18d <- genomeblocksGenes(genomeblocksBCR3_18dGC, Genes18)
genomeblocksGenes18d
genomeblocksGenes19d <- genomeblocksGenes(genomeblocksBCR3_19dGC, Genes19)
genomeblocksGenes19d
genomeblocksGenes20d <- genomeblocksGenes(genomeblocksBCR3_20dGC, Genes20)
genomeblocksGenes20d
genomeblocksGenes21d <- genomeblocksGenes(genomeblocksBCR3_21dGC, Genes21)
genomeblocksGenes21d
genomeblocksGenes22d <- genomeblocksGenes(genomeblocksBCR3_22dGC, Genes22)
genomeblocksGenes22d
genomeblocksGenes23d <- genomeblocksGenes(genomeblocksBCR3_23dGC, Genes23)
genomeblocksGenes23d
genomeblocksGenes24d <- genomeblocksGenes(genomeblocksBCR3_24dGC, Genes24)
genomeblocksGenes24d
genomeblocksGenes25d <- genomeblocksGenes(genomeblocksBCR3_25dGC, Genes25)
genomeblocksGenes25d
genomeblocksGenes26d <- genomeblocksGenes(genomeblocksBCR3_26dGC, Genes26)
genomeblocksGenes26d
genomeblocksGenes27d <- genomeblocksGenes(genomeblocksBCR3_27dGC, Genes27)
genomeblocksGenes27d
genomeblocksGenes28d <- genomeblocksGenes(genomeblocksBCR3_28dGC, Genes28)
genomeblocksGenes28d
genomeblocksGenes29d <- genomeblocksGenes(genomeblocksBCR3_29dGC, Genes29)
genomeblocksGenes29d

genomeblocksGenesalld <- rbind(genomeblocksGenes1d,genomeblocksGenes2d,genomeblocksGenes3d,genomeblocksGenes4d,genomeblocksGenes5d,genomeblocksGenes6d,genomeblocksGenes7d,genomeblocksGenes8d,genomeblocksGenes9d,genomeblocksGenes10d,genomeblocksGenes11d,genomeblocksGenes12d,genomeblocksGenes13d,genomeblocksGenes14d,genomeblocksGenes15d,genomeblocksGenes16d,genomeblocksGenes17d,genomeblocksGenes18d,genomeblocksGenes19d,genomeblocksGenes20d,genomeblocksGenes21d,genomeblocksGenes22d,genomeblocksGenes23d,genomeblocksGenes24d,genomeblocksGenes25d,genomeblocksGenes26d,genomeblocksGenes27d,genomeblocksGenes28d,genomeblocksGenes29d)

genomeblocksGenesalld
boxplot(as.numeric(genomeblocksGenesalld[,"Genes"]) ~ genomeblocksGenesalld[,"type"])

anova(lm(as.numeric(genomeblocksGenesalld[,"Genes"]) ~ as.character(genomeblocksGenesalld[,"type"])))
summary(aov(as.numeric(genomeblocksGenesalld[,"Genes"]) ~ as.character(genomeblocksGenesalld[,"type"])))

```


Simulating the rearrangements!

```{r}

source("fisfusmodeling.R") 

genome = cbind(c("LG1", "LG1","LG1", "LG1", "LG1", "LG2", "LG2", "LG2", "LG2", "LG2", "LG3", "LG3", "LG3", "LG3", "LG3"),c(0,1000000, 3000000, 9000000, 10000000, 0,1000000, 3000000, 9000000, 10000000, 0,1000000, 3000000, 9000000, 10000000), c(1,1000001, 3000001, 9000001, 10000001, 1,1000001, 3000001, 9000001, 10000001, 0,1000001, 3000001, 9000001, 10000001), c(NA, "M1","M2","M3",NA,NA,"M4","M5","M6",NA,NA,"M7","M8","M9",NA ))
colnames(genome) <- c("Chromosome","chromStart","chromEnd","Marker")
genome <- as.data.frame(genome)
genome

rearrangements = 10
pfis = 0.5
fisfusmodelling(genome, rearrangements,pfis)

# Carex scoparia and Carex littledalei got separated ca. 23 mya 
# rate of fission based on ChromoSSE 3.4 fissions per million year and 5 fussion per million year in anagenetic
# rate of fission based on ChromoSSE 0.6 fissions per million year and 0.4 fussion per million year in anagenetic
# number of rearrangements are ca. 9.5 events per million year x 46 my (23 + 23 million years)
# pfis = 4/9.4

rearrangements = 435
pfis = 0.5

genome <- rbind(
c("chr1",0,1,NA),  
bowtie2[bowtie2[,1]=="chr1",][order(bowtie2[bowtie2[,1]=="chr1",2]),], 
c(both[1,1],both[1,3]-1,both[1,3],NA), 
c("chr2",0,1,NA),  
bowtie2[bowtie2[,1]=="chr2",][order(bowtie2[bowtie2[,1]=="chr2",2]),], 
c(both[2,1],both[2,3]-1,both[2,3],NA),
c("chr3",0,1,NA),  
bowtie2[bowtie2[,1]=="chr3",][order(bowtie2[bowtie2[,1]=="chr3",2]),], 
c(both[3,1],both[3,3]-1,both[3,3],NA),
c("chr4",0,1,NA),  
bowtie2[bowtie2[,1]=="chr4",][order(bowtie2[bowtie2[,1]=="chr4",2]),], 
c(both[4,1],both[4,3]-1,both[4,3],NA),
c("chr5",0,1,NA),  
bowtie2[bowtie2[,1]=="chr5",][order(bowtie2[bowtie2[,1]=="chr5",2]),], 
c(both[5,1],both[5,3]-1,both[5,3],NA),
c("chr6",0,1,NA),  
bowtie2[bowtie2[,1]=="chr6",][order(bowtie2[bowtie2[,1]=="chr6",2]),], 
c(both[6,1],both[6,3]-1,both[6,3],NA),
c("chr7",0,1,NA),  
bowtie2[bowtie2[,1]=="chr7",][order(bowtie2[bowtie2[,1]=="chr7",2]),], 
c(both[7,1],both[7,3]-1,both[7,3],NA),
c("chr8",0,1,NA),  
bowtie2[bowtie2[,1]=="chr8",][order(bowtie2[bowtie2[,1]=="chr8",2]),], 
c(both[8,1],both[8,3]-1,both[8,3],NA),
c("chr9",0,1,NA),  
bowtie2[bowtie2[,1]=="chr9",][order(bowtie2[bowtie2[,1]=="chr9",2]),], 
c(both[9,1],both[9,3]-1,both[9,3],NA),
c("chr10",0,1,NA),  
bowtie2[bowtie2[,1]=="chr10",][order(bowtie2[bowtie2[,1]=="chr10",2]),], 
c(both[10,1],both[10,3]-1,both[10,3],NA),
c("chr11",0,1,NA),  
bowtie2[bowtie2[,1]=="chr11",][order(bowtie2[bowtie2[,1]=="chr11",2]),], 
c(both[11,1],both[11,3]-1,both[11,3],NA),
c("chr12",0,1,NA),  
bowtie2[bowtie2[,1]=="chr12",][order(bowtie2[bowtie2[,1]=="chr12",2]),], 
c(both[12,1],both[12,3]-1,both[12,3],NA),
c("chr13",0,1,NA),  
bowtie2[bowtie2[,1]=="chr13",][order(bowtie2[bowtie2[,1]=="chr13",2]),], 
c(both[13,1],both[13,3]-1,both[13,3],NA),
c("chr14",0,1,NA),  
bowtie2[bowtie2[,1]=="chr14",][order(bowtie2[bowtie2[,1]=="chr14",2]),], 
c(both[14,1],both[14,3]-1,both[14,3],NA),
c("chr15",0,1,NA),  
bowtie2[bowtie2[,1]=="chr15",][order(bowtie2[bowtie2[,1]=="chr15",2]),], 
c(both[15,1],both[15,3]-1,both[15,3],NA),
c("chr16",0,1,NA),  
bowtie2[bowtie2[,1]=="chr16",][order(bowtie2[bowtie2[,1]=="chr16",2]),], 
c(both[16,1],both[16,3]-1,both[16,3],NA),
c("chr17",0,1,NA),  
bowtie2[bowtie2[,1]=="chr17",][order(bowtie2[bowtie2[,1]=="chr17",2]),], 
c(both[17,1],both[17,3]-1,both[17,3],NA),
c("chr18",0,1,NA),  
bowtie2[bowtie2[,1]=="chr18",][order(bowtie2[bowtie2[,1]=="chr18",2]),], 
c(both[18,1],both[18,3]-1,both[18,3],NA),
c("chr19",0,1,NA),  
bowtie2[bowtie2[,1]=="chr19",][order(bowtie2[bowtie2[,1]=="chr19",2]),], 
c(both[19,1],both[19,3]-1,both[19,3],NA),
c("chr20",0,1,NA),  
bowtie2[bowtie2[,1]=="chr20",][order(bowtie2[bowtie2[,1]=="chr20",2]),], 
c(both[20,1],both[20,3]-1,both[20,3],NA),
c("chr21",0,1,NA),  
bowtie2[bowtie2[,1]=="chr21",][order(bowtie2[bowtie2[,1]=="chr21",2]),], 
c(both[21,1],both[21,3]-1,both[21,3],NA),
c("chr22",0,1,NA),  
bowtie2[bowtie2[,1]=="chr22",][order(bowtie2[bowtie2[,1]=="chr22",2]),], 
c(both[22,1],both[22,3]-1,both[22,3],NA),
c("chr23",0,1,NA),  
bowtie2[bowtie2[,1]=="chr23",][order(bowtie2[bowtie2[,1]=="chr23",2]),], 
c(both[23,1],both[23,3]-1,both[23,3],NA),
c("chr24",0,1,NA),  
bowtie2[bowtie2[,1]=="chr24",][order(bowtie2[bowtie2[,1]=="chr24",2]),], 
c(both[24,1],both[24,3]-1,both[24,3],NA),
c("chr25",0,1,NA),  
bowtie2[bowtie2[,1]=="chr25",][order(bowtie2[bowtie2[,1]=="chr25",2]),], 
c(both[25,1],both[25,3]-1,both[25,3],NA),
c("chr26",0,1,NA),  
bowtie2[bowtie2[,1]=="chr26",][order(bowtie2[bowtie2[,1]=="chr26",2]),], 
c(both[26,1],both[26,3]-1,both[26,3],NA),
c("chr27",0,1,NA),  
bowtie2[bowtie2[,1]=="chr27",][order(bowtie2[bowtie2[,1]=="chr27",2]),], 
c(both[27,1],both[27,3]-1,both[27,3],NA),
c("chr28",0,1,NA),  
bowtie2[bowtie2[,1]=="chr28",][order(bowtie2[bowtie2[,1]=="chr28",2]),], 
c(both[28,1],both[28,3]-1,both[28,3],NA),
c("chr29",0,1,NA),  
bowtie2[bowtie2[,1]=="chr29",][order(bowtie2[bowtie2[,1]=="chr29",2]),], 
c(both[29,1],both[29,3]-1,both[29,3],NA)
)

littledaleimodel <- fisfusmodelling(genome, rearrangements,pfis)

littledaleimodel436 <- littledaleimodel[[436]]

littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[1],1] <- "LG1"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[2],1] <- "LG2"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[3],1] <- "LG3"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[4],1] <- "LG4"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[5],1] <- "LG5"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[6],1] <- "LG6"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[7],1] <- "LG7"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[8],1] <- "LG8"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[9],1] <- "LG9"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[10],1] <- "LG10"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[11],1] <- "LG11"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[12],1] <- "LG12"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[13],1] <- "LG13"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[14],1] <- "LG14"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[15],1] <- "LG15"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[16],1] <- "LG16"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[17],1] <- "LG17"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[18],1] <- "LG18"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[19],1] <- "LG19"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[20],1] <- "LG20"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[21],1] <- "LG21"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[22],1] <- "LG22"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[23],1] <- "LG23"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[24],1] <- "LG24"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[25],1] <- "LG25"
littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[26],1] <- "LG26"
#littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[27],1] <- "LG27"
#littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[28],1] <- "LG28"
#littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[29],1] <- "LG29"
#littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[30],1] <- "LG30"
#littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[31],1] <- "LG31"
#littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[32],1] <- "LG32"
#littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[33],1] <- "LG33"
#littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[34],1] <- "LG34"
#littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[35],1] <- "LG35"
#littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[36],1] <- "LG36"
#littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[37],1] <- "LG37"
#littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[38],1] <- "LG38"
#littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[39],1] <- "LG39"
#littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[40],1] <- "LG40"



both[1:29,]
maximum <- vector()
for (i in 1:length(unique(littledaleimodel436[,1])))
maximum[i] <- max(littledaleimodel436[littledaleimodel436[,1] == unique(littledaleimodel436[,1])[i],3])  
littledaleimodel436genome <- cbind(unique(littledaleimodel436[,1]),0,maximum,NA,NA)
colnames(littledaleimodel436genome) <- colnames(both[1:5])
genomesim <- rbind(both[1:29,1:5],littledaleimodel436genome) 

genomesim <- as.data.frame(genomesim)

library(RCircos)
chr.exclude <- NULL
cyto.info <- genomesim
tracks.inside <- 0
tracks.outside <- 0
RCircos.Set.Core.Components(cyto.info, chr.exclude,tracks.inside, tracks.outside)


rcircos.params <- RCircos.Get.Plot.Parameters()
rcircos.cyto <- RCircos.Get.Plot.Ideogram()
rcircos.position <- RCircos.Get.Plot.Positions()
RCircos.List.Plot.Parameters()

 


RCircos.Set.Plot.Area()
RCircos.Chromosome.Ideogram.Plot()

genome <- genome[complete.cases(genome),]
littledaleimodel436 <- littledaleimodel436[complete.cases(littledaleimodel436),]

linkssim <- merge(genome, littledaleimodel436, by = "Marker", sort = F, no.dups = TRUE)
length(genome[,4])
length(unique(genome[,4]))
dim(linkssim)
linkssim <- unique(linkssim)
linkssim <- linkssim[,c(2,3,4,5,6,7)]
linkssim
linkssim <- as.data.frame(linkssim)
RCircos.Link.Plot(linkssim,track.num=1,TRUE)
#There is an error here. Maybe consider to use other package

chr1dsim <- linkssim[linkssim$Chromosome.x=="chr1",]
chr1dsim
chr2dsim <- linkssim[linkssim$Chromosome.x=="chr2",]
chr2dsim
chr3dsim <- linkssim[linkssim$Chromosome.x=="chr3",]
chr3dsim
chr4dsim <- linkssim[linkssim$Chromosome.x=="chr4",]
chr4dsim
chr5dsim <- linkssim[linkssim$Chromosome.x=="chr5",]
chr5dsim
chr6dsim <- linkssim[linkssim$Chromosome.x=="chr6",]
chr6dsim
chr7dsim <- linkssim[linkssim$Chromosome.x=="chr7",]
chr7dsim
chr8dsim <- linkssim[linkssim$Chromosome.x=="chr8",]
chr8dsim
chr9dsim <- linkssim[linkssim$Chromosome.x=="chr9",]
chr9dsim
chr10dsim <- linkssim[linkssim$Chromosome.x=="chr10",]
chr10dsim
chr11dsim <- linkssim[linkssim$Chromosome.x=="chr11",]
chr11dsim
chr12dsim <- linkssim[linkssim$Chromosome.x=="chr12",]
chr12dsim
chr13dsim <- linkssim[linkssim$Chromosome.x=="chr13",]
chr13dsim
chr14dsim <- linkssim[linkssim$Chromosome.x=="chr14",]
chr14dsim
chr15dsim <- linkssim[linkssim$Chromosome.x=="chr15",]
chr15dsim
chr16dsim <- linkssim[linkssim$Chromosome.x=="chr16",]
chr16dsim
chr17dsim <- linkssim[linkssim$Chromosome.x=="chr17",]
chr17dsim
chr18dsim <- linkssim[linkssim$Chromosome.x=="chr18",]
chr18dsim
chr19dsim <- linkssim[linkssim$Chromosome.x=="chr19",]
chr19dsim
chr20dsim <- linkssim[linkssim$Chromosome.x=="chr20",]
chr20dsim
chr21dsim <- linkssim[linkssim$Chromosome.x=="chr21",]
chr21dsim
chr22dsim <- linkssim[linkssim$Chromosome.x=="chr22",]
chr22dsim
chr23dsim <- linkssim[linkssim$Chromosome.x=="chr23",]
chr23dsim
chr24dsim <- linkssim[linkssim$Chromosome.x=="chr24",]
chr24dsim
chr25dsim <- linkssim[linkssim$Chromosome.x=="chr25",]
chr25dsim
chr26dsim <- linkssim[linkssim$Chromosome.x=="chr26",]
chr26dsim
chr27dsim <- linkssim[linkssim$Chromosome.x=="chr27",]
chr27dsim
chr28dsim <- linkssim[linkssim$Chromosome.x=="chr28",]
chr28dsim
chr29dsim <- linkssim[linkssim$Chromosome.x=="chr29",]
chr29dsim

source("genomeblocks.R")
outblock1dsim <- genomeblocks(chr1dsim)
outblock1dsim
outblock2dsim <- genomeblocks(chr2dsim)
outblock2dsim
outblock3dsim <- genomeblocks(chr3dsim)
outblock3dsim
outblock4dsim <- genomeblocks(chr4dsim)
outblock4dsim
outblock5dsim <- genomeblocks(chr5dsim)
outblock5dsim
outblock6dsim <- genomeblocks(chr6dsim)
outblock6dsim
outblock7dsim <- genomeblocks(chr7dsim)
outblock7dsim
outblock8dsim <- genomeblocks(chr8dsim)
outblock8dsim
outblock9dsim <- genomeblocks(chr9dsim)
outblock9dsim
outblock10dsim <- genomeblocks(chr10dsim)
outblock10dsim
outblock11dsim <- genomeblocks(chr11dsim)
outblock11dsim
outblock12dsim <- genomeblocks(chr12dsim)
outblock12dsim
outblock13dsim <- genomeblocks(chr13dsim)
outblock13dsim
outblock14dsim <- genomeblocks(chr14dsim)
outblock14dsim
outblock15dsim <- genomeblocks(chr15dsim)
outblock15dsim
outblock16dsim <- genomeblocks(chr16dsim)
outblock16dsim
outblock17dsim <- genomeblocks(chr17dsim)
outblock17dsim
outblock18dsim <- genomeblocks(chr18dsim)
outblock18dsim
outblock19dsim <- genomeblocks(chr19dsim)
outblock19dsim
outblock20dsim <- genomeblocks(chr20dsim)
outblock20dsim
outblock21dsim <- genomeblocks(chr21dsim)
outblock21dsim
outblock22dsim <- genomeblocks(chr22dsim)
outblock22dsim
outblock23dsim <- genomeblocks(chr23dsim)
outblock23dsim
outblock24dsim <- genomeblocks(chr24dsim)
outblock24dsim
outblock25dsim <- genomeblocks(chr25dsim)
outblock25dsim
outblock26dsim <- genomeblocks(chr26dsim)
outblock26dsim
outblock27dsim <- genomeblocks(chr27dsim)
outblock27dsim
outblock28dsim <- genomeblocks(chr28dsim)
outblock28dsim
outblock29dsim <- genomeblocks(chr29dsim)
outblock29dsim

outblockdsim <- rbind(outblock1dsim,outblock2dsim,outblock3dsim,outblock4dsim,outblock5dsim,outblock6dsim,outblock7dsim,outblock8dsim,outblock9dsim,outblock10dsim,outblock11dsim,outblock12dsim,outblock13dsim,outblock14dsim,outblock15dsim,outblock16dsim,outblock17dsim,outblock18dsim,outblock19dsim,outblock20dsim,outblock21dsim,outblock22dsim,outblock23dsim,outblock24dsim,outblock25dsim,outblock26dsim,outblock27dsim,outblock28dsim,outblock29dsim)


dim(outblockdsim[outblockdsim[,4]=="conserved",])
dim(outblockdsim[outblockdsim[,4]=="singlemarker",])
dim(outblockdsim[outblockdsim[,4]=="fisfus",])
sum(as.numeric(outblockdsim[outblockdsim[,4]=="conserved",3]) - as.numeric(outblockdsim[outblockdsim[,4]=="conserved",2]))
mean(as.numeric(outblockdsim[outblockdsim[,4]=="conserved",3]) - as.numeric(outblockdsim[outblockdsim[,4]=="conserved",2]))

#Datos reales
outblockd <- rbind(outblock1d,outblock2d,outblock3d,outblock4d,outblock5d,outblock6d,outblock7d,outblock8d,outblock9d,outblock10d,outblock11d,outblock12d,outblock13d,outblock14d,outblock15d,outblock16d,outblock17d,outblock18d,outblock19d,outblock20d,outblock21d,outblock22d,outblock23d,outblock24d,outblock25d,outblock26d,outblock27d,outblock28d,outblock29d)

dim(outblockd[outblockd[,4]=="conserved",])
dim(outblockd[outblockd[,4]=="singlemarker",])
dim(outblockd[outblockd[,4]=="fisfus",])
sum(as.numeric(outblockd[outblockd[,4]=="conserved",3]) - as.numeric(outblockd[outblockd[,4]=="conserved",2]))
mean(as.numeric(outblockd[outblockd[,4]=="conserved",3]) - as.numeric(outblockd[outblockd[,4]=="conserved",2]))


```

Now with permutations

```{r}
source("fisfusmodelingperm.R")
# Carex scoparia and Carex littledalei got separated ca. 23 mya 
# rate of fission based on ChromoSSE 3.4 fissions per million year and 5 fussion per million year in anagenetic
# rate of fission based on ChromoSSE 0.6 fissions per million year and 0.4 fussion per million year in anagenetic
# number of rearrangements are ca. 9.5 events per million year x 46 my (23 + 23 million years)
# pfis = 4/9.4

rearrangements = 435
pfis = 0.5
perm = 100
#make sure the genome object is right

littledaleimodelperm <- fisfusmodellingperm(genome, rearrangements,pfis,perm)
#littledaleimodelperm <- littledaleimodelpermcopy
#littledaleimodelperm <- littledaleimodelpermcopy

littledaleimodelperm[[1]]

for (i in 1:100) { 
for (j in 1:length(unique(littledaleimodelperm[[i+1]][,1])))  
littledaleimodelperm[[i+1]][littledaleimodelperm[[i+1]][,1] == unique(littledaleimodelperm[[i+1]][,1])[j],1] <- paste("LG",j, sep="")
}



genome <- genome[complete.cases(genome),]

for (i in 1:101)
littledaleimodelperm[[i]] <- littledaleimodelperm[[i]][complete.cases(littledaleimodelperm[[i]]),]
littledaleimodelperm[[2]]

linkssimperm <- list()
for (i in 1:100)
linkssimperm[[i]] <- merge(genome, littledaleimodelperm[[i+1]], by = "Marker", sort = F, no.dups = TRUE)
linkssimperm[[2]]

length(genome[,4])
length(unique(genome[,4]))
dim(linkssimperm[[2]])

for (i in 1:100)
linkssimperm[[i]] <- linkssimperm[[i]][,c(2,3,4,5,6,7)]

linkssimperm[[2]]

chr1dsimperm <- list()
for (i in 1:100)
chr1dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr1",]
chr1dsimperm[[2]]

chr2dsimperm <- list()
for (i in 1:100)
chr2dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr2",]
chr3dsimperm <- list()
for (i in 1:100)
chr3dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr3",]
chr4dsimperm <- list()
for (i in 1:100)
chr4dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr4",]
chr5dsimperm <- list()
for (i in 1:100)
chr5dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr5",]
chr6dsimperm <- list()
for (i in 1:100)
chr6dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr6",]
chr7dsimperm <- list()
for (i in 1:100)
chr7dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr7",]
chr8dsimperm <- list()
for (i in 1:100)
chr8dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr8",]
chr9dsimperm <- list()
for (i in 1:100)
chr9dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr9",]
chr10dsimperm <- list()
for (i in 1:100)
chr10dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr10",]
chr11dsimperm <- list()
for (i in 1:100)
chr11dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr11",]
chr12dsimperm <- list()
for (i in 1:100)
chr12dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr12",]
chr13dsimperm <- list()
for (i in 1:100)
chr13dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr13",]
chr14dsimperm <- list()
for (i in 1:100)
chr14dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr14",]
chr15dsimperm <- list()
for (i in 1:100)
chr15dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr15",]
chr16dsimperm <- list()
for (i in 1:100)
chr16dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr16",]
chr17dsimperm <- list()
for (i in 1:100)
chr17dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr17",]
chr18dsimperm <- list()
for (i in 1:100)
chr18dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr18",]
chr19dsimperm <- list()
for (i in 1:100)
chr19dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr19",]
chr20dsimperm <- list()
for (i in 1:100)
chr20dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr20",]
chr21dsimperm <- list()
for (i in 1:100)
chr21dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr21",]
chr22dsimperm <- list()
for (i in 1:100)
chr22dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr22",]
chr23dsimperm <- list()
for (i in 1:100)
chr23dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr23",]
chr24dsimperm <- list()
for (i in 1:100)
chr24dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr24",]
chr25dsimperm <- list()
for (i in 1:100)
chr25dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr25",]
chr26dsimperm <- list()
for (i in 1:100)
chr26dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr26",]
chr27dsimperm <- list()
for (i in 1:100)
chr27dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr27",]
chr28dsimperm <- list()
for (i in 1:100)
chr28dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr28",]
chr29dsimperm <- list()
for (i in 1:100)
chr29dsimperm[[i]] <- linkssimperm[[i]][linkssimperm[[i]][,1]=="chr29",]

source("genomeblocks.R")
outblock1dsimperm <- list()
for (i in 1:100)
outblock1dsimperm[[i]] <- genomeblocks(chr1dsimperm[[i]])
outblock2dsimperm <- list()
for (i in 1:100)
outblock2dsimperm[[i]] <- genomeblocks(chr2dsimperm[[i]])
outblock3dsimperm <- list()
for (i in 1:100)
outblock3dsimperm[[i]] <- genomeblocks(chr3dsimperm[[i]])
outblock4dsimperm <- list()
for (i in 1:100)
outblock4dsimperm[[i]] <- genomeblocks(chr4dsimperm[[i]])
outblock5dsimperm <- list()
for (i in 1:100)
outblock5dsimperm[[i]] <- genomeblocks(chr5dsimperm[[i]])
outblock6dsimperm <- list()
for (i in 1:100)
outblock6dsimperm[[i]] <- genomeblocks(chr6dsimperm[[i]])
outblock7dsimperm <- list()
for (i in 1:100)
outblock7dsimperm[[i]] <- genomeblocks(chr7dsimperm[[i]])
outblock8dsimperm <- list()
for (i in 1:100)
outblock8dsimperm[[i]] <- genomeblocks(chr8dsimperm[[i]])
outblock9dsimperm <- list()
for (i in 1:100)
outblock9dsimperm[[i]] <- genomeblocks(chr9dsimperm[[i]])
outblock10dsimperm <- list()
for (i in 1:100)
outblock10dsimperm[[i]] <- genomeblocks(chr10dsimperm[[i]])
outblock11dsimperm <- list()
for (i in 1:100)
outblock11dsimperm[[i]] <- genomeblocks(chr11dsimperm[[i]])
outblock12dsimperm <- list()
for (i in 1:100)
outblock12dsimperm[[i]] <- genomeblocks(chr12dsimperm[[i]])
outblock13dsimperm <- list()
for (i in 1:100)
outblock13dsimperm[[i]] <- genomeblocks(chr13dsimperm[[i]])
outblock14dsimperm <- list()
for (i in 1:100)
outblock14dsimperm[[i]] <- genomeblocks(chr14dsimperm[[i]])
outblock15dsimperm <- list()
for (i in 1:100)
outblock15dsimperm[[i]] <- genomeblocks(chr15dsimperm[[i]])
outblock16dsimperm <- list()
for (i in 1:100)
outblock16dsimperm[[i]] <- genomeblocks(chr16dsimperm[[i]])
outblock17dsimperm <- list()
for (i in 1:100)
outblock17dsimperm[[i]] <- genomeblocks(chr17dsimperm[[i]])
outblock18dsimperm <- list()
for (i in 1:100)
outblock18dsimperm[[i]] <- genomeblocks(chr18dsimperm[[i]])
outblock19dsimperm <- list()
for (i in 1:100)
outblock19dsimperm[[i]] <- genomeblocks(chr19dsimperm[[i]])
outblock20dsimperm <- list()
for (i in 1:100)
outblock20dsimperm[[i]] <- genomeblocks(chr20dsimperm[[i]])
outblock21dsimperm <- list()
for (i in 1:100)
outblock21dsimperm[[i]] <- genomeblocks(chr21dsimperm[[i]])
outblock22dsimperm <- list()
for (i in 1:100)
outblock22dsimperm[[i]] <- genomeblocks(chr22dsimperm[[i]])
outblock23dsimperm <- list()
for (i in 1:100)
outblock23dsimperm[[i]] <- genomeblocks(chr23dsimperm[[i]])
outblock24dsimperm <- list()
for (i in 1:100)
outblock24dsimperm[[i]] <- genomeblocks(chr24dsimperm[[i]])
outblock25dsimperm <- list()
for (i in 1:100)
outblock25dsimperm[[i]] <- genomeblocks(chr25dsimperm[[i]])
outblock26dsimperm <- list()
for (i in 1:100)
outblock26dsimperm[[i]] <- genomeblocks(chr26dsimperm[[i]])
outblock27dsimperm <- list()
for (i in 1:100)
outblock27dsimperm[[i]] <- genomeblocks(chr27dsimperm[[i]])
outblock28dsimperm <- list()
for (i in 1:100)
outblock28dsimperm[[i]] <- genomeblocks(chr28dsimperm[[i]])
outblock29dsimperm <- list()
for (i in 1:100)
outblock29dsimperm[[i]] <- genomeblocks(chr29dsimperm[[i]])


outblockdsimperm <- list()
for (i in 1:100)
outblockdsimperm[[i]] <- rbind(outblock1dsimperm[[i]],outblock2dsimperm[[i]],outblock3dsimperm[[i]],outblock4dsimperm[[i]],outblock5dsimperm[[i]],outblock6dsimperm[[i]],outblock7dsimperm[[i]],outblock8dsimperm[[i]],outblock9dsimperm[[i]],outblock10dsimperm[[i]],outblock11dsimperm[[i]],outblock12dsimperm[[i]],outblock13dsimperm[[i]],outblock14dsimperm[[i]],outblock15dsimperm[[i]],outblock16dsimperm[[i]],outblock17dsimperm[[i]],outblock18dsimperm[[i]],outblock19dsimperm[[i]],outblock20dsimperm[[i]],outblock21dsimperm[[i]],outblock22dsimperm[[i]],outblock23dsimperm[[i]],outblock24dsimperm[[i]],outblock25dsimperm[[i]],outblock26dsimperm[[i]],outblock27dsimperm[[i]],outblock28dsimperm[[i]],outblock29dsimperm[[i]])

numberofconserved <- vector()
for (i in 1:100)
numberofconserved[i] <- dim(outblockdsimperm[[i]][outblockdsimperm[[i]][,4]=="conserved",])[1]
numberofconserved
numberofsinglemarker <- vector()
for (i in 1:100)
numberofsinglemarker[i] <- dim(outblockdsimperm[[i]][outblockdsimperm[[i]][,4]=="singlemarker",])[1]
numberofsinglemarker
numberoffisfus <- vector()
for (i in 1:100)
numberoffisfus[i] <- dim(outblockdsimperm[[i]][outblockdsimperm[[i]][,4]=="fisfus",])[1]
numberoffisfus
sumconserved <- vector()
for (i in 1:100)
sumconserved[i] <- sum(as.numeric(outblockdsimperm[[i]][outblockdsimperm[[i]][,4]=="conserved",3]) - as.numeric(outblockdsimperm[[i]][outblockdsimperm[[i]][,4]=="conserved",2]))
sumconserved
meanconserved <- vector()
for (i in 1:100)
meanconserved[i] <- mean(as.numeric(outblockdsimperm[[i]][outblockdsimperm[[i]][,4]=="conserved",3]) - as.numeric(outblockdsimperm[[i]][outblockdsimperm[[i]][,4]=="conserved",2]))
meanconserved

#Datos reales
outblockd <- rbind(outblock1d,outblock2d,outblock3d,outblock4d,outblock5d,outblock6d,outblock7d,outblock8d,outblock9d,outblock10d,outblock11d,outblock12d,outblock13d,outblock14d,outblock15d,outblock16d,outblock17d,outblock18d,outblock19d,outblock20d,outblock21d,outblock22d,outblock23d,outblock24d,outblock25d,outblock26d,outblock27d,outblock28d,outblock29d)

dim(outblockd[outblockd[,4]=="conserved",])[1]
dim(outblockd[outblockd[,4]=="singlemarker",])[1]
dim(outblockd[outblockd[,4]=="fisfus",])[1]
sum(as.numeric(outblockd[outblockd[,4]=="conserved",3]) - as.numeric(outblockd[outblockd[,4]=="conserved",2]))
mean(as.numeric(outblockd[outblockd[,4]=="conserved",3]) - as.numeric(outblockd[outblockd[,4]=="conserved",2]))

#make some histograms!!!!
hist(numberofconserved)
abline(v=dim(outblockd[outblockd[,4]=="conserved",])[1])
hist(numberofsinglemarker)
abline(v=dim(outblockd[outblockd[,4]=="singlemarker",])[1])
hist(numberoffisfus)
abline(v=dim(outblockd[outblockd[,4]=="fisfus",])[1])
hist(sumconserved)
abline(v=sum(as.numeric(outblockd[outblockd[,4]=="conserved",3]) - as.numeric(outblockd[outblockd[,4]=="conserved",2])))
hist(meanconserved)
abline(v=mean(as.numeric(outblockd[outblockd[,4]=="conserved",3]) - as.numeric(outblockd[outblockd[,4]=="conserved",2])))
sort(meanconserved)

```

